<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-06-07" />

<title>One-to-N sequence matches in coronavirus genomes</title>

<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/header-attrs-2.1/header-attrs.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="one-to-n-matches_around-CoV-2-plus-GISAID_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/navigation-1.1/tabsets.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/navigation-1.1/codefolding.js"></script>
<link href="one-to-n-matches_around-CoV-2-plus-GISAID_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">One-to-N sequence matches in coronavirus genomes</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-06-07</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co">#### General parameters for the analysis ####</span></span>
<span id="cb1-2"><a href="#cb1-2"></a>reloadImage &lt;-<span class="st"> </span><span class="ot">FALSE</span></span>
<span id="cb1-3"><a href="#cb1-3"></a>quickTest &lt;-<span class="st"> </span><span class="ot">TRUE</span> <span class="co"># if True, only analyse the 2 first features</span></span>
<span id="cb1-4"><a href="#cb1-4"></a></span>
<span id="cb1-5"><a href="#cb1-5"></a></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co">#### Sequence collection ####</span></span>
<span id="cb1-7"><a href="#cb1-7"></a></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="co">## Supported: </span></span>
<span id="cb1-9"><a href="#cb1-9"></a>collections &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb1-10"><a href="#cb1-10"></a>  <span class="st">&quot;around-CoV-2&quot;</span>, </span>
<span id="cb1-11"><a href="#cb1-11"></a>  <span class="st">&quot;selected&quot;</span>,</span>
<span id="cb1-12"><a href="#cb1-12"></a>  <span class="st">&quot;around-CoV-2-plus-GISAID&quot;</span>, </span>
<span id="cb1-13"><a href="#cb1-13"></a>  <span class="st">&quot;selected-plus-GISAID&quot;</span></span>
<span id="cb1-14"><a href="#cb1-14"></a>  )</span>
<span id="cb1-15"><a href="#cb1-15"></a></span>
<span id="cb1-16"><a href="#cb1-16"></a></span>
<span id="cb1-17"><a href="#cb1-17"></a></span>
<span id="cb1-18"><a href="#cb1-18"></a><span class="co"># collection &lt;- &quot;around-CoV-2&quot; # 14 strains</span></span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="co"># collection &lt;- &quot;all&quot; # 16 strains</span></span>
<span id="cb1-20"><a href="#cb1-20"></a><span class="co"># collection &lt;- &quot;selected&quot; # ~60 strains</span></span>
<span id="cb1-21"><a href="#cb1-21"></a>collection &lt;-<span class="st"> &quot;around-CoV-2-plus-GISAID&quot;</span> <span class="co"># 16 strains</span></span>
<span id="cb1-22"><a href="#cb1-22"></a><span class="co"># collection &lt;- &quot;selected-plus-GISAID&quot; # ~40 strains</span></span>
<span id="cb1-23"><a href="#cb1-23"></a><span class="co"># collection &lt;- &quot;all-plus-GISAID&quot; # ~60 strains</span></span>
<span id="cb1-24"><a href="#cb1-24"></a></span>
<span id="cb1-25"><a href="#cb1-25"></a></span>
<span id="cb1-26"><a href="#cb1-26"></a></span>
<span id="cb1-27"><a href="#cb1-27"></a><span class="co">## Choose a collection-specific path for the figures</span></span>
<span id="cb1-28"><a href="#cb1-28"></a>knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">fig.path =</span> <span class="kw">paste0</span>(<span class="st">&#39;figures/one-to-n-matches_&#39;</span>, collection, <span class="st">&#39;/&#39;</span>, collection, <span class="st">&quot;_&quot;</span>))</span>
<span id="cb1-29"><a href="#cb1-29"></a></span>
<span id="cb1-30"><a href="#cb1-30"></a><span class="co">## Note about GIDAID sequences.</span></span>
<span id="cb1-31"><a href="#cb1-31"></a><span class="co">## </span></span>
<span id="cb1-32"><a href="#cb1-32"></a><span class="co">## A few genomes were not available in NCBI Genbank at the time of </span></span>
<span id="cb1-33"><a href="#cb1-33"></a><span class="co">## this analysis, and had to be downloaded from GISAID. These sequences</span></span>
<span id="cb1-34"><a href="#cb1-34"></a><span class="co">##  can however not be redistributed, they should thus be downloaded </span></span>
<span id="cb1-35"><a href="#cb1-35"></a><span class="co">##  manually to reproduce the full trees. </span></span>
<span id="cb1-36"><a href="#cb1-36"></a></span>
<span id="cb1-37"><a href="#cb1-37"></a></span>
<span id="cb1-38"><a href="#cb1-38"></a><span class="co">## Exclude incomplete genomes (i.e. those containing a lof of Ns) to avoid biases in the distance computations</span></span>
<span id="cb1-39"><a href="#cb1-39"></a>excludeIncomplete &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb1-40"><a href="#cb1-40"></a></span>
<span id="cb1-41"><a href="#cb1-41"></a></span>
<span id="cb1-42"><a href="#cb1-42"></a></span>
<span id="cb1-43"><a href="#cb1-43"></a><span class="co">#### Define directories and files ####</span></span>
<span id="cb1-44"><a href="#cb1-44"></a>dir &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">main =</span> <span class="st">&#39;..&#39;</span>)</span>
<span id="cb1-45"><a href="#cb1-45"></a>dir<span class="op">$</span>R &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;scripts/R&quot;</span>)</span>
<span id="cb1-46"><a href="#cb1-46"></a></span>
<span id="cb1-47"><a href="#cb1-47"></a><span class="co">## Input directory for sequences</span></span>
<span id="cb1-48"><a href="#cb1-48"></a>dir<span class="op">$</span>seqdata &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>)</span>
<span id="cb1-49"><a href="#cb1-49"></a><span class="kw">dir.create</span>(dir<span class="op">$</span>seqdata, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-50"><a href="#cb1-50"></a></span>
<span id="cb1-51"><a href="#cb1-51"></a></span>
<span id="cb1-52"><a href="#cb1-52"></a><span class="co">## Create output directory for aligned sequences</span></span>
<span id="cb1-53"><a href="#cb1-53"></a>dir<span class="op">$</span>results &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;results&quot;</span>)</span>
<span id="cb1-54"><a href="#cb1-54"></a><span class="kw">dir.create</span>(dir<span class="op">$</span>results, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-55"><a href="#cb1-55"></a></span>
<span id="cb1-56"><a href="#cb1-56"></a><span class="co">## Instantiate a list for output files</span></span>
<span id="cb1-57"><a href="#cb1-57"></a>outfiles &lt;-<span class="st"> </span><span class="kw">vector</span>()</span>
<span id="cb1-58"><a href="#cb1-58"></a></span>
<span id="cb1-59"><a href="#cb1-59"></a><span class="co">## Memory image</span></span>
<span id="cb1-60"><a href="#cb1-60"></a>dir<span class="op">$</span>images &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;memory_images&quot;</span>)</span>
<span id="cb1-61"><a href="#cb1-61"></a><span class="kw">dir.create</span>(dir<span class="op">$</span>images, <span class="dt">recursive =</span> <span class="ot">TRUE</span>, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-62"><a href="#cb1-62"></a>outfiles[<span class="st">&quot;Memory image&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(</span>
<span id="cb1-63"><a href="#cb1-63"></a>  dir<span class="op">$</span>images, </span>
<span id="cb1-64"><a href="#cb1-64"></a>  <span class="kw">paste0</span>(collection, <span class="st">&quot;_one-to-n_matches.Rdata&quot;</span>))</span>
<span id="cb1-65"><a href="#cb1-65"></a></span>
<span id="cb1-66"><a href="#cb1-66"></a><span class="co">## Input files</span></span>
<span id="cb1-67"><a href="#cb1-67"></a>infiles &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb1-68"><a href="#cb1-68"></a></span>
<span id="cb1-69"><a href="#cb1-69"></a><span class="co">## Output tables</span></span>
<span id="cb1-70"><a href="#cb1-70"></a><span class="co"># di$output &lt;- file.path(dir.main, &quot;&quot;)</span></span>
<span id="cb1-71"><a href="#cb1-71"></a><span class="co"># dir$tables &lt;- </span></span>
<span id="cb1-72"><a href="#cb1-72"></a></span>
<span id="cb1-73"><a href="#cb1-73"></a><span class="co">## Load custom functions</span></span>
<span id="cb1-74"><a href="#cb1-74"></a><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;align_n_to_one.R&quot;</span>))</span>
<span id="cb1-75"><a href="#cb1-75"></a><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;plot_pip_profiles.R&quot;</span>))</span>
<span id="cb1-76"><a href="#cb1-76"></a><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;plot_pip_profiles_from_list.R&quot;</span>))</span>
<span id="cb1-77"><a href="#cb1-77"></a></span>
<span id="cb1-78"><a href="#cb1-78"></a><span class="co">## A unequivocal pattern to identify the reference genome in the sequence names of the input file</span></span>
<span id="cb1-79"><a href="#cb1-79"></a>refPattern &lt;-<span class="st"> &quot;HuCoV2_WH01_2019&quot;</span></span>
<span id="cb1-80"><a href="#cb1-80"></a></span>
<span id="cb1-81"><a href="#cb1-81"></a><span class="co">## Exclude some genomes with a lot of Ns, because they bias the PIP profiles and alignments and trees</span></span>
<span id="cb1-82"><a href="#cb1-82"></a>excludePatterns &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;PnGu-P2S_2019&quot;</span>)</span>
<span id="cb1-83"><a href="#cb1-83"></a></span>
<span id="cb1-84"><a href="#cb1-84"></a></span>
<span id="cb1-85"><a href="#cb1-85"></a></span>
<span id="cb1-86"><a href="#cb1-86"></a></span>
<span id="cb1-87"><a href="#cb1-87"></a><span class="co">#### Features of interest in the reference genome ####</span></span>
<span id="cb1-88"><a href="#cb1-88"></a></span>
<span id="cb1-89"><a href="#cb1-89"></a>features &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb1-90"><a href="#cb1-90"></a></span>
<span id="cb1-91"><a href="#cb1-91"></a></span>
<span id="cb1-92"><a href="#cb1-92"></a><span class="co">#### Specific features  ####</span></span>
<span id="cb1-93"><a href="#cb1-93"></a></span>
<span id="cb1-94"><a href="#cb1-94"></a><span class="co">## Cited from Sun et al (https://doi.org/10.1016/j.molmed.2020.02.008)in Lin et al. https://doi.org/10.20944/preprints202006.0044.v1</span></span>
<span id="cb1-95"><a href="#cb1-95"></a></span>
<span id="cb1-96"><a href="#cb1-96"></a>features[[<span class="st">&#39;Sun_2020&#39;</span>]]  &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">13522</span>, <span class="dt">end =</span>  <span class="dv">23686</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-97"><a href="#cb1-97"></a></span>
<span id="cb1-98"><a href="#cb1-98"></a><span class="co">## S1, first cleavage product of the spike protein</span></span>
<span id="cb1-99"><a href="#cb1-99"></a>features[[<span class="st">&#39;S1&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21599</span>, <span class="dt">end =</span> <span class="dv">23617</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-100"><a href="#cb1-100"></a></span>
<span id="cb1-101"><a href="#cb1-101"></a><span class="co">## Receptor binding domain</span></span>
<span id="cb1-102"><a href="#cb1-102"></a>features[[<span class="st">&#39;RBD&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">22517</span>, <span class="dt">end =</span> <span class="dv">23185</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-103"><a href="#cb1-103"></a></span>
<span id="cb1-104"><a href="#cb1-104"></a><span class="co">## S2, second cleavage product of the spike protein</span></span>
<span id="cb1-105"><a href="#cb1-105"></a>features[[<span class="st">&#39;S2&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">23618</span>, <span class="dt">end =</span> <span class="dv">25381</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-106"><a href="#cb1-106"></a></span>
<span id="cb1-107"><a href="#cb1-107"></a></span>
<span id="cb1-108"><a href="#cb1-108"></a></span>
<span id="cb1-109"><a href="#cb1-109"></a><span class="co">## Regions around the insertions</span></span>
<span id="cb1-110"><a href="#cb1-110"></a>features[[<span class="st">&#39;Ins1-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21647</span>, <span class="dt">end =</span> <span class="dv">21907</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-111"><a href="#cb1-111"></a>features[[<span class="st">&#39;Ins2-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21899</span>, <span class="dt">end =</span> <span class="dv">22156</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-112"><a href="#cb1-112"></a>features[[<span class="st">&#39;Ins3-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21899</span>, <span class="dt">end =</span> <span class="dv">23152</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-113"><a href="#cb1-113"></a>features[[<span class="st">&#39;Ins4-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">23483</span>, <span class="dt">end =</span> <span class="dv">23734</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-114"><a href="#cb1-114"></a>features[[<span class="st">&#39;Ins4-m240&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">23363</span>, <span class="dt">end =</span> <span class="dv">23614</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-115"><a href="#cb1-115"></a></span>
<span id="cb1-116"><a href="#cb1-116"></a><span class="co">## Receptor-binding domain</span></span>
<span id="cb1-117"><a href="#cb1-117"></a></span>
<span id="cb1-118"><a href="#cb1-118"></a><span class="co">## Potential Pangolin origin after Xiao (https://doi.org/10.1101/2020.02.17.951335)</span></span>
<span id="cb1-119"><a href="#cb1-119"></a>features[[<span class="st">&#39;Recomb-Xiao&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">22871</span>, <span class="dt">end =</span> <span class="dv">23092</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-120"><a href="#cb1-120"></a></span>
<span id="cb1-121"><a href="#cb1-121"></a><span class="co">## Recombinant region 1 seen on the PIP profiles</span></span>
<span id="cb1-122"><a href="#cb1-122"></a>features[[<span class="st">&#39;Recomb-reg-1&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">2900</span>, <span class="dt">end =</span> <span class="dv">3800</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-123"><a href="#cb1-123"></a></span>
<span id="cb1-124"><a href="#cb1-124"></a><span class="co">## Recombinant region 2 seen on the PIP profiles</span></span>
<span id="cb1-125"><a href="#cb1-125"></a>features[[<span class="st">&#39;Recomb-reg-2&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21000</span>, <span class="dt">end =</span> <span class="dv">24000</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-126"><a href="#cb1-126"></a></span>
<span id="cb1-127"><a href="#cb1-127"></a></span>
<span id="cb1-128"><a href="#cb1-128"></a><span class="co">## Recombinant region 3 seen on the PIP profiles</span></span>
<span id="cb1-129"><a href="#cb1-129"></a>features[[<span class="st">&#39;Recomb-reg-3&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27500</span>, <span class="dt">end =</span> <span class="dv">28550</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-130"><a href="#cb1-130"></a></span>
<span id="cb1-131"><a href="#cb1-131"></a><span class="co">## Recombinant region 4 in the RBD</span></span>
<span id="cb1-132"><a href="#cb1-132"></a><span class="co">## Slighty wider than Xiao&#39;s limits in order to see the context</span></span>
<span id="cb1-133"><a href="#cb1-133"></a>features[[<span class="st">&#39;Recomb-RBD&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">22700</span>, <span class="dt">end =</span> <span class="dv">23200</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-134"><a href="#cb1-134"></a><span class="co"># features[[&#39;Recomb-RBD&#39;]] &lt;- c(start = 22760, end = 23160, isCDS = FALSE)</span></span>
<span id="cb1-135"><a href="#cb1-135"></a></span>
<span id="cb1-136"><a href="#cb1-136"></a><span class="co">## Annotated coding sequences</span></span>
<span id="cb1-137"><a href="#cb1-137"></a>features[[<span class="st">&#39;CDS-S&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21563</span>, <span class="dt">end =</span> <span class="dv">25384</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>) <span class="co">## Spike gene</span></span>
<span id="cb1-138"><a href="#cb1-138"></a>features[[<span class="st">&#39;CDS-ORF3a&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">25393</span>, <span class="dt">end =</span> <span class="dv">26220</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-139"><a href="#cb1-139"></a>features[[<span class="st">&#39;CDS-E&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">26245</span>, <span class="dt">end =</span> <span class="dv">26472</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-140"><a href="#cb1-140"></a>features[[<span class="st">&#39;CDS-M&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">26523</span>, <span class="dt">end =</span> <span class="dv">27191</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-141"><a href="#cb1-141"></a>features[[<span class="st">&#39;CDS-ORF6&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27202</span>, <span class="dt">end =</span> <span class="dv">27387</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-142"><a href="#cb1-142"></a>features[[<span class="st">&#39;CDS-ORF7a&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27394</span>, <span class="dt">end =</span> <span class="dv">27759</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-143"><a href="#cb1-143"></a>features[[<span class="st">&#39;CDS-ORF8&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27894</span>, <span class="dt">end =</span> <span class="dv">28259</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-144"><a href="#cb1-144"></a>features[[<span class="st">&#39;CDS-N&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">28274</span>, <span class="dt">end =</span> <span class="dv">29533</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-145"><a href="#cb1-145"></a>features[[<span class="st">&#39;CDS-ORF10&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">29558</span>, <span class="dt">end =</span> <span class="dv">29674</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-146"><a href="#cb1-146"></a></span>
<span id="cb1-147"><a href="#cb1-147"></a><span class="co">## Large ORF covering 2 thirds of the genome, coding for polyprotein</span></span>
<span id="cb1-148"><a href="#cb1-148"></a>features[[<span class="st">&#39;CDS-ORF1ab&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">266</span>, <span class="dt">end =</span> <span class="dv">21555</span>, <span class="dt">isCDS =</span> <span class="ot">TRUE</span>)</span>
<span id="cb1-149"><a href="#cb1-149"></a></span>
<span id="cb1-150"><a href="#cb1-150"></a><span class="co">## All the sequences after the big polyprotein-coding ORF1ab</span></span>
<span id="cb1-151"><a href="#cb1-151"></a>features[[<span class="st">&#39;After-ORF1ab&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21556</span>, <span class="dt">end =</span> <span class="dv">29899</span>, <span class="dt">isCDS =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-152"><a href="#cb1-152"></a></span>
<span id="cb1-153"><a href="#cb1-153"></a><span class="co">## Only keep the two first features if quick test has been specified</span></span>
<span id="cb1-154"><a href="#cb1-154"></a><span class="cf">if</span> (quickTest) {</span>
<span id="cb1-155"><a href="#cb1-155"></a>  features &lt;-<span class="st"> </span>features[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</span>
<span id="cb1-156"><a href="#cb1-156"></a>}</span>
<span id="cb1-157"><a href="#cb1-157"></a></span>
<span id="cb1-158"><a href="#cb1-158"></a><span class="co">## Convert feature list into a feature tbale</span></span>
<span id="cb1-159"><a href="#cb1-159"></a>featureTable &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">t</span>(<span class="kw">as.data.frame.list</span>(</span>
<span id="cb1-160"><a href="#cb1-160"></a>  features, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)),</span>
<span id="cb1-161"><a href="#cb1-161"></a>  <span class="dt">row.names =</span> <span class="kw">names</span>(features)  <span class="co"># to avoid replacement of &quot;-&quot; by &quot;.&quot;</span></span>
<span id="cb1-162"><a href="#cb1-162"></a>  )</span>
<span id="cb1-163"><a href="#cb1-163"></a></span>
<span id="cb1-164"><a href="#cb1-164"></a><span class="co"># class(featureTable)</span></span>
<span id="cb1-165"><a href="#cb1-165"></a>featureTable<span class="op">$</span>length &lt;-<span class="st"> </span>featureTable<span class="op">$</span>end <span class="op">-</span><span class="st"> </span>featureTable<span class="op">$</span>start <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb1-166"><a href="#cb1-166"></a><span class="co"># row.names(featureTable)</span></span>
<span id="cb1-167"><a href="#cb1-167"></a><span class="co"># featureTable[&quot;CDS.S&quot;, ]</span></span>
<span id="cb1-168"><a href="#cb1-168"></a></span>
<span id="cb1-169"><a href="#cb1-169"></a>featureTable &lt;-<span class="st"> </span>featureTable[<span class="kw">order</span>(featureTable<span class="op">$</span>start), ]</span>
<span id="cb1-170"><a href="#cb1-170"></a></span>
<span id="cb1-171"><a href="#cb1-171"></a></span>
<span id="cb1-172"><a href="#cb1-172"></a><span class="kw">message</span>(<span class="st">&quot;Analysing &quot;</span>, <span class="kw">length</span>(features), <span class="st">&quot; features&quot;</span>)</span>
<span id="cb1-173"><a href="#cb1-173"></a><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, <span class="kw">names</span>(features)))</span>
<span id="cb1-174"><a href="#cb1-174"></a></span>
<span id="cb1-175"><a href="#cb1-175"></a><span class="co">## Report the parameters</span></span>
<span id="cb1-176"><a href="#cb1-176"></a><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Reference pattern: &quot;</span>, refPattern)</span></code></pre></div>
<div id="strain-collection-around-cov-2-plus-gisaid" class="section level2">
<h2>Strain collection: around-CoV-2-plus-GISAID</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">## Genome dir and files</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="cf">if</span> (<span class="kw">length</span>(<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="st">&quot;GISAID&quot;</span>, <span class="dt">x =</span> collection)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) {</span>
<span id="cb2-3"><a href="#cb2-3"></a>  useGISAID &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb2-4"><a href="#cb2-4"></a>  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;GISAID_genomes&quot;</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a>  <span class="co"># collections &lt;-  paste0(collections, &quot;-plus-GISAID&quot;)</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>  <span class="co"># collection &lt;-  paste0(collection, &quot;-plus-GISAID&quot;)</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>} <span class="cf">else</span> {</span>
<span id="cb2-8"><a href="#cb2-8"></a>  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;genomes&quot;</span>)</span>
<span id="cb2-9"><a href="#cb2-9"></a>}</span>
<span id="cb2-10"><a href="#cb2-10"></a></span>
<span id="cb2-11"><a href="#cb2-11"></a></span>
<span id="cb2-12"><a href="#cb2-12"></a><span class="co">## Define the input sequences</span></span>
<span id="cb2-13"><a href="#cb2-13"></a>infiles<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(</span>
<span id="cb2-14"><a href="#cb2-14"></a>  dir<span class="op">$</span>sequences, </span>
<span id="cb2-15"><a href="#cb2-15"></a>  <span class="kw">paste0</span>(<span class="st">&quot;genomes_&quot;</span>, collection, <span class="st">&quot;.fasta&quot;</span>))</span>
<span id="cb2-16"><a href="#cb2-16"></a></span>
<span id="cb2-17"><a href="#cb2-17"></a><span class="co">## Check if the input  sequence file exists</span></span>
<span id="cb2-18"><a href="#cb2-18"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(infiles<span class="op">$</span>sequences)) {</span>
<span id="cb2-19"><a href="#cb2-19"></a>  <span class="kw">stop</span>(<span class="st">&quot;Genome sequence file is missing&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, infiles<span class="op">$</span>sequences)</span>
<span id="cb2-20"><a href="#cb2-20"></a>} <span class="cf">else</span> {</span>
<span id="cb2-21"><a href="#cb2-21"></a>  <span class="kw">message</span>(<span class="st">&quot;Genome sequence file&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, infiles<span class="op">$</span>sequences)</span>
<span id="cb2-22"><a href="#cb2-22"></a>}</span>
<span id="cb2-23"><a href="#cb2-23"></a></span>
<span id="cb2-24"><a href="#cb2-24"></a><span class="co">#### Load genome sequences ####</span></span>
<span id="cb2-25"><a href="#cb2-25"></a>genomes &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="dt">filepath =</span> infiles<span class="op">$</span>sequences, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb2-26"><a href="#cb2-26"></a></span>
<span id="cb2-27"><a href="#cb2-27"></a><span class="co">## Shorten sequence names by suppressing the fasta comment (after the space)</span></span>
<span id="cb2-28"><a href="#cb2-28"></a><span class="kw">names</span>(genomes) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">&quot; .*&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="dt">x =</span> <span class="kw">names</span>(genomes), <span class="dt">perl =</span> <span class="ot">TRUE</span>)</span>
<span id="cb2-29"><a href="#cb2-29"></a></span>
<span id="cb2-30"><a href="#cb2-30"></a><span class="co">## Exclude genomes</span></span>
<span id="cb2-31"><a href="#cb2-31"></a><span class="cf">if</span> (excludeIncomplete) {</span>
<span id="cb2-32"><a href="#cb2-32"></a>  excludePattern =<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;(&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;)|(&quot;</span>, excludePatterns), <span class="st">&quot;)&quot;</span>)</span>
<span id="cb2-33"><a href="#cb2-33"></a>  excludedstrainNames &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> excludePattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </span>
<span id="cb2-34"><a href="#cb2-34"></a>                              <span class="dt">value =</span> <span class="ot">TRUE</span>, <span class="dt">invert =</span> <span class="ot">FALSE</span>)</span>
<span id="cb2-35"><a href="#cb2-35"></a>  filteredGenomeIndices &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> excludePattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </span>
<span id="cb2-36"><a href="#cb2-36"></a>                                <span class="dt">value =</span> <span class="ot">FALSE</span>, <span class="dt">invert =</span> <span class="ot">TRUE</span>)</span>
<span id="cb2-37"><a href="#cb2-37"></a>  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Excluded &quot;</span>, <span class="kw">length</span>(excludedstrainNames),<span class="st">&quot; genomes: &quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, excludedstrainNames)) </span>
<span id="cb2-38"><a href="#cb2-38"></a>  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Remaining genomes: &quot;</span>, <span class="kw">length</span>(filteredGenomeIndices))</span>
<span id="cb2-39"><a href="#cb2-39"></a>  genomes &lt;-<span class="st"> </span>genomes[filteredGenomeIndices]</span>
<span id="cb2-40"><a href="#cb2-40"></a><span class="co"># names(genomes)</span></span>
<span id="cb2-41"><a href="#cb2-41"></a>}</span>
<span id="cb2-42"><a href="#cb2-42"></a></span>
<span id="cb2-43"><a href="#cb2-43"></a><span class="co">## Report the number of genoomes</span></span>
<span id="cb2-44"><a href="#cb2-44"></a>strainNames &lt;-<span class="st"> </span><span class="kw">names</span>(genomes)</span>
<span id="cb2-45"><a href="#cb2-45"></a>nbStrains &lt;-<span class="st"> </span><span class="kw">length</span>(strainNames)</span>
<span id="cb2-46"><a href="#cb2-46"></a><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Loaded &quot;</span>, nbStrains, <span class="st">&quot; genomes from file &quot;</span>, infiles<span class="op">$</span>sequences)</span>
<span id="cb2-47"><a href="#cb2-47"></a><span class="co"># View(genomes)</span></span>
<span id="cb2-48"><a href="#cb2-48"></a></span>
<span id="cb2-49"><a href="#cb2-49"></a></span>
<span id="cb2-50"><a href="#cb2-50"></a><span class="co">#### Define reference and query genomes ####</span></span>
<span id="cb2-51"><a href="#cb2-51"></a>refStrainName &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> refPattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </span>
<span id="cb2-52"><a href="#cb2-52"></a>                      <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</span>
<span id="cb2-53"><a href="#cb2-53"></a><span class="cf">if</span> (<span class="kw">is.null</span>(refStrainName)) {</span>
<span id="cb2-54"><a href="#cb2-54"></a>  <span class="kw">stop</span>(<span class="st">&quot;Could not identify reference genome with pattern &quot;</span>, refPattern)</span>
<span id="cb2-55"><a href="#cb2-55"></a>}</span>
<span id="cb2-56"><a href="#cb2-56"></a><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Reference genome name: &quot;</span>, refStrainName)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>allFeatureAlignments &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">## Reload memory image if specified in parameters</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="cf">if</span> (reloadImage) {</span>
<span id="cb3-5"><a href="#cb3-5"></a>  <span class="kw">load</span>(outfiles[<span class="st">&quot;Memory image&quot;</span>])</span>
<span id="cb3-6"><a href="#cb3-6"></a>  reloadImage &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb3-7"><a href="#cb3-7"></a>}</span></code></pre></div>
</div>
<div id="parameters" class="section level2">
<h2>Parameters</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co">#### Define a list of parameters ####</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>parameters &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb4-3"><a href="#cb4-3"></a>parameters<span class="op">$</span>collection &lt;-<span class="st"> </span>collection</span>
<span id="cb4-4"><a href="#cb4-4"></a>parameters<span class="op">$</span>refStrainName &lt;-<span class="st"> </span>refStrainName</span>
<span id="cb4-5"><a href="#cb4-5"></a>parameters<span class="op">$</span>nbStrains &lt;-<span class="st"> </span>nbStrains</span>
<span id="cb4-6"><a href="#cb4-6"></a>parameters<span class="op">$</span>sequenceDir &lt;-<span class="st"> </span>dir<span class="op">$</span>sequences</span>
<span id="cb4-7"><a href="#cb4-7"></a>parameters<span class="op">$</span>sequenceFile &lt;-<span class="st"> </span>infiles<span class="op">$</span>sequences</span>
<span id="cb4-8"><a href="#cb4-8"></a>parameters<span class="op">$</span>memoryImage &lt;-<span class="st"> </span>outfiles[<span class="st">&quot;Memory image&quot;</span>]</span>
<span id="cb4-9"><a href="#cb4-9"></a></span>
<span id="cb4-10"><a href="#cb4-10"></a></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">## Exclude metagenomic assemblies</span></span>
<span id="cb4-12"><a href="#cb4-12"></a>parameters<span class="op">$</span>pasDeTambouille &lt;-<span class="st"> </span><span class="ot">FALSE</span></span>
<span id="cb4-13"><a href="#cb4-13"></a><span class="cf">if</span> (parameters<span class="op">$</span>pasDeTambouille) {</span>
<span id="cb4-14"><a href="#cb4-14"></a>  excludePatterns &lt;-<span class="st"> </span><span class="kw">append</span>(excludePatterns, <span class="st">&quot;BtYu-RmYN02_2019&quot;</span>)</span>
<span id="cb4-15"><a href="#cb4-15"></a>}</span>
<span id="cb4-16"><a href="#cb4-16"></a></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="kw">kable</span>(<span class="kw">t</span>(<span class="kw">as.data.frame.list</span>(parameters)), </span>
<span id="cb4-18"><a href="#cb4-18"></a>      <span class="dt">caption =</span> <span class="st">&quot;Parameters of the analysis&quot;</span>,</span>
<span id="cb4-19"><a href="#cb4-19"></a>      <span class="dt">col.names =</span> <span class="st">&quot;Parameter&quot;</span>)</span></code></pre></div>
<table>
<caption>Parameters of the analysis</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">Parameter</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>collection</td>
<td align="left">around-CoV-2-plus-GISAID</td>
</tr>
<tr class="even">
<td>refStrainName</td>
<td align="left">HuCoV2_WH01_2019</td>
</tr>
<tr class="odd">
<td>nbStrains</td>
<td align="left">17</td>
</tr>
<tr class="even">
<td>sequenceDir</td>
<td align="left">../data/GISAID_genomes</td>
</tr>
<tr class="odd">
<td>sequenceFile</td>
<td align="left">../data/GISAID_genomes/genomes_around-CoV-2-plus-GISAID.fasta</td>
</tr>
<tr class="even">
<td>memoryImage</td>
<td align="left">../memory_images/around-CoV-2-plus-GISAID_one-to-n_matches.Rdata</td>
</tr>
<tr class="odd">
<td>pasDeTambouille</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
</div>
<div id="strain-statistics" class="section level2">
<h2>Strain statistics</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co">#### Compute statistics about sequence sizes </span><span class="al">###</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>strainStats &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="dt">n =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genomes),</span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="dt">row.names =</span> <span class="kw">names</span>(genomes),</span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="dt">status =</span> <span class="kw">rep</span>(<span class="st">&quot;Query&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(strainNames))</span>
<span id="cb5-6"><a href="#cb5-6"></a>)</span>
<span id="cb5-7"><a href="#cb5-7"></a>strainStats[,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> </span><span class="kw">as.vector</span>(strainStats[,<span class="st">&quot;status&quot;</span>])</span>
<span id="cb5-8"><a href="#cb5-8"></a>strainStats[refStrainName,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> &quot;Reference&quot;</span></span>
<span id="cb5-9"><a href="#cb5-9"></a>g &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="cf">for</span> (g <span class="cf">in</span> strainNames) {</span>
<span id="cb5-11"><a href="#cb5-11"></a>  strainStats[g, <span class="st">&quot;length&quot;</span>] &lt;-<span class="st"> </span><span class="kw">length</span>(genomes[[g]])</span>
<span id="cb5-12"><a href="#cb5-12"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co">#### Define the color associated to each sequence ####</span></span>
<span id="cb6-2"><a href="#cb6-2"></a></span>
<span id="cb6-3"><a href="#cb6-3"></a></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co">## Color palette per species</span></span>
<span id="cb6-5"><a href="#cb6-5"></a>speciesPalette &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="dt">Human =</span> <span class="st">&quot;#880000&quot;</span>,</span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="dt">Bat =</span> <span class="st">&quot;#888888&quot;</span>,</span>
<span id="cb6-8"><a href="#cb6-8"></a>  <span class="dt">Pangolin =</span> <span class="st">&quot;#447700&quot;</span>,</span>
<span id="cb6-9"><a href="#cb6-9"></a>  <span class="dt">Camel =</span> <span class="st">&quot;#BB8800&quot;</span>,</span>
<span id="cb6-10"><a href="#cb6-10"></a>  <span class="dt">Pig =</span> <span class="st">&quot;#FFBBBB&quot;</span>,</span>
<span id="cb6-11"><a href="#cb6-11"></a>  <span class="dt">Civet =</span> <span class="st">&quot;#00BBFF&quot;</span></span>
<span id="cb6-12"><a href="#cb6-12"></a>)</span>
<span id="cb6-13"><a href="#cb6-13"></a></span>
<span id="cb6-14"><a href="#cb6-14"></a><span class="co">## Species prefix in the tip labels</span></span>
<span id="cb6-15"><a href="#cb6-15"></a>speciesPrefix &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Hu&quot;</span> =<span class="st"> &quot;Human&quot;</span>,</span>
<span id="cb6-16"><a href="#cb6-16"></a>                   <span class="st">&quot;Bt&quot;</span> =<span class="st"> &quot;Bat&quot;</span>,</span>
<span id="cb6-17"><a href="#cb6-17"></a>                   <span class="st">&quot;Pn&quot;</span> =<span class="st"> &quot;Pangolin&quot;</span>,</span>
<span id="cb6-18"><a href="#cb6-18"></a>                   <span class="st">&quot;Cm&quot;</span> =<span class="st"> &quot;Camel&quot;</span>,</span>
<span id="cb6-19"><a href="#cb6-19"></a>                   <span class="st">&quot;Pi&quot;</span> =<span class="st"> &quot;Pig&quot;</span>,</span>
<span id="cb6-20"><a href="#cb6-20"></a>                   <span class="st">&quot;Cv&quot;</span> =<span class="st"> &quot;Civet&quot;</span>)</span>
<span id="cb6-21"><a href="#cb6-21"></a></span>
<span id="cb6-22"><a href="#cb6-22"></a><span class="co">## Strain-specific colors</span></span>
<span id="cb6-23"><a href="#cb6-23"></a>strainColor &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb6-24"><a href="#cb6-24"></a>  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span> =<span class="st"> &quot;red&quot;</span>,</span>
<span id="cb6-25"><a href="#cb6-25"></a>  <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span> =<span class="st"> &quot;#0044BB&quot;</span>,</span>
<span id="cb6-26"><a href="#cb6-26"></a>  <span class="st">&quot;PnGu1_2019&quot;</span> =<span class="st"> &quot;#00CC00&quot;</span>,</span>
<span id="cb6-27"><a href="#cb6-27"></a>  <span class="st">&quot;PnMP789&quot;</span> =<span class="st"> &quot;#00FF88&quot;</span>,</span>
<span id="cb6-28"><a href="#cb6-28"></a>  <span class="st">&quot;BtRaTG13_&quot;</span> =<span class="st"> &quot;#FF6600&quot;</span>,</span>
<span id="cb6-29"><a href="#cb6-29"></a>  <span class="st">&quot;BtYu-RmYN&quot;</span> =<span class="st"> &quot;#FFBB22&quot;</span>,</span>
<span id="cb6-30"><a href="#cb6-30"></a>  <span class="st">&quot;BtZXC21&quot;</span> =<span class="st"> &quot;black&quot;</span>, </span>
<span id="cb6-31"><a href="#cb6-31"></a>  <span class="st">&quot;BtZC45&quot;</span> =<span class="st"> &quot;black&quot;</span>)</span>
<span id="cb6-32"><a href="#cb6-32"></a></span>
<span id="cb6-33"><a href="#cb6-33"></a><span class="co">## Identify species per tip</span></span>
<span id="cb6-34"><a href="#cb6-34"></a><span class="cf">for</span> (prefix <span class="cf">in</span> <span class="kw">names</span>(speciesPrefix)) {</span>
<span id="cb6-35"><a href="#cb6-35"></a>  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, prefix), <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;species&quot;</span>] &lt;-<span class="st"> </span>speciesPrefix[prefix]</span>
<span id="cb6-36"><a href="#cb6-36"></a>  </span>
<span id="cb6-37"><a href="#cb6-37"></a>}</span>
<span id="cb6-38"><a href="#cb6-38"></a></span>
<span id="cb6-39"><a href="#cb6-39"></a><span class="co">## Assign acolor to each species</span></span>
<span id="cb6-40"><a href="#cb6-40"></a>strainStats<span class="op">$</span>color &lt;-<span class="st"> &quot;grey&quot;</span> <span class="co"># default</span></span>
<span id="cb6-41"><a href="#cb6-41"></a>strainStats<span class="op">$</span>color &lt;-<span class="st"> </span>speciesPalette[<span class="kw">as.vector</span>(strainStats<span class="op">$</span>species)]</span>
<span id="cb6-42"><a href="#cb6-42"></a></span>
<span id="cb6-43"><a href="#cb6-43"></a><span class="cf">for</span> (strain <span class="cf">in</span> <span class="kw">names</span>(strainColor)) {</span>
<span id="cb6-44"><a href="#cb6-44"></a>  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, strain), </span>
<span id="cb6-45"><a href="#cb6-45"></a>                  <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;color&quot;</span>] &lt;-<span class="st"> </span>strainColor[strain]</span>
<span id="cb6-46"><a href="#cb6-46"></a>  </span>
<span id="cb6-47"><a href="#cb6-47"></a>}</span>
<span id="cb6-48"><a href="#cb6-48"></a></span>
<span id="cb6-49"><a href="#cb6-49"></a></span>
<span id="cb6-50"><a href="#cb6-50"></a><span class="co">## Assign specific color to some nodes</span></span>
<span id="cb6-51"><a href="#cb6-51"></a></span>
<span id="cb6-52"><a href="#cb6-52"></a><span class="co">## Define a color for each strain</span></span>
<span id="cb6-53"><a href="#cb6-53"></a>strainColors &lt;-<span class="st"> </span>(<span class="kw">unlist</span>(strainStats<span class="op">$</span>color))</span>
<span id="cb6-54"><a href="#cb6-54"></a><span class="kw">names</span>(strainColors) &lt;-<span class="st"> </span><span class="kw">row.names</span>(strainStats)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw">kable</span>(strainStats, <span class="dt">caption =</span> <span class="st">&quot;Reference and query genomes&quot;</span>)</span></code></pre></div>
<table>
<caption>Reference and query genomes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">n</th>
<th align="left">status</th>
<th align="right">length</th>
<th align="left">species</th>
<th align="left">color</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BtBM48-31</td>
<td align="right">1</td>
<td align="left">Query</td>
<td align="right">29276</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtGX2013</td>
<td align="right">2</td>
<td align="left">Query</td>
<td align="right">29161</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="odd">
<td>BtHKU3-12</td>
<td align="right">3</td>
<td align="left">Query</td>
<td align="right">29704</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">4</td>
<td align="left">Query</td>
<td align="right">29855</td>
<td align="left">Bat</td>
<td align="left">#FF6600</td>
</tr>
<tr class="odd">
<td>BtRs4874</td>
<td align="right">5</td>
<td align="left">Query</td>
<td align="right">30311</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtYN2013</td>
<td align="right">6</td>
<td align="left">Query</td>
<td align="right">29142</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="odd">
<td>BtYN2018B</td>
<td align="right">7</td>
<td align="left">Query</td>
<td align="right">30256</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtZC45</td>
<td align="right">8</td>
<td align="left">Query</td>
<td align="right">29802</td>
<td align="left">Bat</td>
<td align="left">black</td>
</tr>
<tr class="odd">
<td>BtZXC21</td>
<td align="right">9</td>
<td align="left">Query</td>
<td align="right">29732</td>
<td align="left">Bat</td>
<td align="left">black</td>
</tr>
<tr class="even">
<td>Cv007-2004</td>
<td align="right">10</td>
<td align="left">Query</td>
<td align="right">29540</td>
<td align="left">Civet</td>
<td align="left">#00BBFF</td>
</tr>
<tr class="odd">
<td>HuCoV2_WH01_2019</td>
<td align="right">11</td>
<td align="left">Reference</td>
<td align="right">29899</td>
<td align="left">Human</td>
<td align="left">red</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="right">12</td>
<td align="left">Query</td>
<td align="right">29727</td>
<td align="left">Human</td>
<td align="left">#0044BB</td>
</tr>
<tr class="odd">
<td>PnGX-P1E_2017</td>
<td align="right">13</td>
<td align="left">Query</td>
<td align="right">29801</td>
<td align="left">Pangolin</td>
<td align="left">#447700</td>
</tr>
<tr class="even">
<td>PnGX-P2V_2018</td>
<td align="right">14</td>
<td align="left">Query</td>
<td align="right">29795</td>
<td align="left">Pangolin</td>
<td align="left">#447700</td>
</tr>
<tr class="odd">
<td>PnMP789</td>
<td align="right">15</td>
<td align="left">Query</td>
<td align="right">29521</td>
<td align="left">Pangolin</td>
<td align="left">#00FF88</td>
</tr>
<tr class="even">
<td>BtYu-RmYN02_2019</td>
<td align="right">16</td>
<td align="left">Query</td>
<td align="right">29671</td>
<td align="left">Bat</td>
<td align="left">#FFBB22</td>
</tr>
<tr class="odd">
<td>PnGu1_2019</td>
<td align="right">17</td>
<td align="left">Query</td>
<td align="right">29825</td>
<td align="left">Pangolin</td>
<td align="left">#00CC00</td>
</tr>
</tbody>
</table>
<p>The collection around-CoV-2-plus-GISAID contains 17 virus genome sequences.</p>
</div>
<div id="features" class="section level2">
<h2>Features</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co">## Print feature coordinates</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw">kable</span>(featureTable, <span class="dt">caption =</span> <span class="st">&quot;Feature coordinates. &quot;</span>)</span></code></pre></div>
<table>
<caption>Feature coordinates.</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">isCDS</th>
<th align="right">length</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Sun_2020</td>
<td align="right">13522</td>
<td align="right">23686</td>
<td align="right">0</td>
<td align="right">10165</td>
</tr>
<tr class="even">
<td>S1</td>
<td align="right">21599</td>
<td align="right">23617</td>
<td align="right">1</td>
<td align="right">2019</td>
</tr>
</tbody>
</table>
</div>
<div id="feature-matches-and-alignments" class="section level2">
<h2>Feature matches and alignments</h2>
<ol style="list-style-type: decimal">
<li><p>We perform a global pairwise alignment (Needle-Waterman algorithm) between each feature of the reference strain (HuCoV2_WH01_2019) and each one of the query genomes.</p></li>
<li><p>The collected nucleic sequences are then aligned based on their translated sequences. This ensures a more relevant placing of the gaps. These translation-based alignments are exported in aminoacids and DNA formats (the DNA sequences are obtained by reverse-mapping the aligned aminoacid sequences onto the input DNA sequences).</p></li>
<li><p>For each alignment (AA and DNA) wee compute a consensus matrix, indicating the frequency of each residue (row of the consensus matrix) in each position of the alignment (column of the matrix).</p></li>
</ol>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co">#### One-to-N alignmemnt of user-speficied genomic features ####</span></span>
<span id="cb9-2"><a href="#cb9-2"></a></span>
<span id="cb9-3"><a href="#cb9-3"></a></span>
<span id="cb9-4"><a href="#cb9-4"></a>i &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb9-5"><a href="#cb9-5"></a>testFeature &lt;-<span class="st"> &quot;Recomb-RBD&quot;</span></span>
<span id="cb9-6"><a href="#cb9-6"></a>i &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">names</span>(features) <span class="op">==</span><span class="st"> </span>testFeature)</span>
<span id="cb9-7"><a href="#cb9-7"></a></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co">## Rune one-ton matches and translation-based mumti^ple alignment + PIP profiles</span></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(features)) {</span>
<span id="cb9-10"><a href="#cb9-10"></a>  featureName &lt;-<span class="st"> </span><span class="kw">names</span>(features)[i]</span>
<span id="cb9-11"><a href="#cb9-11"></a>  </span>
<span id="cb9-12"><a href="#cb9-12"></a>  <span class="co">## Compute feature length</span></span>
<span id="cb9-13"><a href="#cb9-13"></a>  featureLimits &lt;-<span class="st"> </span>features[[featureName]]</span>
<span id="cb9-14"><a href="#cb9-14"></a>  featureStart &lt;-<span class="st"> </span>featureLimits[[<span class="st">&quot;start&quot;</span>]]</span>
<span id="cb9-15"><a href="#cb9-15"></a>  featureEnd &lt;-<span class="st"> </span>featureLimits[[<span class="st">&quot;end&quot;</span>]]</span>
<span id="cb9-16"><a href="#cb9-16"></a>  featureLength &lt;-<span class="st"> </span></span>
<span id="cb9-17"><a href="#cb9-17"></a><span class="st">    </span>featureEnd <span class="op">-</span><span class="st"> </span>featureStart  <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb9-18"><a href="#cb9-18"></a>  </span>
<span id="cb9-19"><a href="#cb9-19"></a>  <span class="kw">message</span>(<span class="st">&quot;Searching matches for feature &quot;</span>, </span>
<span id="cb9-20"><a href="#cb9-20"></a>          i, <span class="st">&quot;/&quot;</span>, <span class="kw">length</span>(features), <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,</span>
<span id="cb9-21"><a href="#cb9-21"></a>          featureName, </span>
<span id="cb9-22"><a href="#cb9-22"></a>          <span class="st">&quot; (&quot;</span>, featureLimits[<span class="dv">1</span>], <span class="st">&quot;-&quot;</span>, featureLimits[<span class="dv">2</span>], <span class="st">&quot;)&quot;</span>, </span>
<span id="cb9-23"><a href="#cb9-23"></a>          <span class="st">&quot; in collection &quot;</span>, collection)</span>
<span id="cb9-24"><a href="#cb9-24"></a>  </span>
<span id="cb9-25"><a href="#cb9-25"></a>  <span class="co">## Print a section title with the feature name and limits</span></span>
<span id="cb9-26"><a href="#cb9-26"></a>  <span class="kw">cat</span>(<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">### &quot;</span>,  featureName, </span>
<span id="cb9-27"><a href="#cb9-27"></a>      <span class="st">&quot; (&quot;</span>, featureLimits[<span class="dv">1</span>], <span class="st">&quot;-&quot;</span>, featureLimits[<span class="dv">2</span>], </span>
<span id="cb9-28"><a href="#cb9-28"></a>      <span class="st">&quot;; &quot;</span>, featureLength,<span class="st">&quot;bp)&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb9-29"><a href="#cb9-29"></a></span>
<span id="cb9-30"><a href="#cb9-30"></a></span>
<span id="cb9-31"><a href="#cb9-31"></a>  <span class="co">## Choose a collection- and feature-specific path for the figures</span></span>
<span id="cb9-32"><a href="#cb9-32"></a>  knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">fig.path =</span> <span class="kw">paste0</span>(</span>
<span id="cb9-33"><a href="#cb9-33"></a>    <span class="st">&#39;figures/one-to-n-matches_&#39;</span>, collection, <span class="st">&#39;/&#39;</span>,</span>
<span id="cb9-34"><a href="#cb9-34"></a>    collection, <span class="st">&#39;_vs_&#39;</span>, featureName))</span>
<span id="cb9-35"><a href="#cb9-35"></a></span>
<span id="cb9-36"><a href="#cb9-36"></a>  </span>
<span id="cb9-37"><a href="#cb9-37"></a>  <span class="co">#### Define feature-specific output directory ####</span></span>
<span id="cb9-38"><a href="#cb9-38"></a>  featurePrefix &lt;-<span class="st"> </span><span class="kw">paste0</span>(</span>
<span id="cb9-39"><a href="#cb9-39"></a>    featureName,</span>
<span id="cb9-40"><a href="#cb9-40"></a>    <span class="st">&quot;_&quot;</span>, collection)</span>
<span id="cb9-41"><a href="#cb9-41"></a>  </span>
<span id="cb9-42"><a href="#cb9-42"></a>  dir[[featureName]] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>results, featurePrefix)</span>
<span id="cb9-43"><a href="#cb9-43"></a>  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Output directory: &quot;</span>, dir[[featureName]] )</span>
<span id="cb9-44"><a href="#cb9-44"></a>  <span class="kw">dir.create</span>(dir[[featureName]], <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</span>
<span id="cb9-45"><a href="#cb9-45"></a>  </span>
<span id="cb9-46"><a href="#cb9-46"></a>  <span class="co">#### Find feature matches by 1-to-1 alignment with each genome ####</span></span>
<span id="cb9-47"><a href="#cb9-47"></a></span>
<span id="cb9-48"><a href="#cb9-48"></a>  <span class="co">## Define output file for unaligned matching sequences</span></span>
<span id="cb9-49"><a href="#cb9-49"></a>  outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_matches&quot;</span>)]  &lt;-<span class="st"> </span><span class="kw">file.path</span>(</span>
<span id="cb9-50"><a href="#cb9-50"></a>    dir[[featureName]], <span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_matches.fasta&quot;</span>))</span>
<span id="cb9-51"><a href="#cb9-51"></a>  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Matching sequences: &quot;</span>, </span>
<span id="cb9-52"><a href="#cb9-52"></a>          outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_matches&quot;</span>)])</span>
<span id="cb9-53"><a href="#cb9-53"></a>  </span>
<span id="cb9-54"><a href="#cb9-54"></a>  <span class="co">## Get sequence of the feature from the reference genome</span></span>
<span id="cb9-55"><a href="#cb9-55"></a>  refSeq &lt;-<span class="st"> </span><span class="kw">subseq</span>(genomes[refStrainName], </span>
<span id="cb9-56"><a href="#cb9-56"></a>                   <span class="dt">start =</span> featureLimits[<span class="dv">1</span>], </span>
<span id="cb9-57"><a href="#cb9-57"></a>                   <span class="dt">end =</span> featureLimits[<span class="dv">2</span>])</span>
<span id="cb9-58"><a href="#cb9-58"></a></span>
<span id="cb9-59"><a href="#cb9-59"></a>  <span class="co">## Match the reference feature with all the genomes</span></span>
<span id="cb9-60"><a href="#cb9-60"></a>  <span class="cf">if</span> (reloadImage) {</span>
<span id="cb9-61"><a href="#cb9-61"></a>    featureAlignmentsNto1 &lt;-<span class="st"> </span>allFeatureAlignments[[featureName]]</span>
<span id="cb9-62"><a href="#cb9-62"></a>  } <span class="cf">else</span> {  </span>
<span id="cb9-63"><a href="#cb9-63"></a>    featureAlignmentsNto1 &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</span>
<span id="cb9-64"><a href="#cb9-64"></a>      <span class="dt">refSequence =</span> refSeq, </span>
<span id="cb9-65"><a href="#cb9-65"></a>      <span class="dt">querySequences =</span> genomes, </span>
<span id="cb9-66"><a href="#cb9-66"></a>      <span class="dt">type =</span> <span class="st">&quot;global-local&quot;</span>,</span>
<span id="cb9-67"><a href="#cb9-67"></a>      <span class="dt">outfile =</span> outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_matches&quot;</span>)])</span>
<span id="cb9-68"><a href="#cb9-68"></a>    allFeatureAlignments[[featureName]] &lt;-<span class="st"> </span>featureAlignmentsNto1</span>
<span id="cb9-69"><a href="#cb9-69"></a>    </span>
<span id="cb9-70"><a href="#cb9-70"></a>  }</span>
<span id="cb9-71"><a href="#cb9-71"></a>  </span>
<span id="cb9-72"><a href="#cb9-72"></a>  </span>
<span id="cb9-73"><a href="#cb9-73"></a>  </span>
<span id="cb9-74"><a href="#cb9-74"></a>   </span>
<span id="cb9-75"><a href="#cb9-75"></a>  </span>
<span id="cb9-76"><a href="#cb9-76"></a>  </span>
<span id="cb9-77"><a href="#cb9-77"></a>  <span class="co">#### Translation-based alignments ####</span></span>
<span id="cb9-78"><a href="#cb9-78"></a>  <span class="cf">if</span> (featureTable[featureName, <span class="st">&quot;isCDS&quot;</span>]) {</span>
<span id="cb9-79"><a href="#cb9-79"></a>    <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Translation-based multiple alignment of feature &quot;</span>, featureName, <span class="st">&quot;; DNA output&quot;</span>)</span>
<span id="cb9-80"><a href="#cb9-80"></a>    </span>
<span id="cb9-81"><a href="#cb9-81"></a>    <span class="co">#### Align translated sequenecs and return correponding nucleic sequences ####</span></span>
<span id="cb9-82"><a href="#cb9-82"></a>    featureAlignmentsNto1<span class="op">$</span>tralignedDNA &lt;-<span class="st"> </span><span class="kw">AlignTranslation</span>(</span>
<span id="cb9-83"><a href="#cb9-83"></a>      <span class="kw">DNAStringSet</span>(featureAlignmentsNto1<span class="op">$</span>sequences), </span>
<span id="cb9-84"><a href="#cb9-84"></a>      <span class="dt">type =</span> <span class="st">&quot;DNAStringSet&quot;</span>,</span>
<span id="cb9-85"><a href="#cb9-85"></a>      <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb9-86"><a href="#cb9-86"></a>    </span>
<span id="cb9-87"><a href="#cb9-87"></a>    <span class="co">## Save multiple-aligned DNA sequences</span></span>
<span id="cb9-88"><a href="#cb9-88"></a>    outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedDNA&quot;</span>)]  &lt;-<span class="st"> </span><span class="kw">file.path</span>(</span>
<span id="cb9-89"><a href="#cb9-89"></a>      dir[[featureName]], <span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedDNA.fasta&quot;</span>))</span>
<span id="cb9-90"><a href="#cb9-90"></a>    <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Translation-based alignment saved in file &quot;</span>, </span>
<span id="cb9-91"><a href="#cb9-91"></a>            outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedDNA&quot;</span>)])</span>
<span id="cb9-92"><a href="#cb9-92"></a>    <span class="kw">writeXStringSet</span>(</span>
<span id="cb9-93"><a href="#cb9-93"></a>      <span class="dt">x =</span> featureAlignmentsNto1<span class="op">$</span>tralignedDNA, </span>
<span id="cb9-94"><a href="#cb9-94"></a>      <span class="dt">filepath =</span> outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedDNA&quot;</span>)] )</span>
<span id="cb9-95"><a href="#cb9-95"></a></span>
<span id="cb9-96"><a href="#cb9-96"></a>    <span class="co">## Convert alignment to position-weight matrix</span></span>
<span id="cb9-97"><a href="#cb9-97"></a>    featureAlignmentsNto1<span class="op">$</span>tralignedDNAConsensus &lt;-<span class="st"> </span><span class="kw">consensusMatrix</span>(</span>
<span id="cb9-98"><a href="#cb9-98"></a>      <span class="dt">x =</span> featureAlignmentsNto1<span class="op">$</span>tralignedDNA)</span>
<span id="cb9-99"><a href="#cb9-99"></a>    </span>
<span id="cb9-100"><a href="#cb9-100"></a>    <span class="co">#### Align translated sequenecs and return correponding AA sequences ####</span></span>
<span id="cb9-101"><a href="#cb9-101"></a>    <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Translation-based multiple alignment of feature &quot;</span>, featureName, <span class="st">&quot;; AA output&quot;</span>)</span>
<span id="cb9-102"><a href="#cb9-102"></a>    featureAlignmentsNto1<span class="op">$</span>tralignedAA &lt;-<span class="st"> </span><span class="kw">AlignTranslation</span>(</span>
<span id="cb9-103"><a href="#cb9-103"></a>      <span class="kw">DNAStringSet</span>(featureAlignmentsNto1<span class="op">$</span>sequences), </span>
<span id="cb9-104"><a href="#cb9-104"></a>      <span class="dt">type =</span> <span class="st">&quot;AAStringSet&quot;</span>,</span>
<span id="cb9-105"><a href="#cb9-105"></a>      <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb9-106"><a href="#cb9-106"></a></span>
<span id="cb9-107"><a href="#cb9-107"></a>    <span class="co">## Convert alignment to position-weight matrix</span></span>
<span id="cb9-108"><a href="#cb9-108"></a>    featureAlignmentsNto1<span class="op">$</span>tralignedAAConsensus &lt;-<span class="st"> </span><span class="kw">consensusMatrix</span>(</span>
<span id="cb9-109"><a href="#cb9-109"></a>      <span class="dt">x =</span> featureAlignmentsNto1<span class="op">$</span>tralignedAA)</span>
<span id="cb9-110"><a href="#cb9-110"></a>  </span>
<span id="cb9-111"><a href="#cb9-111"></a>    <span class="co">## Save multiple-aligned AA sequences</span></span>
<span id="cb9-112"><a href="#cb9-112"></a>    outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedAA&quot;</span>)]  &lt;-<span class="st"> </span><span class="kw">file.path</span>(</span>
<span id="cb9-113"><a href="#cb9-113"></a>      dir[[featureName]], <span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedAA.fasta&quot;</span>))</span>
<span id="cb9-114"><a href="#cb9-114"></a>    <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Translation-based alignment saved in file &quot;</span>, </span>
<span id="cb9-115"><a href="#cb9-115"></a>            outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedAA&quot;</span>)])</span>
<span id="cb9-116"><a href="#cb9-116"></a>    <span class="kw">writeXStringSet</span>(</span>
<span id="cb9-117"><a href="#cb9-117"></a>      <span class="dt">x =</span> featureAlignmentsNto1<span class="op">$</span>tralignedAA, </span>
<span id="cb9-118"><a href="#cb9-118"></a>      <span class="dt">filepath =</span> outfiles[<span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;_tralignedAA&quot;</span>)] )</span>
<span id="cb9-119"><a href="#cb9-119"></a>  }</span>
<span id="cb9-120"><a href="#cb9-120"></a>  </span>
<span id="cb9-121"><a href="#cb9-121"></a>  <span class="co">## Compute sequence order by decreasing PIP score</span></span>
<span id="cb9-122"><a href="#cb9-122"></a>  seqOrder &lt;-<span class="st"> </span><span class="kw">order</span>(featureAlignmentsNto1<span class="op">$</span>stat<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</span>
<span id="cb9-123"><a href="#cb9-123"></a>  </span>
<span id="cb9-124"><a href="#cb9-124"></a></span>
<span id="cb9-125"><a href="#cb9-125"></a>  <span class="co">## Print matching statistics</span></span>
<span id="cb9-126"><a href="#cb9-126"></a>  <span class="kw">kable</span>(featureAlignmentsNto1<span class="op">$</span>stats[seqOrder, ], </span>
<span id="cb9-127"><a href="#cb9-127"></a>        <span class="dt">caption =</span> <span class="kw">paste0</span>(</span>
<span id="cb9-128"><a href="#cb9-128"></a>          <span class="st">&quot;One-to-N alignment of feature &quot;</span>, </span>
<span id="cb9-129"><a href="#cb9-129"></a>          featureName))</span>
<span id="cb9-130"><a href="#cb9-130"></a>  </span>
<span id="cb9-131"><a href="#cb9-131"></a>  <span class="co">#### PIP profiles ####</span></span>
<span id="cb9-132"><a href="#cb9-132"></a>  <span class="co">## Choose window size  </span></span>
<span id="cb9-133"><a href="#cb9-133"></a>  <span class="cf">if</span> (featureLength <span class="op">&gt;</span><span class="st"> </span><span class="dv">20000</span>) {</span>
<span id="cb9-134"><a href="#cb9-134"></a>    windowSize &lt;-<span class="st"> </span><span class="dv">800</span> </span>
<span id="cb9-135"><a href="#cb9-135"></a>  } <span class="cf">else</span> <span class="cf">if</span> (featureLength <span class="op">&gt;</span><span class="st"> </span><span class="dv">5000</span>) {</span>
<span id="cb9-136"><a href="#cb9-136"></a>    windowSize &lt;-<span class="st"> </span><span class="dv">500</span> </span>
<span id="cb9-137"><a href="#cb9-137"></a>  } <span class="cf">else</span> <span class="cf">if</span> (featureLength <span class="op">&gt;</span><span class="st"> </span><span class="dv">3000</span>) {</span>
<span id="cb9-138"><a href="#cb9-138"></a>    windowSize &lt;-<span class="st"> </span><span class="dv">200</span> </span>
<span id="cb9-139"><a href="#cb9-139"></a>  } <span class="cf">else</span> <span class="cf">if</span> (featureLength <span class="op">&gt;</span><span class="st"> </span><span class="dv">400</span>) {</span>
<span id="cb9-140"><a href="#cb9-140"></a>    windowSize &lt;-<span class="st"> </span><span class="dv">100</span> </span>
<span id="cb9-141"><a href="#cb9-141"></a>  } <span class="cf">else</span> {</span>
<span id="cb9-142"><a href="#cb9-142"></a>    windowSize &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="dv">50</span>, <span class="dv">10</span> <span class="op">*</span><span class="st"> </span><span class="kw">round</span>(featureLength <span class="op">/</span><span class="st"> </span><span class="dv">100</span>))</span>
<span id="cb9-143"><a href="#cb9-143"></a>  }</span>
<span id="cb9-144"><a href="#cb9-144"></a></span>
<span id="cb9-145"><a href="#cb9-145"></a>  </span>
<span id="cb9-146"><a href="#cb9-146"></a>  <span class="co">## DNA PIP profile of one-to-N alignment</span></span>
<span id="cb9-147"><a href="#cb9-147"></a>  <span class="kw">PlotPIPprofilesFromList</span>(</span>
<span id="cb9-148"><a href="#cb9-148"></a><span class="co">#    alignments = featureAlignmentsNto1$alignments[seqOrder],</span></span>
<span id="cb9-149"><a href="#cb9-149"></a>    <span class="dt">alignments =</span> featureAlignmentsNto1<span class="op">$</span>alignments,</span>
<span id="cb9-150"><a href="#cb9-150"></a>    <span class="dt">reversePlot =</span> <span class="ot">TRUE</span>,</span>
<span id="cb9-151"><a href="#cb9-151"></a>    <span class="dt">windowSize =</span> windowSize,</span>
<span id="cb9-152"><a href="#cb9-152"></a>    <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;PIP profiles: collection &quot;</span>, collection,</span>
<span id="cb9-153"><a href="#cb9-153"></a>                  <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Genomic feature &quot;</span>, featureName, <span class="st">&quot; (&quot;</span>, featureStart, <span class="st">&quot;:&quot;</span>, featureEnd ,<span class="st">&quot;, &quot;</span>, featureLength, <span class="st">&quot; bp)&quot;</span>),</span>
<span id="cb9-154"><a href="#cb9-154"></a>    <span class="co">#    colors = NULL,</span></span>
<span id="cb9-155"><a href="#cb9-155"></a>    <span class="dt">colors =</span> strainColors,</span>
<span id="cb9-156"><a href="#cb9-156"></a>    <span class="dt">legendMargin =</span> <span class="fl">0.25</span>,</span>
<span id="cb9-157"><a href="#cb9-157"></a>    <span class="dt">legendCorner =</span> <span class="st">&quot;topright&quot;</span>, </span>
<span id="cb9-158"><a href="#cb9-158"></a>    <span class="dt">lwd =</span> <span class="dv">1</span>,</span>
<span id="cb9-159"><a href="#cb9-159"></a>    <span class="dt">legendCex =</span> <span class="fl">0.5</span>, </span>
<span id="cb9-160"><a href="#cb9-160"></a>    <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">100</span>))</span>
<span id="cb9-161"><a href="#cb9-161"></a></span>
<span id="cb9-162"><a href="#cb9-162"></a>  <span class="co">## Identify the correspodences between matches and sequences</span></span>
<span id="cb9-163"><a href="#cb9-163"></a>  seqNames &lt;-<span class="st"> </span><span class="kw">names</span>(featureAlignmentsNto1<span class="op">$</span>alignments)</span>
<span id="cb9-164"><a href="#cb9-164"></a>  refSeqName &lt;-<span class="st"> </span>refStrainName</span>
<span id="cb9-165"><a href="#cb9-165"></a>  matchNames &lt;-<span class="st"> </span><span class="kw">names</span>(featureAlignmentsNto1<span class="op">$</span>sequences) </span>
<span id="cb9-166"><a href="#cb9-166"></a>  refMatchName &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, refStrainName), matchNames, </span>
<span id="cb9-167"><a href="#cb9-167"></a>                       <span class="dt">perl =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</span>
<span id="cb9-168"><a href="#cb9-168"></a></span>
<span id="cb9-169"><a href="#cb9-169"></a>  <span class="co">## Define colors</span></span>
<span id="cb9-170"><a href="#cb9-170"></a>  matchColors &lt;-<span class="st"> </span>strainColors[seqNames]</span>
<span id="cb9-171"><a href="#cb9-171"></a>  <span class="kw">names</span>(matchColors) &lt;-<span class="st"> </span>matchNames</span>
<span id="cb9-172"><a href="#cb9-172"></a>  </span>
<span id="cb9-173"><a href="#cb9-173"></a>  <span class="cf">if</span> (featureTable[featureName, <span class="st">&quot;isCDS&quot;</span>]) {</span>
<span id="cb9-174"><a href="#cb9-174"></a>    </span>
<span id="cb9-175"><a href="#cb9-175"></a>    <span class="co">#### DNA PIP profile of translation-based multiple alignment ####</span></span>
<span id="cb9-176"><a href="#cb9-176"></a>    <span class="kw">PlotPIPprofiles</span>(</span>
<span id="cb9-177"><a href="#cb9-177"></a>      <span class="dt">alignment =</span> featureAlignmentsNto1<span class="op">$</span>tralignedDNA, </span>
<span id="cb9-178"><a href="#cb9-178"></a>      <span class="dt">refSeqName =</span> refMatchName,</span>
<span id="cb9-179"><a href="#cb9-179"></a>      <span class="dt">plotRefGaps =</span> <span class="ot">FALSE</span>,</span>
<span id="cb9-180"><a href="#cb9-180"></a>      <span class="dt">reversePlot =</span> <span class="ot">TRUE</span>,</span>
<span id="cb9-181"><a href="#cb9-181"></a>      <span class="dt">windowSize =</span> windowSize,</span>
<span id="cb9-182"><a href="#cb9-182"></a>      <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;DNA PIP profiles – collection: &quot;</span>, collection,</span>
<span id="cb9-183"><a href="#cb9-183"></a>                    <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Genomic feature &quot;</span>, featureName, <span class="st">&quot; (&quot;</span>, featureStart, <span class="st">&quot;:&quot;</span>, featureEnd ,<span class="st">&quot;, &quot;</span>, featureLength, <span class="st">&quot; bp)&quot;</span>),</span>
<span id="cb9-184"><a href="#cb9-184"></a>      <span class="dt">colors =</span> matchColors,</span>
<span id="cb9-185"><a href="#cb9-185"></a>      <span class="dt">legendMargin =</span> <span class="fl">0.3</span>,</span>
<span id="cb9-186"><a href="#cb9-186"></a>      <span class="dt">legendCorner =</span> <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb9-187"><a href="#cb9-187"></a>      <span class="dt">lwd =</span> <span class="dv">1</span>,</span>
<span id="cb9-188"><a href="#cb9-188"></a>      <span class="dt">legendCex =</span> <span class="fl">0.5</span>,</span>
<span id="cb9-189"><a href="#cb9-189"></a>      <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">100</span>))</span>
<span id="cb9-190"><a href="#cb9-190"></a>    </span>
<span id="cb9-191"><a href="#cb9-191"></a>    <span class="co">## AA PIP profile of translation-based multiple alignment</span></span>
<span id="cb9-192"><a href="#cb9-192"></a>    <span class="kw">PlotPIPprofiles</span>(</span>
<span id="cb9-193"><a href="#cb9-193"></a>      <span class="co">#    alignments = featureAlignmentsNto1$alignments[seqOrder],</span></span>
<span id="cb9-194"><a href="#cb9-194"></a>      <span class="dt">alignment =</span> featureAlignmentsNto1<span class="op">$</span>tralignedAA, </span>
<span id="cb9-195"><a href="#cb9-195"></a>      <span class="dt">refSeqName =</span> refMatchName,</span>
<span id="cb9-196"><a href="#cb9-196"></a>      <span class="dt">plotRefGaps =</span> <span class="ot">FALSE</span>,</span>
<span id="cb9-197"><a href="#cb9-197"></a>      <span class="dt">reversePlot =</span> <span class="ot">TRUE</span>,</span>
<span id="cb9-198"><a href="#cb9-198"></a>      <span class="dt">windowSize =</span> windowSize<span class="op">/</span><span class="dv">3</span>,</span>
<span id="cb9-199"><a href="#cb9-199"></a>      <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;AA PIP profiles: collection &quot;</span>, collection,</span>
<span id="cb9-200"><a href="#cb9-200"></a>                    <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Genomic feature &quot;</span>, featureName, <span class="st">&quot; (&quot;</span>, featureStart, <span class="st">&quot;:&quot;</span>, featureEnd ,<span class="st">&quot;, &quot;</span>, featureLength, <span class="st">&quot; bp)&quot;</span>),</span>
<span id="cb9-201"><a href="#cb9-201"></a>      <span class="dt">colors =</span> matchColors,</span>
<span id="cb9-202"><a href="#cb9-202"></a>      <span class="dt">legendMargin =</span> <span class="fl">0.3</span>,</span>
<span id="cb9-203"><a href="#cb9-203"></a>      <span class="dt">legendCorner =</span> <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb9-204"><a href="#cb9-204"></a>      <span class="dt">lwd =</span> <span class="dv">1</span>,</span>
<span id="cb9-205"><a href="#cb9-205"></a>      <span class="dt">legendCex =</span> <span class="fl">0.5</span>,</span>
<span id="cb9-206"><a href="#cb9-206"></a>      <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">100</span>))</span>
<span id="cb9-207"><a href="#cb9-207"></a>  }    </span>
<span id="cb9-208"><a href="#cb9-208"></a>  </span>
<span id="cb9-209"><a href="#cb9-209"></a>  </span>
<span id="cb9-210"><a href="#cb9-210"></a>}</span></code></pre></div>
<div id="sun_2020-13522-23686-10165bp" class="section level3">
<h3>Sun_2020 (13522-23686; 10165bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches_around-CoV-2-plus-GISAID/around-CoV-2-plus-GISAID_one2N_feature_matches-1.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="s1-21599-23617-2019bp" class="section level3">
<h3>S1 (21599-23617; 2019bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches_around-CoV-2-plus-GISAID/around-CoV-2-plus-GISAID_one2N_feature_matches-2.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches_around-CoV-2-plus-GISAID/around-CoV-2-plus-GISAID_one2N_feature_matches-3.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches_around-CoV-2-plus-GISAID/around-CoV-2-plus-GISAID_one2N_feature_matches-4.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># }</span></span></code></pre></div>
<!-- ## Translation-based multiple alignments -->
<!-- For the features corresponding to coding sequences (and in frame), it is more accurate to run aminoacid-based alignment, and to align in a second time the RNA codons onto the aligned aminoacids.  -->
<!-- ```{r align_translation} -->
<!-- # featureName <- "CDS-S" -->
<!-- featureName <- "S1" -->
<!-- for (featureName in row.names(featureTable)) { -->
<!--   if (featureTable[featureName, "isCDS"]) { -->
<!--     message("Translation-based multiple alignment of feature ", featureName) -->
<!--     featureAlignmentsNto1 <- allFeatureAlignments[[featureName]] -->
<!--     rawSequences <- DNAStringSet(featureAlignmentsNto1$sequences) -->
<!--     class(rawSequences) -->
<!--     traligned <- AlignTranslation( -->
<!--       rawSequences,  -->
<!--       type = "DNAStringSet", -->
<!--       verbose = FALSE) -->
<!--     allFeatureAlignments[[featureName]]$trAlignment <- traligned -->
<!--     # names(allFeatureAlignments[[featureName]]) -->
<!--     ## Save alignment -->
<!--     featurePrefix <- paste0( -->
<!--       featureName, -->
<!--       "_", collection) -->
<!--     ## Define output file for unaligned matching sequences -->
<!--     outfiles[paste0(featurePrefix, "_traligned")]  <- file.path( -->
<!--       dir[[featureName]], paste0(featurePrefix, "_traligned.fasta")) -->
<!--     message("\tTranslation-based alignment saved in file ",  -->
<!--             outfiles[paste0(featurePrefix, "_traligned")]) -->
<!--     ## Save aligned sequences in fasta format -->
<!--     writeXStringSet( -->
<!--       x = traligned,  -->
<!--       filepath = outfiles[paste0(featurePrefix, "_traligned")]) -->
<!--   } -->
<!--   ## Convert alignment to position-weight matrix -->
<!--   consensusM <- consensusMatrix(x = traligned) -->
<!--   allFeatureAlignments[[featureName]]$consensusM <- consensusM -->
<!-- } -->
<!-- ``` -->
<!-- ```{r one2N_feature_matches_PIP, fig.width=10, fig.height=6, out.width="100%", fig.cap="Feature-specific Percent Identical Positions (PIP) profiles. "} -->
<!-- #### One-to-N alignmemnt of user-speficied genomic features #### -->
<!-- allFeatureAlignments <- list() -->
<!-- # featureName <- "CDS-S" # for the test -->
<!-- featureName <- "S1" # for the test -->
<!-- # featureName <- "Recomb-reg-3" # for the test -->
<!-- # featureName <- "RBD" # for the test -->
<!-- # for (collection in (collections)) { -->
<!-- for (featureName in names(features)) { -->
<!--   featureLimits <- features[[featureName]] -->
<!--   featureStart <- featureLimits[["start"]] -->
<!--   featureEnd <- featureLimits[["end"]] -->
<!--   featureLength <-  -->
<!--      featureEnd - featureStart  + 1 -->
<!--   message("Searching matches for feature ", featureName,  -->
<!--           " (", featureLimits[1], "-", featureLimits[2], ")",  -->
<!--           " in collection ", collection) -->
<!--   #### N-to-1 alignments of spike-coding sequences #### -->
<!--   featurePrefix <- paste0( -->
<!--     featureName, -->
<!--     "_", collection) -->
<!--   dir[[featureName]] <- file.path(dir$results, featureName) -->
<!--   dir.create(dir[[featureName]], showWarnings = FALSE, recursive = TRUE) -->
<!--   outfiles[featureName]  <- file.path( -->
<!--     dir[[featureName]], paste0(featurePrefix, ".fasta")) -->
<!--   message("\tOutput directory: ", dir[[featureName]] ) -->
<!--   ## Get sequence of the feature from the reference genome -->
<!--   refSeq <- subseq(genomes[refStrainName],  -->
<!--                    start = featureLimits[1],  -->
<!--                    end = featureLimits[2]) -->
<!--   featureAlignmentsNto1 <- allFeatureAlignments[[featureName]] -->
<!--   seqOrder <- order(featureAlignmentsNto1$stat$score, decreasing = TRUE) -->
<!--   if (featureLength > 20000) { -->
<!--     windowSize <- 800  -->
<!--   } else if (featureLength > 5000) { -->
<!--     windowSize <- 500  -->
<!--   } else if (featureLength > 3000) { -->
<!--     windowSize <- 200  -->
<!--   } else { -->
<!--     windowSize <- max(100, 10 * round(featureLength / 100)) -->
<!--   } -->
<!--   ## PIP profile of one-to-N alignment -->
<!--   plotPIPprofiles( -->
<!--     alignments = featureAlignmentsNto1$alignments[seqOrder], -->
<!--     reversePlot = TRUE, -->
<!--     windowSize = windowSize,  -->
<!--     main = paste0(featureName, " (", featureStart, ":", featureEnd ,", ", featureLength, " bp)"),  -->
<!-- #    colors = NULL, -->
<!--     colors = strainColors, -->
<!--     legendMargin = 0.25,  -->
<!--     legendCorner = "topright", lwd = 1, -->
<!--     legendCex = 0.5, ylim = c(30, 100)) -->
<!--   kable(featureAlignmentsNto1$stats[seqOrder,],  -->
<!--         caption = paste0( -->
<!--           "One-to-N alignment of feature ",  -->
<!--           featureName)) -->
<!-- } -->
<!-- # } -->
<!-- ``` -->
</div>
</div>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw">kable</span>(<span class="kw">t</span>(<span class="kw">as.data.frame</span>(dir)), <span class="dt">col.names =</span> <span class="st">&quot;Dir&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Directories&quot;</span>)</span></code></pre></div>
<table>
<caption>Directories</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">Dir</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>main</td>
<td align="left">..</td>
</tr>
<tr class="even">
<td>R</td>
<td align="left">../scripts/R</td>
</tr>
<tr class="odd">
<td>seqdata</td>
<td align="left">../data</td>
</tr>
<tr class="even">
<td>results</td>
<td align="left">../results</td>
</tr>
<tr class="odd">
<td>images</td>
<td align="left">../memory_images</td>
</tr>
<tr class="even">
<td>sequences</td>
<td align="left">../data/GISAID_genomes</td>
</tr>
<tr class="odd">
<td>Sun_2020</td>
<td align="left">../results/Sun_2020_around-CoV-2-plus-GISAID</td>
</tr>
<tr class="even">
<td>S1</td>
<td align="left">../results/S1_around-CoV-2-plus-GISAID</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>outfileTable &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">path =</span> <span class="kw">as.vector</span>(outfiles))</span>
<span id="cb12-2"><a href="#cb12-2"></a>outfileTable<span class="op">$</span>basename &lt;-<span class="st"> </span><span class="kw">basename</span>(<span class="kw">as.vector</span>(outfileTable<span class="op">$</span>path))</span>
<span id="cb12-3"><a href="#cb12-3"></a>outfileTable<span class="op">$</span>dir &lt;-<span class="st"> </span><span class="kw">dirname</span>(<span class="kw">as.vector</span>(outfileTable<span class="op">$</span>path))</span>
<span id="cb12-4"><a href="#cb12-4"></a>outfileTable<span class="op">$</span>link &lt;-<span class="st"> </span><span class="kw">paste0</span>(</span>
<span id="cb12-5"><a href="#cb12-5"></a>  <span class="st">&quot;[&quot;</span>, outfileTable<span class="op">$</span>basename, <span class="st">&quot;](&quot;</span>, outfileTable<span class="op">$</span>path, <span class="st">&quot;)&quot;</span></span>
<span id="cb12-6"><a href="#cb12-6"></a>)</span>
<span id="cb12-7"><a href="#cb12-7"></a></span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="kw">kable</span>(outfileTable[, <span class="kw">c</span>(<span class="st">&quot;dir&quot;</span>, <span class="st">&quot;link&quot;</span>)], </span>
<span id="cb12-9"><a href="#cb12-9"></a>      <span class="dt">col.names =</span> <span class="kw">c</span>(<span class="st">&quot;dir&quot;</span>, <span class="st">&quot;file&quot;</span>), </span>
<span id="cb12-10"><a href="#cb12-10"></a>      <span class="dt">caption =</span> <span class="st">&quot;Output files&quot;</span>)</span></code></pre></div>
<table>
<caption>Output files</caption>
<thead>
<tr class="header">
<th align="left">dir</th>
<th align="left">file</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">../memory_images</td>
<td align="left"><a href="../memory_images/around-CoV-2-plus-GISAID_one-to-n_matches.Rdata">around-CoV-2-plus-GISAID_one-to-n_matches.Rdata</a></td>
</tr>
<tr class="even">
<td align="left">../results/Sun_2020_around-CoV-2-plus-GISAID</td>
<td align="left"><a href="../results/Sun_2020_around-CoV-2-plus-GISAID/Sun_2020_around-CoV-2-plus-GISAID_matches.fasta">Sun_2020_around-CoV-2-plus-GISAID_matches.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../results/S1_around-CoV-2-plus-GISAID</td>
<td align="left"><a href="../results/S1_around-CoV-2-plus-GISAID/S1_around-CoV-2-plus-GISAID_matches.fasta">S1_around-CoV-2-plus-GISAID_matches.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../results/S1_around-CoV-2-plus-GISAID</td>
<td align="left"><a href="../results/S1_around-CoV-2-plus-GISAID/S1_around-CoV-2-plus-GISAID_tralignedDNA.fasta">S1_around-CoV-2-plus-GISAID_tralignedDNA.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../results/S1_around-CoV-2-plus-GISAID</td>
<td align="left"><a href="../results/S1_around-CoV-2-plus-GISAID/S1_around-CoV-2-plus-GISAID_tralignedAA.fasta">S1_around-CoV-2-plus-GISAID_tralignedAA.fasta</a></td>
</tr>
</tbody>
</table>
</div>
<div id="memory-image" class="section level2">
<h2>Memory image</h2>
<p>We store the result ina memory image, in oder to be able reloading it to plot PIP profiles with different parameters.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw">save.image</span>(<span class="dt">file =</span> outfiles[<span class="st">&quot;Memory image&quot;</span>])</span></code></pre></div>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] DECIPHER_2.16.1     RSQLite_2.2.0       Biostrings_2.56.0   XVector_0.28.0      IRanges_2.22.2      S4Vectors_0.26.1    BiocGenerics_0.34.0 knitr_1.28         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6        magrittr_1.5        zlibbioc_1.34.0     bit_1.1-15.2        rlang_0.4.6         highr_0.8           stringr_1.4.0       blob_1.2.1          tools_4.0.0         xfun_0.14           DBI_1.1.0           htmltools_0.4.0     yaml_2.2.1          bit64_0.9-7        
[15] digest_0.6.25       crayon_1.3.4        BiocManager_1.30.10 vctrs_0.3.0         memoise_1.1.0       evaluate_0.14       rmarkdown_2.1       stringi_1.4.6       compiler_4.0.0     </code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
