<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-05-23" />

<title>One-to-N sequence matches in coronavirus genomes</title>

<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="one-to-n-matches_around-CoV-2-plus-GISAID_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/navigation-1.1/tabsets.js"></script>
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/navigation-1.1/codefolding.js"></script>
<link href="one-to-n-matches_around-CoV-2-plus-GISAID_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="one-to-n-matches_around-CoV-2-plus-GISAID_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">One-to-N sequence matches in coronavirus genomes</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-05-23</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#### General parameters for the analysis ####</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">reloadImage &lt;-<span class="st"> </span><span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">#### Sequence collection ####</span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">## Supported: </span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8">collections &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">  <span class="st">&quot;around-CoV-2&quot;</span>, </a>
<a class="sourceLine" id="cb1-10" data-line-number="10">  <span class="st">&quot;selected&quot;</span>,</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">  <span class="st">&quot;around-CoV-2-plus-GISAID&quot;</span>, </a>
<a class="sourceLine" id="cb1-12" data-line-number="12">  <span class="st">&quot;selected-plus-GISAID&quot;</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13">  )</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co"># collection &lt;- &quot;around-CoV-2&quot; # 14 genomes</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co"># collection &lt;- &quot;all&quot; # 16 genomes</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="co"># collection &lt;- &quot;selected&quot; # ~60 genomes</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20">collection &lt;-<span class="st"> &quot;around-CoV-2-plus-GISAID&quot;</span> <span class="co"># 16 genomes</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co"># collection &lt;- &quot;selected-plus-GISAID&quot; # ~60 genomes</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co"># collection &lt;- &quot;all-plus-GISAID&quot; # 16 genomes</span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23"></a>
<a class="sourceLine" id="cb1-24" data-line-number="24"><span class="co">## Note about GIDAID sequences.</span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="co">## </span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">## A few genomes were not available in NCBI Genbank at the time of </span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="co">## this analysis, and had to be downloaded from GISAID. These sequences</span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">##  can however not be redistributed, they should thus be downloaded </span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29"><span class="co">##  manually to reproduce the full trees. </span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="co">## Exclude incomplete genomes (i.e. those containing a lof of Ns) to avoid biases in the distance computations</span></a>
<a class="sourceLine" id="cb1-33" data-line-number="33">excludeIncomplete &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34"></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"></a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="co">#### Define directories and files ####</span></a>
<a class="sourceLine" id="cb1-38" data-line-number="38">dir &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">main =</span> <span class="st">&#39;..&#39;</span>)</a>
<a class="sourceLine" id="cb1-39" data-line-number="39">dir<span class="op">$</span>R &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;scripts/R&quot;</span>)</a>
<a class="sourceLine" id="cb1-40" data-line-number="40"></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="co">#### Create output directory for sequences ####</span></a>
<a class="sourceLine" id="cb1-42" data-line-number="42">dir<span class="op">$</span>seqdata &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb1-43" data-line-number="43"><span class="kw">dir.create</span>(dir<span class="op">$</span>seqdata, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-44" data-line-number="44"></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"><span class="co">## Instantiate a list for output files</span></a>
<a class="sourceLine" id="cb1-46" data-line-number="46">outfiles &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb1-47" data-line-number="47"></a>
<a class="sourceLine" id="cb1-48" data-line-number="48"><span class="co">## Memory image</span></a>
<a class="sourceLine" id="cb1-49" data-line-number="49">dir<span class="op">$</span>images &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;memory_images&quot;</span>)</a>
<a class="sourceLine" id="cb1-50" data-line-number="50"><span class="kw">dir.create</span>(dir<span class="op">$</span>images, <span class="dt">recursive =</span> <span class="ot">TRUE</span>, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb1-51" data-line-number="51">outfiles[<span class="st">&quot;Memory image&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb1-52" data-line-number="52">  dir<span class="op">$</span>images, </a>
<a class="sourceLine" id="cb1-53" data-line-number="53">  <span class="kw">paste0</span>(collection, <span class="st">&quot;_one-to-n_matches.Rdata&quot;</span>))</a>
<a class="sourceLine" id="cb1-54" data-line-number="54"></a>
<a class="sourceLine" id="cb1-55" data-line-number="55"><span class="co">## Input files</span></a>
<a class="sourceLine" id="cb1-56" data-line-number="56">infiles &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-57" data-line-number="57"></a>
<a class="sourceLine" id="cb1-58" data-line-number="58"><span class="co">## Output tables</span></a>
<a class="sourceLine" id="cb1-59" data-line-number="59"><span class="co"># di$output &lt;- file.path(dir.main, &quot;&quot;)</span></a>
<a class="sourceLine" id="cb1-60" data-line-number="60"><span class="co"># dir$tables &lt;- </span></a>
<a class="sourceLine" id="cb1-61" data-line-number="61"></a>
<a class="sourceLine" id="cb1-62" data-line-number="62"><span class="co">## Load custom functions</span></a>
<a class="sourceLine" id="cb1-63" data-line-number="63"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;align_n_to_one.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-64" data-line-number="64"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;plot_pip_profiles.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-65" data-line-number="65"></a>
<a class="sourceLine" id="cb1-66" data-line-number="66"><span class="co">## A unequivocal pattern to identify the reference genome in the sequence names of the input file</span></a>
<a class="sourceLine" id="cb1-67" data-line-number="67">refPattern &lt;-<span class="st"> &quot;HuCoV2_WH01_2019&quot;</span></a>
<a class="sourceLine" id="cb1-68" data-line-number="68"></a>
<a class="sourceLine" id="cb1-69" data-line-number="69"><span class="co">## Exclude some genomes with a lot of Ns, because they bias the PIP profiles and alignments and trees</span></a>
<a class="sourceLine" id="cb1-70" data-line-number="70">excludePatterns &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;PnMP789&quot;</span>, <span class="st">&quot;PnGu-P2S_2019&quot;</span>)</a>
<a class="sourceLine" id="cb1-71" data-line-number="71"></a>
<a class="sourceLine" id="cb1-72" data-line-number="72"><span class="co">#### Features of interest in the reference genome ####</span></a>
<a class="sourceLine" id="cb1-73" data-line-number="73"></a>
<a class="sourceLine" id="cb1-74" data-line-number="74">features &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-75" data-line-number="75"></a>
<a class="sourceLine" id="cb1-76" data-line-number="76"></a>
<a class="sourceLine" id="cb1-77" data-line-number="77"><span class="co">#### Specific features  ####</span></a>
<a class="sourceLine" id="cb1-78" data-line-number="78"></a>
<a class="sourceLine" id="cb1-79" data-line-number="79"><span class="co">## Regions around the insertions</span></a>
<a class="sourceLine" id="cb1-80" data-line-number="80">features[[<span class="st">&#39;Ins1-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21647</span>, <span class="dt">end =</span> <span class="dv">21907</span>)</a>
<a class="sourceLine" id="cb1-81" data-line-number="81">features[[<span class="st">&#39;Ins2-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21899</span>, <span class="dt">end =</span> <span class="dv">22156</span>)</a>
<a class="sourceLine" id="cb1-82" data-line-number="82">features[[<span class="st">&#39;Ins3-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21899</span>, <span class="dt">end =</span> <span class="dv">23152</span>)</a>
<a class="sourceLine" id="cb1-83" data-line-number="83">features[[<span class="st">&#39;Ins4-pm120&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">23483</span>, <span class="dt">end =</span> <span class="dv">23734</span>)</a>
<a class="sourceLine" id="cb1-84" data-line-number="84">features[[<span class="st">&#39;Ins4-m240&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">23363</span>, <span class="dt">end =</span> <span class="dv">23614</span>)</a>
<a class="sourceLine" id="cb1-85" data-line-number="85"></a>
<a class="sourceLine" id="cb1-86" data-line-number="86"><span class="co">## Potential Pangolin origin after Xiao (https://doi.org/10.1101/2020.02.17.951335)</span></a>
<a class="sourceLine" id="cb1-87" data-line-number="87">features[[<span class="st">&#39;Recomb-Xiao&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">22871</span>, <span class="dt">end =</span> <span class="dv">23092</span>)</a>
<a class="sourceLine" id="cb1-88" data-line-number="88"><span class="co">## Recombinant region 1 seen on the PIP profiles</span></a>
<a class="sourceLine" id="cb1-89" data-line-number="89">features[[<span class="st">&#39;Recomb-reg-1&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21500</span>, <span class="dt">end =</span> <span class="dv">22800</span>)</a>
<a class="sourceLine" id="cb1-90" data-line-number="90"><span class="co">## Recombinant region 2 seen on the PIP profiles</span></a>
<a class="sourceLine" id="cb1-91" data-line-number="91">features[[<span class="st">&#39;Recomb-reg-2&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">22800</span>, <span class="dt">end =</span> <span class="dv">24000</span>)</a>
<a class="sourceLine" id="cb1-92" data-line-number="92"><span class="co">## Recombinant region 3 seen on the PIP profiles</span></a>
<a class="sourceLine" id="cb1-93" data-line-number="93">features[[<span class="st">&#39;Recomb-reg-3&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27800</span>, <span class="dt">end =</span> <span class="dv">28350</span>)</a>
<a class="sourceLine" id="cb1-94" data-line-number="94"></a>
<a class="sourceLine" id="cb1-95" data-line-number="95"><span class="co">## Annotated coding sequences</span></a>
<a class="sourceLine" id="cb1-96" data-line-number="96">features[[<span class="st">&#39;CDS-S&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21563</span>, <span class="dt">end =</span> <span class="dv">25384</span>) <span class="co">## Spike gene</span></a>
<a class="sourceLine" id="cb1-97" data-line-number="97">features[[<span class="st">&#39;CDS-ORF3a&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">25393</span>, <span class="dt">end =</span> <span class="dv">26220</span>)</a>
<a class="sourceLine" id="cb1-98" data-line-number="98">features[[<span class="st">&#39;CDS-E&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">26245</span>, <span class="dt">end =</span> <span class="dv">26472</span>)</a>
<a class="sourceLine" id="cb1-99" data-line-number="99">features[[<span class="st">&#39;CDS-M&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">26523</span>, <span class="dt">end =</span> <span class="dv">27191</span>)</a>
<a class="sourceLine" id="cb1-100" data-line-number="100">features[[<span class="st">&#39;CDS-ORF6&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27202</span>, <span class="dt">end =</span> <span class="dv">27387</span>)</a>
<a class="sourceLine" id="cb1-101" data-line-number="101">features[[<span class="st">&#39;CDS-ORF7a&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27394</span>, <span class="dt">end =</span> <span class="dv">27759</span>)</a>
<a class="sourceLine" id="cb1-102" data-line-number="102">features[[<span class="st">&#39;CDS-ORF8&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27894</span>, <span class="dt">end =</span> <span class="dv">28259</span>)</a>
<a class="sourceLine" id="cb1-103" data-line-number="103">features[[<span class="st">&#39;CDS-N&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">28274</span>, <span class="dt">end =</span> <span class="dv">29533</span>)</a>
<a class="sourceLine" id="cb1-104" data-line-number="104">features[[<span class="st">&#39;CDS-ORF10&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">29558</span>, <span class="dt">end =</span> <span class="dv">29674</span>)</a>
<a class="sourceLine" id="cb1-105" data-line-number="105">features[[<span class="st">&#39;CDS-ORF1ab&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">266</span>, <span class="dt">end =</span> <span class="dv">21555</span>)</a>
<a class="sourceLine" id="cb1-106" data-line-number="106"></a>
<a class="sourceLine" id="cb1-107" data-line-number="107"><span class="co">## All the sequences after the bif ORF coding for the polyproteinn 1ab</span></a>
<a class="sourceLine" id="cb1-108" data-line-number="108">features[[<span class="st">&#39;After-ORF1ab&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21556</span>, <span class="dt">end =</span> <span class="dv">29899</span>)</a>
<a class="sourceLine" id="cb1-109" data-line-number="109"></a>
<a class="sourceLine" id="cb1-110" data-line-number="110"><span class="co">## Report the parameters</span></a>
<a class="sourceLine" id="cb1-111" data-line-number="111"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Reference strain: &quot;</span>, refPattern)</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">## Genome dir and files</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="cf">if</span> (<span class="kw">length</span>(<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="st">&quot;GISAID&quot;</span>, <span class="dt">x =</span> collection)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) {</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">  useGISAID &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;GISAID_genomes&quot;</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  <span class="co"># collections &lt;-  paste0(collections, &quot;-plus-GISAID&quot;)</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  <span class="co"># collection &lt;-  paste0(collection, &quot;-plus-GISAID&quot;)</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;genomes&quot;</span>)</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">}</a>
<a class="sourceLine" id="cb2-10" data-line-number="10"></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">## Define the input sequences</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">infiles<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb2-14" data-line-number="14">  dir<span class="op">$</span>sequences, </a>
<a class="sourceLine" id="cb2-15" data-line-number="15">  <span class="kw">paste0</span>(<span class="st">&quot;genomes_&quot;</span>, collection, <span class="st">&quot;.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb2-16" data-line-number="16"></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">## Check if the input  sequence file exists</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(infiles<span class="op">$</span>sequences)) {</a>
<a class="sourceLine" id="cb2-19" data-line-number="19">  <span class="kw">stop</span>(<span class="st">&quot;Genome sequence file is missing&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb2-20" data-line-number="20">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">  <span class="kw">message</span>(<span class="st">&quot;Genome sequence file&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb2-22" data-line-number="22">}</a>
<a class="sourceLine" id="cb2-23" data-line-number="23"></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co">#### Load genome sequences ####</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25">genomes &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="dt">filepath =</span> infiles<span class="op">$</span>sequences, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</a>
<a class="sourceLine" id="cb2-26" data-line-number="26"></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">## Shorten sequence names by suppressing the fasta comment (after the space)</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28"><span class="kw">names</span>(genomes) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">&quot; .*&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="dt">x =</span> <span class="kw">names</span>(genomes), <span class="dt">perl =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-29" data-line-number="29"></a>
<a class="sourceLine" id="cb2-30" data-line-number="30"><span class="co">## Exclude genomes</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="cf">if</span> (excludeIncomplete) {</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">  excludePattern =<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;(&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;)|(&quot;</span>, excludePatterns), <span class="st">&quot;)&quot;</span>)</a>
<a class="sourceLine" id="cb2-33" data-line-number="33">  excludedstrainNames &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> excludePattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </a>
<a class="sourceLine" id="cb2-34" data-line-number="34">                              <span class="dt">value =</span> <span class="ot">TRUE</span>, <span class="dt">invert =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-35" data-line-number="35">  filteredGenomeIndices &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> excludePattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </a>
<a class="sourceLine" id="cb2-36" data-line-number="36">                                <span class="dt">value =</span> <span class="ot">FALSE</span>, <span class="dt">invert =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-37" data-line-number="37">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Excluded &quot;</span>, <span class="kw">length</span>(excludedstrainNames),<span class="st">&quot; genomes: &quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, excludedstrainNames)) </a>
<a class="sourceLine" id="cb2-38" data-line-number="38">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Remaining genomes: &quot;</span>, <span class="kw">length</span>(filteredGenomeIndices))</a>
<a class="sourceLine" id="cb2-39" data-line-number="39">  genomes &lt;-<span class="st"> </span>genomes[filteredGenomeIndices]</a>
<a class="sourceLine" id="cb2-40" data-line-number="40"><span class="co"># names(genomes)</span></a>
<a class="sourceLine" id="cb2-41" data-line-number="41">}</a>
<a class="sourceLine" id="cb2-42" data-line-number="42"></a>
<a class="sourceLine" id="cb2-43" data-line-number="43"><span class="co">## Report the number of genoomes</span></a>
<a class="sourceLine" id="cb2-44" data-line-number="44">strainNames &lt;-<span class="st"> </span><span class="kw">names</span>(genomes)</a>
<a class="sourceLine" id="cb2-45" data-line-number="45">nbStrains &lt;-<span class="st"> </span><span class="kw">length</span>(strainNames)</a>
<a class="sourceLine" id="cb2-46" data-line-number="46"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Loaded &quot;</span>, nbStrains, <span class="st">&quot; genomes from file &quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb2-47" data-line-number="47"><span class="co"># View(genomes)</span></a>
<a class="sourceLine" id="cb2-48" data-line-number="48"></a>
<a class="sourceLine" id="cb2-49" data-line-number="49"></a>
<a class="sourceLine" id="cb2-50" data-line-number="50"><span class="co">#### Define reference and query genomes ####</span></a>
<a class="sourceLine" id="cb2-51" data-line-number="51">refStrainName &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> refPattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </a>
<a class="sourceLine" id="cb2-52" data-line-number="52">                      <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-53" data-line-number="53"><span class="cf">if</span> (<span class="kw">is.null</span>(refStrainName)) {</a>
<a class="sourceLine" id="cb2-54" data-line-number="54">  <span class="kw">stop</span>(<span class="st">&quot;Could not identify reference genome with pattern &quot;</span>, refPattern)</a>
<a class="sourceLine" id="cb2-55" data-line-number="55">}</a>
<a class="sourceLine" id="cb2-56" data-line-number="56"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Reference genome name: &quot;</span>, refStrainName)</a>
<a class="sourceLine" id="cb2-57" data-line-number="57"></a>
<a class="sourceLine" id="cb2-58" data-line-number="58"><span class="co">## Compute some statistics about sequence sizes</span></a>
<a class="sourceLine" id="cb2-59" data-line-number="59">strainStats &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb2-60" data-line-number="60">  <span class="dt">n =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genomes),</a>
<a class="sourceLine" id="cb2-61" data-line-number="61">  <span class="dt">row.names =</span> <span class="kw">names</span>(genomes),</a>
<a class="sourceLine" id="cb2-62" data-line-number="62">  <span class="dt">status =</span> <span class="kw">rep</span>(<span class="st">&quot;Query&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(strainNames))</a>
<a class="sourceLine" id="cb2-63" data-line-number="63">)</a>
<a class="sourceLine" id="cb2-64" data-line-number="64">strainStats[,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> </span><span class="kw">as.vector</span>(strainStats[,<span class="st">&quot;status&quot;</span>])</a>
<a class="sourceLine" id="cb2-65" data-line-number="65">strainStats[refStrainName,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> &quot;Reference&quot;</span></a>
<a class="sourceLine" id="cb2-66" data-line-number="66">g &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb2-67" data-line-number="67"><span class="cf">for</span> (g <span class="cf">in</span> strainNames) {</a>
<a class="sourceLine" id="cb2-68" data-line-number="68">  strainStats[g, <span class="st">&quot;length&quot;</span>] &lt;-<span class="st"> </span><span class="kw">length</span>(genomes[[g]])</a>
<a class="sourceLine" id="cb2-69" data-line-number="69">}</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#### Define the color associated to each sequence ####</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">## Color palette per species</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">speciesPalette &lt;-<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">  <span class="dt">Human =</span> <span class="st">&quot;#880000&quot;</span>,</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">  <span class="dt">Bat =</span> <span class="st">&quot;#888888&quot;</span>,</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">  <span class="dt">Pangolin =</span> <span class="st">&quot;#448800&quot;</span>,</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">  <span class="dt">Camel =</span> <span class="st">&quot;#BB8800&quot;</span>,</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">  <span class="dt">Pig =</span> <span class="st">&quot;#FFBBBB&quot;</span>,</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">  <span class="dt">Civet =</span> <span class="st">&quot;#00BBFF&quot;</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11">)</a>
<a class="sourceLine" id="cb3-12" data-line-number="12"></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">## Species prefix in the tip labels</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14">speciesPrefix &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Hu&quot;</span> =<span class="st"> &quot;Human&quot;</span>,</a>
<a class="sourceLine" id="cb3-15" data-line-number="15">                   <span class="st">&quot;Bt&quot;</span> =<span class="st"> &quot;Bat&quot;</span>,</a>
<a class="sourceLine" id="cb3-16" data-line-number="16">                   <span class="st">&quot;Pn&quot;</span> =<span class="st"> &quot;Pangolin&quot;</span>,</a>
<a class="sourceLine" id="cb3-17" data-line-number="17">                   <span class="st">&quot;Cm&quot;</span> =<span class="st"> &quot;Camel&quot;</span>,</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">                   <span class="st">&quot;Pi&quot;</span> =<span class="st"> &quot;Pig&quot;</span>,</a>
<a class="sourceLine" id="cb3-19" data-line-number="19">                   <span class="st">&quot;Cv&quot;</span> =<span class="st"> &quot;Civet&quot;</span>)</a>
<a class="sourceLine" id="cb3-20" data-line-number="20"></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">## Strain-specific colors</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22">strainColor &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span> =<span class="st"> &quot;red&quot;</span>,</a>
<a class="sourceLine" id="cb3-24" data-line-number="24">  <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span> =<span class="st"> &quot;#0044BB&quot;</span>,</a>
<a class="sourceLine" id="cb3-25" data-line-number="25">  <span class="st">&quot;PnGu1_2019&quot;</span> =<span class="st"> &quot;#00BB00&quot;</span>,</a>
<a class="sourceLine" id="cb3-26" data-line-number="26">  <span class="st">&quot;BtRaTG13_&quot;</span> =<span class="st"> &quot;#FF6600&quot;</span>,</a>
<a class="sourceLine" id="cb3-27" data-line-number="27">  <span class="st">&quot;BtYu-RmYN&quot;</span> =<span class="st"> &quot;#FFBB22&quot;</span>,</a>
<a class="sourceLine" id="cb3-28" data-line-number="28">  <span class="st">&quot;BtZXC21&quot;</span> =<span class="st"> &quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb3-29" data-line-number="29">  <span class="st">&quot;BtZC45&quot;</span> =<span class="st"> &quot;black&quot;</span>)</a>
<a class="sourceLine" id="cb3-30" data-line-number="30"></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">## Identify species per tip</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="cf">for</span> (prefix <span class="cf">in</span> <span class="kw">names</span>(speciesPrefix)) {</a>
<a class="sourceLine" id="cb3-33" data-line-number="33">  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, prefix), <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;species&quot;</span>] &lt;-<span class="st"> </span>speciesPrefix[prefix]</a>
<a class="sourceLine" id="cb3-34" data-line-number="34">  </a>
<a class="sourceLine" id="cb3-35" data-line-number="35">}</a>
<a class="sourceLine" id="cb3-36" data-line-number="36"></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">## Assign acolor to each species</span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38">strainStats<span class="op">$</span>color &lt;-<span class="st"> &quot;grey&quot;</span> <span class="co"># default</span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39">strainStats<span class="op">$</span>color &lt;-<span class="st"> </span>speciesPalette[<span class="kw">as.vector</span>(strainStats<span class="op">$</span>species)]</a>
<a class="sourceLine" id="cb3-40" data-line-number="40"></a>
<a class="sourceLine" id="cb3-41" data-line-number="41"><span class="cf">for</span> (strain <span class="cf">in</span> <span class="kw">names</span>(strainColor)) {</a>
<a class="sourceLine" id="cb3-42" data-line-number="42">  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, strain), </a>
<a class="sourceLine" id="cb3-43" data-line-number="43">                  <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;color&quot;</span>] &lt;-<span class="st"> </span>strainColor[strain]</a>
<a class="sourceLine" id="cb3-44" data-line-number="44">  </a>
<a class="sourceLine" id="cb3-45" data-line-number="45">}</a>
<a class="sourceLine" id="cb3-46" data-line-number="46"></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"></a>
<a class="sourceLine" id="cb3-48" data-line-number="48"><span class="co">## Assign specific color to some nodes</span></a>
<a class="sourceLine" id="cb3-49" data-line-number="49"></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="co">## Define a color for each strain</span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51">strainColors &lt;-<span class="st"> </span>(<span class="kw">unlist</span>(strainStats<span class="op">$</span>color))</a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="kw">names</span>(strainColors) &lt;-<span class="st"> </span><span class="kw">row.names</span>(strainStats)</a></code></pre></div>
<div id="parameters" class="section level2">
<h2>Parameters</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">## Define a list of parameters</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">parameters &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">parameters<span class="op">$</span>collection &lt;-<span class="st"> </span>collection</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">parameters<span class="op">$</span>refStrainName &lt;-<span class="st"> </span>refStrainName</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">parameters<span class="op">$</span>nbStrains &lt;-<span class="st"> </span>nbStrains</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">parameters<span class="op">$</span>sequenceDir &lt;-<span class="st"> </span>dir<span class="op">$</span>sequences</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">parameters<span class="op">$</span>sequenceFile &lt;-<span class="st"> </span>infiles<span class="op">$</span>sequences</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">parameters<span class="op">$</span>memoryImage &lt;-<span class="st"> </span>outfiles[<span class="st">&quot;Memory image&quot;</span>]</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="kw">kable</span>(<span class="kw">t</span>(<span class="kw">as.data.frame.list</span>(parameters)), <span class="dt">caption =</span> <span class="st">&quot;Parameters of the analysis&quot;</span>, </a>
<a class="sourceLine" id="cb4-11" data-line-number="11">      <span class="dt">col.names =</span> <span class="st">&quot;Parameter&quot;</span>)</a></code></pre></div>
<table>
<caption>Parameters of the analysis</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">Parameter</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>collection</td>
<td align="left">around-CoV-2-plus-GISAID</td>
</tr>
<tr class="even">
<td>refStrainName</td>
<td align="left">HuCoV2_WH01_2019</td>
</tr>
<tr class="odd">
<td>nbStrains</td>
<td align="left">16</td>
</tr>
<tr class="even">
<td>sequenceDir</td>
<td align="left">../data/GISAID_genomes</td>
</tr>
<tr class="odd">
<td>sequenceFile</td>
<td align="left">../data/GISAID_genomes/genomes_around-CoV-2-plus-GISAID.fasta</td>
</tr>
<tr class="even">
<td>memoryImage</td>
<td align="left">../memory_images/around-CoV-2-plus-GISAID_one-to-n_matches.Rdata</td>
</tr>
</tbody>
</table>
</div>
<div id="genome-statistics" class="section level2">
<h2>Genome statistics</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">kable</span>(strainStats, <span class="dt">caption =</span> <span class="st">&quot;Reference and query genomes&quot;</span>)</a></code></pre></div>
<table>
<caption>Reference and query genomes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">n</th>
<th align="left">status</th>
<th align="right">length</th>
<th align="left">species</th>
<th align="left">color</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BtBM48-31</td>
<td align="right">1</td>
<td align="left">Query</td>
<td align="right">29276</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtGX2013</td>
<td align="right">2</td>
<td align="left">Query</td>
<td align="right">29161</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="odd">
<td>BtHKU3-12</td>
<td align="right">3</td>
<td align="left">Query</td>
<td align="right">29704</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">4</td>
<td align="left">Query</td>
<td align="right">29855</td>
<td align="left">Bat</td>
<td align="left">#FF6600</td>
</tr>
<tr class="odd">
<td>BtRs4874</td>
<td align="right">5</td>
<td align="left">Query</td>
<td align="right">30311</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtYN2013</td>
<td align="right">6</td>
<td align="left">Query</td>
<td align="right">29142</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="odd">
<td>BtYN2018B</td>
<td align="right">7</td>
<td align="left">Query</td>
<td align="right">30256</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtZC45</td>
<td align="right">8</td>
<td align="left">Query</td>
<td align="right">29802</td>
<td align="left">Bat</td>
<td align="left">black</td>
</tr>
<tr class="odd">
<td>BtZXC21</td>
<td align="right">9</td>
<td align="left">Query</td>
<td align="right">29732</td>
<td align="left">Bat</td>
<td align="left">black</td>
</tr>
<tr class="even">
<td>Cv007-2004</td>
<td align="right">10</td>
<td align="left">Query</td>
<td align="right">29540</td>
<td align="left">Civet</td>
<td align="left">#00BBFF</td>
</tr>
<tr class="odd">
<td>HuCoV2_WH01_2019</td>
<td align="right">11</td>
<td align="left">Reference</td>
<td align="right">29899</td>
<td align="left">Human</td>
<td align="left">red</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="right">12</td>
<td align="left">Query</td>
<td align="right">29727</td>
<td align="left">Human</td>
<td align="left">#0044BB</td>
</tr>
<tr class="odd">
<td>PnGX-P1E_2017</td>
<td align="right">13</td>
<td align="left">Query</td>
<td align="right">29801</td>
<td align="left">Pangolin</td>
<td align="left">#448800</td>
</tr>
<tr class="even">
<td>PnGX-P2V_2018</td>
<td align="right">14</td>
<td align="left">Query</td>
<td align="right">29795</td>
<td align="left">Pangolin</td>
<td align="left">#448800</td>
</tr>
<tr class="odd">
<td>BtYu-RmYN02_2019</td>
<td align="right">15</td>
<td align="left">Query</td>
<td align="right">29671</td>
<td align="left">Bat</td>
<td align="left">#FFBB22</td>
</tr>
<tr class="even">
<td>PnGu1_2019</td>
<td align="right">16</td>
<td align="left">Query</td>
<td align="right">29825</td>
<td align="left">Pangolin</td>
<td align="left">#00BB00</td>
</tr>
</tbody>
</table>
<p>The collection around-CoV-2-plus-GISAID contains 16 virus genome sequences.</p>
</div>
<div id="one-to-n-alignemnts-of-selected-features" class="section level2">
<h2>One-to-N alignemnts of selected features</h2>
<p>We perform a global pairwise alignment (Needle-Waterman algorithm) between each feature of the reference (HuCoV2_WH01_2019) and each one of the query genomes.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co">#### One-to-N alignmemnt of user-speficied genomic features ####</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">featureName &lt;-<span class="st"> &quot;CDS-S&quot;</span> <span class="co"># for the test</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co"># featureName &lt;- &quot;S1&quot; # for the test</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co"># featureName &lt;- &quot;Recomb-reg-3&quot; # for the test</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co"># featureName &lt;- &quot;RBD&quot; # for the test</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co"># for (collection in (collections)) {</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="cf">if</span> (reloadImage) {</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">  <span class="kw">load</span>(outfiles[<span class="st">&quot;Memory image&quot;</span>])</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">  reloadImage &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"></a>
<a class="sourceLine" id="cb6-11" data-line-number="11">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb6-12" data-line-number="12">  allFeatureAlignments &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb6-13" data-line-number="13">}</a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="cf">for</span> (featureName <span class="cf">in</span> <span class="kw">names</span>(features)) {</a>
<a class="sourceLine" id="cb6-15" data-line-number="15">  featureLimits &lt;-<span class="st"> </span>features[[featureName]]</a>
<a class="sourceLine" id="cb6-16" data-line-number="16">  featureStart &lt;-<span class="st"> </span>featureLimits[[<span class="st">&quot;start&quot;</span>]]</a>
<a class="sourceLine" id="cb6-17" data-line-number="17">  featureEnd &lt;-<span class="st"> </span>featureLimits[[<span class="st">&quot;end&quot;</span>]]</a>
<a class="sourceLine" id="cb6-18" data-line-number="18">  featureLength &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="st">    </span>featureEnd <span class="op">-</span><span class="st"> </span>featureStart  <span class="op">+</span><span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20">  </a>
<a class="sourceLine" id="cb6-21" data-line-number="21">  <span class="kw">message</span>(<span class="st">&quot;Searching matches for feature &quot;</span>, featureName, </a>
<a class="sourceLine" id="cb6-22" data-line-number="22">          <span class="st">&quot; (&quot;</span>, featureLimits[<span class="dv">1</span>], <span class="st">&quot;-&quot;</span>, featureLimits[<span class="dv">2</span>], <span class="st">&quot;)&quot;</span>, </a>
<a class="sourceLine" id="cb6-23" data-line-number="23">          <span class="st">&quot; in collection &quot;</span>, collection)</a>
<a class="sourceLine" id="cb6-24" data-line-number="24">  </a>
<a class="sourceLine" id="cb6-25" data-line-number="25">  <span class="kw">cat</span>(<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">### &quot;</span>,  featureName, </a>
<a class="sourceLine" id="cb6-26" data-line-number="26">      <span class="st">&quot; (&quot;</span>, featureLimits[<span class="dv">1</span>], <span class="st">&quot;-&quot;</span>, featureLimits[<span class="dv">2</span>], </a>
<a class="sourceLine" id="cb6-27" data-line-number="27">      <span class="st">&quot;; &quot;</span>, featureLength,<span class="st">&quot;bp)&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb6-28" data-line-number="28"></a>
<a class="sourceLine" id="cb6-29" data-line-number="29">  <span class="co">#### N-to-1 alignments of spike-coding sequences ####</span></a>
<a class="sourceLine" id="cb6-30" data-line-number="30">  featurePrefix &lt;-<span class="st"> </span><span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb6-31" data-line-number="31">    featureName,</a>
<a class="sourceLine" id="cb6-32" data-line-number="32">    <span class="st">&quot;_&quot;</span>, collection)</a>
<a class="sourceLine" id="cb6-33" data-line-number="33">  </a>
<a class="sourceLine" id="cb6-34" data-line-number="34">  dir[[featureName]] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>seqdata, featureName)</a>
<a class="sourceLine" id="cb6-35" data-line-number="35">  <span class="kw">dir.create</span>(dir[[featureName]], <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb6-36" data-line-number="36">  outfiles[featureName]  &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb6-37" data-line-number="37">    dir[[featureName]], <span class="kw">paste0</span>(featurePrefix, <span class="st">&quot;.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb6-38" data-line-number="38">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Output directory: &quot;</span>, dir[[featureName]] )</a>
<a class="sourceLine" id="cb6-39" data-line-number="39">  </a>
<a class="sourceLine" id="cb6-40" data-line-number="40">  <span class="co">## Get sequence of the feature from the reference genome</span></a>
<a class="sourceLine" id="cb6-41" data-line-number="41">  refSeq &lt;-<span class="st"> </span><span class="kw">subseq</span>(genomes[refStrainName], </a>
<a class="sourceLine" id="cb6-42" data-line-number="42">                   <span class="dt">start =</span> featureLimits[<span class="dv">1</span>], </a>
<a class="sourceLine" id="cb6-43" data-line-number="43">                   <span class="dt">end =</span> featureLimits[<span class="dv">2</span>])</a>
<a class="sourceLine" id="cb6-44" data-line-number="44">  </a>
<a class="sourceLine" id="cb6-45" data-line-number="45">  <span class="co">## Match the reference feature with all the genomes</span></a>
<a class="sourceLine" id="cb6-46" data-line-number="46">  <span class="cf">if</span> (reloadImage) {</a>
<a class="sourceLine" id="cb6-47" data-line-number="47">    featureAlignmentsNto1 &lt;-<span class="st"> </span>allFeatureAlignments[[featureName]]</a>
<a class="sourceLine" id="cb6-48" data-line-number="48">  } <span class="cf">else</span> {  </a>
<a class="sourceLine" id="cb6-49" data-line-number="49">    featureAlignmentsNto1 &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb6-50" data-line-number="50">      <span class="dt">refSequence =</span> refSeq, </a>
<a class="sourceLine" id="cb6-51" data-line-number="51">      <span class="dt">querySequences =</span> genomes, </a>
<a class="sourceLine" id="cb6-52" data-line-number="52">      <span class="dt">type =</span> <span class="st">&quot;global-local&quot;</span>,</a>
<a class="sourceLine" id="cb6-53" data-line-number="53">      <span class="dt">outfile =</span> outfiles[featureName])</a>
<a class="sourceLine" id="cb6-54" data-line-number="54">    </a>
<a class="sourceLine" id="cb6-55" data-line-number="55">    allFeatureAlignments[[featureName]] &lt;-<span class="st"> </span>featureAlignmentsNto1</a>
<a class="sourceLine" id="cb6-56" data-line-number="56">  }</a>
<a class="sourceLine" id="cb6-57" data-line-number="57">  </a>
<a class="sourceLine" id="cb6-58" data-line-number="58">  <span class="co">## Choose window size  </span></a>
<a class="sourceLine" id="cb6-59" data-line-number="59">  <span class="cf">if</span> (featureLength <span class="op">&gt;</span><span class="st"> </span><span class="dv">20000</span>) {</a>
<a class="sourceLine" id="cb6-60" data-line-number="60">    windowSize &lt;-<span class="st"> </span><span class="dv">800</span> </a>
<a class="sourceLine" id="cb6-61" data-line-number="61">  } <span class="cf">else</span> <span class="cf">if</span> (featureLength <span class="op">&gt;</span><span class="st"> </span><span class="dv">5000</span>) {</a>
<a class="sourceLine" id="cb6-62" data-line-number="62">    windowSize &lt;-<span class="st"> </span><span class="dv">500</span> </a>
<a class="sourceLine" id="cb6-63" data-line-number="63">  } <span class="cf">else</span> <span class="cf">if</span> (featureLength <span class="op">&gt;</span><span class="st"> </span><span class="dv">3000</span>) {</a>
<a class="sourceLine" id="cb6-64" data-line-number="64">    windowSize &lt;-<span class="st"> </span><span class="dv">200</span> </a>
<a class="sourceLine" id="cb6-65" data-line-number="65">  } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb6-66" data-line-number="66">    windowSize &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="dv">50</span>, <span class="dv">10</span> <span class="op">*</span><span class="st"> </span><span class="kw">round</span>(featureLength <span class="op">/</span><span class="st"> </span><span class="dv">100</span>))</a>
<a class="sourceLine" id="cb6-67" data-line-number="67">  }</a>
<a class="sourceLine" id="cb6-68" data-line-number="68"></a>
<a class="sourceLine" id="cb6-69" data-line-number="69">  <span class="co">## Compute sequence order by decreasing PIP score</span></a>
<a class="sourceLine" id="cb6-70" data-line-number="70">  seqOrder &lt;-<span class="st"> </span><span class="kw">order</span>(featureAlignmentsNto1<span class="op">$</span>stat<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb6-71" data-line-number="71"></a>
<a class="sourceLine" id="cb6-72" data-line-number="72">  <span class="co">## PIP profile of one-to-N alignment</span></a>
<a class="sourceLine" id="cb6-73" data-line-number="73">  <span class="kw">plotPIPprofiles</span>(</a>
<a class="sourceLine" id="cb6-74" data-line-number="74"><span class="co">#    alignments = featureAlignmentsNto1$alignments[seqOrder],</span></a>
<a class="sourceLine" id="cb6-75" data-line-number="75">    <span class="dt">alignments =</span> featureAlignmentsNto1<span class="op">$</span>alignments,</a>
<a class="sourceLine" id="cb6-76" data-line-number="76">    <span class="dt">reversePlot =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb6-77" data-line-number="77">    <span class="dt">windowSize =</span> windowSize, </a>
<a class="sourceLine" id="cb6-78" data-line-number="78">    <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;PIP profiles: collection &quot;</span>, collection, </a>
<a class="sourceLine" id="cb6-79" data-line-number="79">                  <span class="st">&quot;Genomic feature &quot;</span>, featureName, <span class="st">&quot; (&quot;</span>, featureStart, <span class="st">&quot;:&quot;</span>, featureEnd ,<span class="st">&quot;, &quot;</span>, featureLength, <span class="st">&quot; bp)&quot;</span>), </a>
<a class="sourceLine" id="cb6-80" data-line-number="80">    <span class="co">#    colors = NULL,</span></a>
<a class="sourceLine" id="cb6-81" data-line-number="81">    <span class="dt">colors =</span> strainColors,</a>
<a class="sourceLine" id="cb6-82" data-line-number="82">    <span class="dt">legendMargin =</span> <span class="fl">0.25</span>, </a>
<a class="sourceLine" id="cb6-83" data-line-number="83">    <span class="dt">legendCorner =</span> <span class="st">&quot;topright&quot;</span>, <span class="dt">lwd =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb6-84" data-line-number="84">    <span class="dt">legendCex =</span> <span class="fl">0.5</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">100</span>))</a>
<a class="sourceLine" id="cb6-85" data-line-number="85">  </a>
<a class="sourceLine" id="cb6-86" data-line-number="86">  </a>
<a class="sourceLine" id="cb6-87" data-line-number="87">  <span class="kw">kable</span>(featureAlignmentsNto1<span class="op">$</span>stats[seqOrder,], </a>
<a class="sourceLine" id="cb6-88" data-line-number="88">        <span class="dt">caption =</span> <span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb6-89" data-line-number="89">          <span class="st">&quot;One-to-N alignment of feature &quot;</span>, </a>
<a class="sourceLine" id="cb6-90" data-line-number="90">          featureName))</a>
<a class="sourceLine" id="cb6-91" data-line-number="91">  </a>
<a class="sourceLine" id="cb6-92" data-line-number="92">}</a></code></pre></div>
<div id="ins1-pm120-21647-21907-261bp" class="section level3">
<h3>Ins1-pm120 (21647-21907; 261bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-1.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="ins2-pm120-21899-22156-258bp" class="section level3">
<h3>Ins2-pm120 (21899-22156; 258bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-2.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="ins3-pm120-21899-23152-1254bp" class="section level3">
<h3>Ins3-pm120 (21899-23152; 1254bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-3.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="ins4-pm120-23483-23734-252bp" class="section level3">
<h3>Ins4-pm120 (23483-23734; 252bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-4.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="ins4-m240-23363-23614-252bp" class="section level3">
<h3>Ins4-m240 (23363-23614; 252bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-5.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="recomb-xiao-22871-23092-222bp" class="section level3">
<h3>Recomb-Xiao (22871-23092; 222bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-6.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="recomb-reg-1-21500-22800-1301bp" class="section level3">
<h3>Recomb-reg-1 (21500-22800; 1301bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-7.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="recomb-reg-2-22800-24000-1201bp" class="section level3">
<h3>Recomb-reg-2 (22800-24000; 1201bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-8.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="recomb-reg-3-27800-28350-551bp" class="section level3">
<h3>Recomb-reg-3 (27800-28350; 551bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-9.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-s-21563-25384-3822bp" class="section level3">
<h3>CDS-S (21563-25384; 3822bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-10.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-orf3a-25393-26220-828bp" class="section level3">
<h3>CDS-ORF3a (25393-26220; 828bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-11.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-e-26245-26472-228bp" class="section level3">
<h3>CDS-E (26245-26472; 228bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-12.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-m-26523-27191-669bp" class="section level3">
<h3>CDS-M (26523-27191; 669bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-13.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-orf6-27202-27387-186bp" class="section level3">
<h3>CDS-ORF6 (27202-27387; 186bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-14.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-orf7a-27394-27759-366bp" class="section level3">
<h3>CDS-ORF7a (27394-27759; 366bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-15.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-orf8-27894-28259-366bp" class="section level3">
<h3>CDS-ORF8 (27894-28259; 366bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-16.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-n-28274-29533-1260bp" class="section level3">
<h3>CDS-N (28274-29533; 1260bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-17.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-orf10-29558-29674-117bp" class="section level3">
<h3>CDS-ORF10 (29558-29674; 117bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-18.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="cds-orf1ab-266-21555-21290bp" class="section level3">
<h3>CDS-ORF1ab (266-21555; 21290bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-19.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
</div>
<div id="after-orf1ab-21556-29899-8344bp" class="section level3">
<h3>After-ORF1ab (21556-29899; 8344bp)</h3>
<div class="figure" style="text-align: center">
<img src="figures/one-to-n-matches/one2N_feature_matches-20.png" alt="Feature-specific Percent Identical Positions (PIP) profiles. " width="100%" />
<p class="caption">
Feature-specific Percent Identical Positions (PIP) profiles.
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># }</span></a></code></pre></div>
<!-- ```{r one2N_feature_matches_PIP, fig.width=10, fig.height=6, out.width="100%", fig.cap="Feature-specific Percent Identical Positions (PIP) profiles. "} -->
<!-- #### One-to-N alignmemnt of user-speficied genomic features #### -->
<!-- allFeatureAlignments <- list() -->
<!-- # featureName <- "CDS-S" # for the test -->
<!-- featureName <- "S1" # for the test -->
<!-- # featureName <- "Recomb-reg-3" # for the test -->
<!-- # featureName <- "RBD" # for the test -->
<!-- # for (collection in (collections)) { -->
<!-- for (featureName in names(features)) { -->
<!--   featureLimits <- features[[featureName]] -->
<!--   featureStart <- featureLimits[["start"]] -->
<!--   featureEnd <- featureLimits[["end"]] -->
<!--   featureLength <-  -->
<!--      featureEnd - featureStart  + 1 -->
<!--   message("Searching matches for feature ", featureName,  -->
<!--           " (", featureLimits[1], "-", featureLimits[2], ")",  -->
<!--           " in collection ", collection) -->
<!--   #### N-to-1 alignments of spike-coding sequences #### -->
<!--   featurePrefix <- paste0( -->
<!--     featureName, -->
<!--     "_", collection) -->
<!--   dir[[featureName]] <- file.path(dir$seqdata, featureName) -->
<!--   dir.create(dir[[featureName]], showWarnings = FALSE, recursive = TRUE) -->
<!--   outfiles[featureName]  <- file.path( -->
<!--     dir[[featureName]], paste0(featurePrefix, ".fasta")) -->
<!--   message("\tOutput directory: ", dir[[featureName]] ) -->
<!--   ## Get sequence of the feature from the reference genome -->
<!--   refSeq <- subseq(genomes[refStrainName],  -->
<!--                    start = featureLimits[1],  -->
<!--                    end = featureLimits[2]) -->
<!--   featureAlignmentsNto1 <- allFeatureAlignments[[featureName]] -->
<!--   seqOrder <- order(featureAlignmentsNto1$stat$score, decreasing = TRUE) -->
<!--   if (featureLength > 20000) { -->
<!--     windowSize <- 800  -->
<!--   } else if (featureLength > 5000) { -->
<!--     windowSize <- 500  -->
<!--   } else if (featureLength > 3000) { -->
<!--     windowSize <- 200  -->
<!--   } else { -->
<!--     windowSize <- max(100, 10 * round(featureLength / 100)) -->
<!--   } -->
<!--   ## PIP profile of one-to-N alignment -->
<!--   plotPIPprofiles( -->
<!--     alignments = featureAlignmentsNto1$alignments[seqOrder], -->
<!--     reversePlot = TRUE, -->
<!--     windowSize = windowSize,  -->
<!--     main = paste0(featureName, " (", featureStart, ":", featureEnd ,", ", featureLength, " bp)"),  -->
<!-- #    colors = NULL, -->
<!--     colors = strainColors, -->
<!--     legendMargin = 0.25,  -->
<!--     legendCorner = "topright", lwd = 1, -->
<!--     legendCex = 0.5, ylim = c(30, 100)) -->
<!--   kable(featureAlignmentsNto1$stats[seqOrder,],  -->
<!--         caption = paste0( -->
<!--           "One-to-N alignment of feature ",  -->
<!--           featureName)) -->
<!-- } -->
<!-- # } -->
<!-- ``` -->
</div>
</div>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">kable</span>(<span class="kw">t</span>(<span class="kw">as.data.frame</span>(dir)), <span class="dt">col.names =</span> <span class="st">&quot;Dir&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Directories&quot;</span>)</a></code></pre></div>
<table>
<caption>Directories</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">Dir</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>main</td>
<td align="left">..</td>
</tr>
<tr class="even">
<td>R</td>
<td align="left">../scripts/R</td>
</tr>
<tr class="odd">
<td>seqdata</td>
<td align="left">../data</td>
</tr>
<tr class="even">
<td>images</td>
<td align="left">../memory_images</td>
</tr>
<tr class="odd">
<td>sequences</td>
<td align="left">../data/GISAID_genomes</td>
</tr>
<tr class="even">
<td>Ins1.pm120</td>
<td align="left">../data/Ins1-pm120</td>
</tr>
<tr class="odd">
<td>Ins2.pm120</td>
<td align="left">../data/Ins2-pm120</td>
</tr>
<tr class="even">
<td>Ins3.pm120</td>
<td align="left">../data/Ins3-pm120</td>
</tr>
<tr class="odd">
<td>Ins4.pm120</td>
<td align="left">../data/Ins4-pm120</td>
</tr>
<tr class="even">
<td>Ins4.m240</td>
<td align="left">../data/Ins4-m240</td>
</tr>
<tr class="odd">
<td>Recomb.Xiao</td>
<td align="left">../data/Recomb-Xiao</td>
</tr>
<tr class="even">
<td>Recomb.reg.1</td>
<td align="left">../data/Recomb-reg-1</td>
</tr>
<tr class="odd">
<td>Recomb.reg.2</td>
<td align="left">../data/Recomb-reg-2</td>
</tr>
<tr class="even">
<td>Recomb.reg.3</td>
<td align="left">../data/Recomb-reg-3</td>
</tr>
<tr class="odd">
<td>CDS.S</td>
<td align="left">../data/CDS-S</td>
</tr>
<tr class="even">
<td>CDS.ORF3a</td>
<td align="left">../data/CDS-ORF3a</td>
</tr>
<tr class="odd">
<td>CDS.E</td>
<td align="left">../data/CDS-E</td>
</tr>
<tr class="even">
<td>CDS.M</td>
<td align="left">../data/CDS-M</td>
</tr>
<tr class="odd">
<td>CDS.ORF6</td>
<td align="left">../data/CDS-ORF6</td>
</tr>
<tr class="even">
<td>CDS.ORF7a</td>
<td align="left">../data/CDS-ORF7a</td>
</tr>
<tr class="odd">
<td>CDS.ORF8</td>
<td align="left">../data/CDS-ORF8</td>
</tr>
<tr class="even">
<td>CDS.N</td>
<td align="left">../data/CDS-N</td>
</tr>
<tr class="odd">
<td>CDS.ORF10</td>
<td align="left">../data/CDS-ORF10</td>
</tr>
<tr class="even">
<td>CDS.ORF1ab</td>
<td align="left">../data/CDS-ORF1ab</td>
</tr>
<tr class="odd">
<td>After.ORF1ab</td>
<td align="left">../data/After-ORF1ab</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">outfileTable &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">path =</span> <span class="kw">as.vector</span>(outfiles))</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">outfileTable<span class="op">$</span>basename &lt;-<span class="st"> </span><span class="kw">basename</span>(<span class="kw">as.vector</span>(outfileTable<span class="op">$</span>path))</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">outfileTable<span class="op">$</span>dir &lt;-<span class="st"> </span><span class="kw">dirname</span>(<span class="kw">as.vector</span>(outfileTable<span class="op">$</span>path))</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">outfileTable<span class="op">$</span>link &lt;-<span class="st"> </span><span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">  <span class="st">&quot;[&quot;</span>, outfileTable<span class="op">$</span>basename, <span class="st">&quot;](&quot;</span>, outfileTable<span class="op">$</span>path, <span class="st">&quot;)&quot;</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6">)</a>
<a class="sourceLine" id="cb9-7" data-line-number="7"></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="kw">kable</span>(outfileTable[, <span class="kw">c</span>(<span class="st">&quot;dir&quot;</span>, <span class="st">&quot;link&quot;</span>)], </a>
<a class="sourceLine" id="cb9-9" data-line-number="9">      <span class="dt">col.names =</span> <span class="kw">c</span>(<span class="st">&quot;dir&quot;</span>, <span class="st">&quot;file&quot;</span>), </a>
<a class="sourceLine" id="cb9-10" data-line-number="10">      <span class="dt">caption =</span> <span class="st">&quot;Output files&quot;</span>)</a></code></pre></div>
<table>
<caption>Output files</caption>
<thead>
<tr class="header">
<th align="left">dir</th>
<th align="left">file</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">../memory_images</td>
<td align="left"><a href="../memory_images/around-CoV-2-plus-GISAID_one-to-n_matches.Rdata">around-CoV-2-plus-GISAID_one-to-n_matches.Rdata</a></td>
</tr>
<tr class="even">
<td align="left">../data/Ins1-pm120</td>
<td align="left"><a href="../data/Ins1-pm120/Ins1-pm120_around-CoV-2-plus-GISAID.fasta">Ins1-pm120_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/Ins2-pm120</td>
<td align="left"><a href="../data/Ins2-pm120/Ins2-pm120_around-CoV-2-plus-GISAID.fasta">Ins2-pm120_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/Ins3-pm120</td>
<td align="left"><a href="../data/Ins3-pm120/Ins3-pm120_around-CoV-2-plus-GISAID.fasta">Ins3-pm120_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/Ins4-pm120</td>
<td align="left"><a href="../data/Ins4-pm120/Ins4-pm120_around-CoV-2-plus-GISAID.fasta">Ins4-pm120_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/Ins4-m240</td>
<td align="left"><a href="../data/Ins4-m240/Ins4-m240_around-CoV-2-plus-GISAID.fasta">Ins4-m240_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/Recomb-Xiao</td>
<td align="left"><a href="../data/Recomb-Xiao/Recomb-Xiao_around-CoV-2-plus-GISAID.fasta">Recomb-Xiao_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/Recomb-reg-1</td>
<td align="left"><a href="../data/Recomb-reg-1/Recomb-reg-1_around-CoV-2-plus-GISAID.fasta">Recomb-reg-1_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/Recomb-reg-2</td>
<td align="left"><a href="../data/Recomb-reg-2/Recomb-reg-2_around-CoV-2-plus-GISAID.fasta">Recomb-reg-2_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/Recomb-reg-3</td>
<td align="left"><a href="../data/Recomb-reg-3/Recomb-reg-3_around-CoV-2-plus-GISAID.fasta">Recomb-reg-3_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/CDS-S</td>
<td align="left"><a href="../data/CDS-S/CDS-S_around-CoV-2-plus-GISAID.fasta">CDS-S_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/CDS-ORF3a</td>
<td align="left"><a href="../data/CDS-ORF3a/CDS-ORF3a_around-CoV-2-plus-GISAID.fasta">CDS-ORF3a_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/CDS-E</td>
<td align="left"><a href="../data/CDS-E/CDS-E_around-CoV-2-plus-GISAID.fasta">CDS-E_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/CDS-M</td>
<td align="left"><a href="../data/CDS-M/CDS-M_around-CoV-2-plus-GISAID.fasta">CDS-M_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/CDS-ORF6</td>
<td align="left"><a href="../data/CDS-ORF6/CDS-ORF6_around-CoV-2-plus-GISAID.fasta">CDS-ORF6_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/CDS-ORF7a</td>
<td align="left"><a href="../data/CDS-ORF7a/CDS-ORF7a_around-CoV-2-plus-GISAID.fasta">CDS-ORF7a_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/CDS-ORF8</td>
<td align="left"><a href="../data/CDS-ORF8/CDS-ORF8_around-CoV-2-plus-GISAID.fasta">CDS-ORF8_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/CDS-N</td>
<td align="left"><a href="../data/CDS-N/CDS-N_around-CoV-2-plus-GISAID.fasta">CDS-N_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/CDS-ORF10</td>
<td align="left"><a href="../data/CDS-ORF10/CDS-ORF10_around-CoV-2-plus-GISAID.fasta">CDS-ORF10_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="even">
<td align="left">../data/CDS-ORF1ab</td>
<td align="left"><a href="../data/CDS-ORF1ab/CDS-ORF1ab_around-CoV-2-plus-GISAID.fasta">CDS-ORF1ab_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
<tr class="odd">
<td align="left">../data/After-ORF1ab</td>
<td align="left"><a href="../data/After-ORF1ab/After-ORF1ab_around-CoV-2-plus-GISAID.fasta">After-ORF1ab_around-CoV-2-plus-GISAID.fasta</a></td>
</tr>
</tbody>
</table>
</div>
<div id="memory-image" class="section level2">
<h2>Memory image</h2>
<p>We store the result ina memory image, in oder to be able reloading it to plot PIP profiles with different parameters.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">save.image</span>(<span class="dt">file =</span> outfiles[<span class="st">&quot;Memory image&quot;</span>])</a></code></pre></div>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Biostrings_2.52.0   XVector_0.24.0      IRanges_2.18.3      S4Vectors_0.22.1    BiocGenerics_0.30.0 knitr_1.28         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4          digest_0.6.25       magrittr_1.5        evaluate_0.14       highr_0.8           zlibbioc_1.30.0     rlang_0.4.5         stringi_1.4.6       rmarkdown_2.1       tools_3.6.1         stringr_1.4.0       xfun_0.12           yaml_2.2.1          compiler_3.6.1     
[15] BiocManager_1.30.10 htmltools_0.4.0    </code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
