<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-05-15" />

<title>Phylogenetic analysis of coronavirus sequences</title>

<script src="phylogeny_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="phylogeny_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="phylogeny_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="phylogeny_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="phylogeny_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="phylogeny_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="phylogeny_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="phylogeny_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="phylogeny_files/navigation-1.1/tabsets.js"></script>
<script src="phylogeny_files/navigation-1.1/codefolding.js"></script>
<link href="phylogeny_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="phylogeny_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Phylogenetic analysis of coronavirus sequences</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-05-15</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">dir &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">dir[<span class="st">&quot;main&quot;</span>] &lt;-<span class="st"> &quot;../&quot;</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4">dir[<span class="st">&quot;results&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir[<span class="st">&quot;main&quot;</span>], <span class="st">&quot;results&quot;</span>)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">dir[<span class="st">&quot;genomes&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir[<span class="st">&quot;results&quot;</span>], <span class="st">&quot;genome_phylogeny&quot;</span>, <span class="st">&quot;clustalw_alignments&quot;</span>)</a></code></pre></div>
<div id="phylogeny-from-full-genomes" class="section level2">
<h2>Phylogeny from full genomes</h2>
<p>We inferred a phylogeny of virus strains based ontheir full genomes.</p>
<p>A multiple alignment of genome sequences was performed with a progressive method (<code>clustalw</code>). The tree of virus strains was inferred with a maximum likelihood approach (<code>phyml</code> software).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">genomeTreeFile &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">  dir[<span class="st">&quot;genomes&quot;</span>],</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">  <span class="st">&quot;coronavirus_selected-plus-GISAID_genomes_clustalw_gblocks.phy_phyml_tree.phb&quot;</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">## Load the tree with phytools</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">genomeTree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">read.tree</span>(<span class="dt">file =</span> genomeTreeFile)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co"># genomeTree &lt;- phytools::read.newick(file = genomeTreeFile)</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co"># plot(genomeTree)</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co"># nodelabels()</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">## Enroot the tree</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co"># genomeTree$tip.label</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co"># outgroup &lt;- c(&quot;HuOC43&quot;,</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#               &quot;CmMERS&quot;,</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#               &quot;HuMERS_17&quot;,</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#               &quot;PiPRCV&quot;,</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#               &quot;HuTGEV&quot;,</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#               &quot;PiSADS&quot;,</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="co">#               &quot;Hu229E&quot;,</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">#               &quot;HuNL63&quot;)</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="co"># genomeTree &lt;- root(genomeTree, outgroup = outgroup, resolve.root = TRUE)</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co">## Define the root</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co"># nodelabels()</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25">rootNode &lt;-<span class="st"> </span><span class="dv">41</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26">genomeTree &lt;-<span class="st"> </span><span class="kw">reroot</span>(genomeTree, <span class="dt">node =</span> rootNode)</a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co"># #genomeTree &lt;-reroot(genomeTree, interactive=TRUE)</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28">genomeTree &lt;-<span class="st"> </span><span class="kw">rotate</span>(genomeTree, <span class="dv">39</span>)</a>
<a class="sourceLine" id="cb2-29" data-line-number="29">genomeTree &lt;-<span class="st"> </span><span class="kw">rotate</span>(genomeTree, <span class="dv">44</span>)</a>
<a class="sourceLine" id="cb2-30" data-line-number="30">genomeTree &lt;-<span class="st"> </span><span class="kw">rotate</span>(genomeTree, <span class="dv">45</span>)</a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="co"># nodelabels()</span></a>
<a class="sourceLine" id="cb2-32" data-line-number="32"></a>
<a class="sourceLine" id="cb2-33" data-line-number="33"><span class="cf">if</span> (<span class="kw">is.binary</span>(genomeTree)) {</a>
<a class="sourceLine" id="cb2-34" data-line-number="34">  treeType &lt;-<span class="st"> &quot;binary&quot;</span></a>
<a class="sourceLine" id="cb2-35" data-line-number="35">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb2-36" data-line-number="36">  treeType &lt;-<span class="st"> &quot;non-binary&quot;</span></a>
<a class="sourceLine" id="cb2-37" data-line-number="37">}</a>
<a class="sourceLine" id="cb2-38" data-line-number="38"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Loaded &quot;</span>, treeType, <span class="st">&quot; genome tree&quot;</span>, </a>
<a class="sourceLine" id="cb2-39" data-line-number="39">        <span class="st">&quot; with &quot;</span>, <span class="kw">Ntip</span>(genomeTree), <span class="st">&quot; tips&quot;</span>,</a>
<a class="sourceLine" id="cb2-40" data-line-number="40">        <span class="st">&quot; and &quot;</span>, <span class="kw">Nnode</span>(genomeTree), <span class="st">&quot; internal nodes&quot;</span>)</a>
<a class="sourceLine" id="cb2-41" data-line-number="41"></a>
<a class="sourceLine" id="cb2-42" data-line-number="42"><span class="co">## Check theinternal structure of the tree</span></a>
<a class="sourceLine" id="cb2-43" data-line-number="43"><span class="co"># str(genomeTree)</span></a>
<a class="sourceLine" id="cb2-44" data-line-number="44"><span class="co"># genomeTree$node.label</span></a>
<a class="sourceLine" id="cb2-45" data-line-number="45"><span class="co">#  dim(genomeTree$edge)</span></a>
<a class="sourceLine" id="cb2-46" data-line-number="46"><span class="co">#  genomeTree$tip.label</span></a>
<a class="sourceLine" id="cb2-47" data-line-number="47"><span class="co">#  class(genomeTree)</span></a>
<a class="sourceLine" id="cb2-48" data-line-number="48"></a>
<a class="sourceLine" id="cb2-49" data-line-number="49"><span class="co">## Color palette per species</span></a>
<a class="sourceLine" id="cb2-50" data-line-number="50">speciesPalette &lt;-<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb2-51" data-line-number="51">  <span class="dt">Human =</span> <span class="st">&quot;#880000&quot;</span>,</a>
<a class="sourceLine" id="cb2-52" data-line-number="52">  <span class="dt">Bat =</span> <span class="st">&quot;#BBBBBB&quot;</span>,</a>
<a class="sourceLine" id="cb2-53" data-line-number="53">  <span class="dt">Pangolin =</span> <span class="st">&quot;#00BB00&quot;</span>,</a>
<a class="sourceLine" id="cb2-54" data-line-number="54">  <span class="dt">Camel =</span> <span class="st">&quot;#BB8800&quot;</span>,</a>
<a class="sourceLine" id="cb2-55" data-line-number="55">  <span class="dt">Pig =</span> <span class="st">&quot;#FFBBBB&quot;</span>,</a>
<a class="sourceLine" id="cb2-56" data-line-number="56">  <span class="dt">Civet =</span> <span class="st">&quot;#0000BB&quot;</span></a>
<a class="sourceLine" id="cb2-57" data-line-number="57">)</a>
<a class="sourceLine" id="cb2-58" data-line-number="58"></a>
<a class="sourceLine" id="cb2-59" data-line-number="59"></a>
<a class="sourceLine" id="cb2-60" data-line-number="60"><span class="co">## Tip parametes</span></a>
<a class="sourceLine" id="cb2-61" data-line-number="61">tipParam &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb2-62" data-line-number="62">  <span class="dt">row.names =</span> genomeTree<span class="op">$</span>tip.label,</a>
<a class="sourceLine" id="cb2-63" data-line-number="63">  <span class="dt">color =</span> <span class="kw">rep</span>(<span class="dt">x =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(genomeTree<span class="op">$</span>tip.label)))</a>
<a class="sourceLine" id="cb2-64" data-line-number="64">tipParam<span class="op">$</span>species &lt;-<span class="st"> &quot;NONE&quot;</span></a>
<a class="sourceLine" id="cb2-65" data-line-number="65"></a>
<a class="sourceLine" id="cb2-66" data-line-number="66"><span class="co">## Identify species per tip</span></a>
<a class="sourceLine" id="cb2-67" data-line-number="67">speciesPrefix &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Hu&quot;</span> =<span class="st"> &quot;Human&quot;</span>,</a>
<a class="sourceLine" id="cb2-68" data-line-number="68">                   <span class="st">&quot;Bt&quot;</span> =<span class="st"> &quot;Bat&quot;</span>,</a>
<a class="sourceLine" id="cb2-69" data-line-number="69">                   <span class="st">&quot;Pn&quot;</span> =<span class="st"> &quot;Pangolin&quot;</span>,</a>
<a class="sourceLine" id="cb2-70" data-line-number="70">                   <span class="st">&quot;Cm&quot;</span> =<span class="st"> &quot;Camel&quot;</span>,</a>
<a class="sourceLine" id="cb2-71" data-line-number="71">                   <span class="st">&quot;Pi&quot;</span> =<span class="st"> &quot;Pig&quot;</span>,</a>
<a class="sourceLine" id="cb2-72" data-line-number="72">                   <span class="st">&quot;Cv&quot;</span> =<span class="st"> &quot;Civet&quot;</span>)</a>
<a class="sourceLine" id="cb2-73" data-line-number="73"><span class="cf">for</span> (prefix <span class="cf">in</span> <span class="kw">names</span>(speciesPrefix)) {</a>
<a class="sourceLine" id="cb2-74" data-line-number="74">  tipParam[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, prefix), <span class="dt">x =</span> <span class="kw">row.names</span>(tipParam), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;species&quot;</span>] &lt;-<span class="st"> </span>speciesPrefix[prefix]</a>
<a class="sourceLine" id="cb2-75" data-line-number="75">  </a>
<a class="sourceLine" id="cb2-76" data-line-number="76">}</a>
<a class="sourceLine" id="cb2-77" data-line-number="77"><span class="co"># table(tipParam$species)</span></a>
<a class="sourceLine" id="cb2-78" data-line-number="78"><span class="co"># View(tipParam)</span></a>
<a class="sourceLine" id="cb2-79" data-line-number="79"></a>
<a class="sourceLine" id="cb2-80" data-line-number="80"><span class="co">## Assign tip color accoding to species</span></a>
<a class="sourceLine" id="cb2-81" data-line-number="81">tipParam<span class="op">$</span>color &lt;-<span class="st"> </span><span class="kw">unlist</span>(speciesPalette[tipParam<span class="op">$</span>species])</a>
<a class="sourceLine" id="cb2-82" data-line-number="82"><span class="co"># table(tipParam$color, tipParam$species)</span></a>
<a class="sourceLine" id="cb2-83" data-line-number="83"><span class="co"># table(tipParam$color)</span></a>
<a class="sourceLine" id="cb2-84" data-line-number="84"></a>
<a class="sourceLine" id="cb2-85" data-line-number="85"><span class="co">## Identify CoV2 tips</span></a>
<a class="sourceLine" id="cb2-86" data-line-number="86">cov2Tips &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="st">&quot;CoV2&quot;</span>, <span class="dt">x =</span> genomeTree<span class="op">$</span>tip.label, <span class="dt">perl =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#### Plot the genome tree ####</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">genomeTree &lt;-<span class="st"> </span><span class="kw">paintSubTree</span>(<span class="dt">tree =</span> genomeTree, <span class="dt">node =</span> <span class="dv">49</span>, <span class="dt">state =</span> <span class="st">&quot;CoV2&quot;</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">## Plot the tree</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">ape<span class="op">::</span><span class="kw">plot.phylo</span>(genomeTree, </a>
<a class="sourceLine" id="cb3-7" data-line-number="7">                <span class="dt">type =</span> <span class="st">&quot;phylogram&quot;</span>, </a>
<a class="sourceLine" id="cb3-8" data-line-number="8">                <span class="dt">show.node.label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">                <span class="dt">show.tip.label =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">                <span class="dt">edge.color =</span> <span class="st">&quot;grey&quot;</span>,</a>
<a class="sourceLine" id="cb3-11" data-line-number="11">                <span class="dt">tip.color =</span> tipParam<span class="op">$</span>color,</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">                <span class="co">#         ftype = &quot;off&quot;,</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13">                <span class="dt">edge.width =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">                <span class="dt">label.offset =</span> <span class="fl">0.1</span>, </a>
<a class="sourceLine" id="cb3-15" data-line-number="15">                <span class="dt">lwd =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb3-16" data-line-number="16">                <span class="co"># fsize = 0.7,</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17">                <span class="dt">cex =</span> <span class="fl">0.7</span>,</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">                <span class="dt">color =</span> speciesPalette<span class="op">$</span>Bat, </a>
<a class="sourceLine" id="cb3-19" data-line-number="19">                <span class="dt">nomargin =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb3-20" data-line-number="20">                <span class="dt">font =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="kw">add.scale.bar</span>(<span class="dt">cex =</span> <span class="dv">1</span>, <span class="dt">font =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="dt">length =</span> <span class="fl">0.2</span>, </a>
<a class="sourceLine" id="cb3-22" data-line-number="22">              <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">lcol =</span> <span class="st">&quot;blue&quot;</span>)</a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co"># add.arrow(genomeTree, tip = cov2Tips, </span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#           arrl = 0.3, </span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#           col = speciesPalette$Human)</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co"># length(genomeTree$node.label)</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co"># nodelabels(cex = 0.5)</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co"># sort(as.numeric(genomeTree$node.label))</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co"># tiplabels()</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">## Identify some clades</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="kw">cladelabels</span>(genomeTree, <span class="st">&quot;CoV2&quot;</span>, <span class="dv">49</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>, <span class="dt">orientation =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="dt">offset =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="kw">cladelabels</span>(genomeTree, <span class="st">&quot;MERS&quot;</span>,<span class="dv">46</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>, <span class="dt">orientation =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="dt">offset =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="kw">cladelabels</span>(genomeTree, <span class="st">&quot;SARS&quot;</span>, <span class="dv">72</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>, <span class="dt">orientation =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="dt">offset =</span> <span class="dv">3</span>)</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_genome_tree-1.png" alt="Genome tree of selected coronaviruses.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="80%" />
<p class="caption">
Genome tree of selected coronaviruses. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
