<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-05-17" />

<title>Phylogenetic analysis of coronavirus sequences</title>

<script src="phylogeny_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="phylogeny_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="phylogeny_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="phylogeny_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="phylogeny_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="phylogeny_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="phylogeny_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="phylogeny_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="phylogeny_files/navigation-1.1/tabsets.js"></script>
<script src="phylogeny_files/navigation-1.1/codefolding.js"></script>
<link href="phylogeny_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="phylogeny_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Phylogenetic analysis of coronavirus sequences</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-05-17</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">## Color palette per species</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">speciesPalette &lt;-<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">  <span class="dt">Human =</span> <span class="st">&quot;#880000&quot;</span>,</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">  <span class="dt">Bat =</span> <span class="st">&quot;#BBBBBB&quot;</span>,</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">  <span class="dt">Pangolin =</span> <span class="st">&quot;#00BB00&quot;</span>,</a>
<a class="sourceLine" id="cb1-6" data-line-number="6">  <span class="dt">Camel =</span> <span class="st">&quot;#BB8800&quot;</span>,</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">  <span class="dt">Pig =</span> <span class="st">&quot;#FFBBBB&quot;</span>,</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">  <span class="dt">Civet =</span> <span class="st">&quot;#0000BB&quot;</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co">## Species prefix in the tip labels</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12">speciesPrefix &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Hu&quot;</span> =<span class="st"> &quot;Human&quot;</span>,</a>
<a class="sourceLine" id="cb1-13" data-line-number="13">                   <span class="st">&quot;Bt&quot;</span> =<span class="st"> &quot;Bat&quot;</span>,</a>
<a class="sourceLine" id="cb1-14" data-line-number="14">                   <span class="st">&quot;Pn&quot;</span> =<span class="st"> &quot;Pangolin&quot;</span>,</a>
<a class="sourceLine" id="cb1-15" data-line-number="15">                   <span class="st">&quot;Cm&quot;</span> =<span class="st"> &quot;Camel&quot;</span>,</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">                   <span class="st">&quot;Pi&quot;</span> =<span class="st"> &quot;Pig&quot;</span>,</a>
<a class="sourceLine" id="cb1-17" data-line-number="17">                   <span class="st">&quot;Cv&quot;</span> =<span class="st"> &quot;Civet&quot;</span>)</a>
<a class="sourceLine" id="cb1-18" data-line-number="18"></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="co">## Define feature types</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20">features &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-21" data-line-number="21">  <span class="co"># &quot;genomes&quot;, </span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22">  <span class="co"># &quot;S-gene&quot;, </span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23">  <span class="st">&quot;S1&quot;</span>,  </a>
<a class="sourceLine" id="cb1-24" data-line-number="24">  <span class="st">&quot;S2&quot;</span>, </a>
<a class="sourceLine" id="cb1-25" data-line-number="25">  <span class="st">&quot;RBD&quot;</span>, </a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">#  &quot;Recomb-Xiao&quot;,</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27">  <span class="st">&quot;Recomb-reg-1&quot;</span>,</a>
<a class="sourceLine" id="cb1-28" data-line-number="28">  <span class="st">&quot;Recomb-reg-2&quot;</span>,</a>
<a class="sourceLine" id="cb1-29" data-line-number="29">  <span class="st">&quot;Recomb-reg-3&quot;</span>,</a>
<a class="sourceLine" id="cb1-30" data-line-number="30">  <span class="st">&quot;CDS-ORF1ab&quot;</span>,</a>
<a class="sourceLine" id="cb1-31" data-line-number="31">  <span class="st">&quot;CDS-S&quot;</span>,</a>
<a class="sourceLine" id="cb1-32" data-line-number="32">  <span class="st">&quot;CDS-ORF3a&quot;</span>,</a>
<a class="sourceLine" id="cb1-33" data-line-number="33">  <span class="st">&quot;CDS-E&quot;</span>,</a>
<a class="sourceLine" id="cb1-34" data-line-number="34">  <span class="st">&quot;CDS-M&quot;</span>,</a>
<a class="sourceLine" id="cb1-35" data-line-number="35">  <span class="st">&quot;CDS-ORF6&quot;</span>,</a>
<a class="sourceLine" id="cb1-36" data-line-number="36">  <span class="st">&quot;CDS-ORF7a&quot;</span>,</a>
<a class="sourceLine" id="cb1-37" data-line-number="37">  <span class="st">&quot;CDS-ORF8&quot;</span>,</a>
<a class="sourceLine" id="cb1-38" data-line-number="38">  <span class="st">&quot;CDS-N&quot;</span>,</a>
<a class="sourceLine" id="cb1-39" data-line-number="39">  <span class="st">&quot;CDS-ORF10&quot;</span>)</a>
<a class="sourceLine" id="cb1-40" data-line-number="40">feature &lt;-<span class="st"> &quot;S-gene&quot;</span></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"></a>
<a class="sourceLine" id="cb1-42" data-line-number="42"><span class="co">## Define collections</span></a>
<a class="sourceLine" id="cb1-43" data-line-number="43">collections &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;around-CoV-2&quot;</span>, <span class="st">&quot;selected&quot;</span>)</a>
<a class="sourceLine" id="cb1-44" data-line-number="44">collection &lt;-<span class="st"> &quot;around-CoV-2&quot;</span> <span class="co"># default for testing</span></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"></a>
<a class="sourceLine" id="cb1-46" data-line-number="46"><span class="co">## Outgroup per collection</span></a>
<a class="sourceLine" id="cb1-47" data-line-number="47">outgroups &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-48" data-line-number="48">outgroups[[<span class="st">&quot;selected&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-49" data-line-number="49">  <span class="st">&quot;HuOC43&quot;</span>,</a>
<a class="sourceLine" id="cb1-50" data-line-number="50">  <span class="st">&quot;PiPRCV&quot;</span>,</a>
<a class="sourceLine" id="cb1-51" data-line-number="51">  <span class="st">&quot;HuTGEV&quot;</span>,</a>
<a class="sourceLine" id="cb1-52" data-line-number="52">  <span class="st">&quot;PiSADS&quot;</span>,</a>
<a class="sourceLine" id="cb1-53" data-line-number="53">  <span class="st">&quot;Hu229E&quot;</span>,</a>
<a class="sourceLine" id="cb1-54" data-line-number="54">  <span class="st">&quot;HuNL63&quot;</span>)</a>
<a class="sourceLine" id="cb1-55" data-line-number="55">outgroups[[<span class="st">&quot;around-CoV-2&quot;</span>]] &lt;-<span class="st"> &quot;BtBM48-31&quot;</span></a>
<a class="sourceLine" id="cb1-56" data-line-number="56"></a>
<a class="sourceLine" id="cb1-57" data-line-number="57"><span class="co">## Use GISAID data</span></a>
<a class="sourceLine" id="cb1-58" data-line-number="58">useGISAID &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb1-59" data-line-number="59"><span class="cf">if</span> (useGISAID) {</a>
<a class="sourceLine" id="cb1-60" data-line-number="60">  collections &lt;-<span class="st"> </span><span class="kw">paste0</span>(collections, <span class="st">&quot;-plus-GISAID&quot;</span>)</a>
<a class="sourceLine" id="cb1-61" data-line-number="61">  <span class="cf">for</span> (collection <span class="cf">in</span> <span class="kw">names</span>(outgroups)) {</a>
<a class="sourceLine" id="cb1-62" data-line-number="62">    outgroups[[<span class="kw">paste0</span>(collection, <span class="st">&quot;-plus-GISAID&quot;</span>)]] &lt;-<span class="st"> </span>outgroups[[collection]]</a>
<a class="sourceLine" id="cb1-63" data-line-number="63">  }</a>
<a class="sourceLine" id="cb1-64" data-line-number="64">  collection &lt;-<span class="st"> </span><span class="kw">paste0</span>(collection, <span class="st">&quot;-plus-GISAID&quot;</span>)</a>
<a class="sourceLine" id="cb1-65" data-line-number="65">}</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">dir &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">dir[<span class="st">&quot;main&quot;</span>] &lt;-<span class="st"> &quot;..&quot;</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">dir[<span class="st">&quot;results&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir[<span class="st">&quot;main&quot;</span>], <span class="st">&quot;results&quot;</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">dir[<span class="st">&quot;genomes&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir[<span class="st">&quot;results&quot;</span>], <span class="st">&quot;genome_phylogeny&quot;</span>, <span class="st">&quot;clustalw_alignments&quot;</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">dir[<span class="st">&quot;R&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir[<span class="st">&quot;main&quot;</span>], <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;R&quot;</span>)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co"># list.files(dir[&quot;R&quot;])</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="kw">source</span>(<span class="kw">file.path</span>(dir[<span class="st">&quot;R&quot;</span>], <span class="st">&quot;load_tree.R&quot;</span>))</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">source</span>(<span class="kw">file.path</span>(dir[<span class="st">&quot;R&quot;</span>], <span class="st">&quot;plot_my_tree.R&quot;</span>))</a></code></pre></div>
<div id="phylogeny-from-full-genomes" class="section level2">
<h2>Phylogeny from full genomes</h2>
<p>We inferred a phylogeny of virus strains based ontheir full genomes.</p>
<p>A multiple alignment of genome sequences was performed with a progressive method (<code>clustalw</code>). The tree of virus strains was inferred with a maximum likelihood approach (<code>phyml</code> software).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#### Load and plot the genome tree ####</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">genomeTreeFile &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">  dir[<span class="st">&quot;genomes&quot;</span>],</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">  <span class="st">&quot;coronavirus_selected-plus-GISAID_genomes_clustalw_gblocks.phy_phyml_tree.phb&quot;</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7">genomeTree &lt;-<span class="st"> </span><span class="kw">loadTree</span>(</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">  <span class="dt">treeFile =</span> genomeTreeFile, </a>
<a class="sourceLine" id="cb3-9" data-line-number="9">  <span class="dt">outgroup =</span> outgroups[[<span class="st">&#39;selected&#39;</span>]], </a>
<a class="sourceLine" id="cb3-10" data-line-number="10">  <span class="dt">rootNode =</span> <span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb3-11" data-line-number="11">  <span class="dt">nodesToRotate =</span> <span class="kw">c</span>(<span class="dv">39</span>, <span class="dv">75</span>, <span class="dv">42</span>))</a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co"># genomeTree &lt;- paintSubTree(tree = genomeTree, node = 49, state = &quot;CoV2&quot;)</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="kw">plotMyTree</span>(genomeTree, <span class="dt">main  =</span> <span class="st">&quot;Genome-based virus tree&quot;</span>, </a>
<a class="sourceLine" id="cb3-15" data-line-number="15">           <span class="dt">scaleLength =</span> <span class="fl">0.2</span>,</a>
<a class="sourceLine" id="cb3-16" data-line-number="16">           <span class="dt">show.node.label =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co"># nodelabels(cex = 0.4)</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">## Identify some clades</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="kw">cladelabels</span>(genomeTree<span class="op">$</span>tree, <span class="st">&quot;CoV2&quot;</span>, <span class="dv">46</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>, <span class="dt">orientation =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="dt">offset =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="kw">cladelabels</span>(genomeTree<span class="op">$</span>tree, <span class="st">&quot;MERS&quot;</span>, <span class="dv">43</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>, <span class="dt">orientation =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="dt">offset =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="kw">cladelabels</span>(genomeTree<span class="op">$</span>tree, <span class="st">&quot;SARS&quot;</span>, <span class="dv">69</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>, <span class="dt">orientation =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="dt">offset =</span> <span class="dv">5</span>)</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_genome_tree-1.png" alt="Genome tree of selected coronaviruses.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Genome tree of selected coronaviruses. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
</div>
<div id="tree-per-feature" class="section level2">
<h2>Tree per feature</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">#### Load and plot feature-specific trees ####</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">## Define vectors to hold the results and enable tree comparisons</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">treeFiles &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">treeData &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb4-6" data-line-number="6"></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">## Define the path to the tre file</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8">feature &lt;-<span class="st"> &quot;S1&quot;</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="cf">for</span> (feature <span class="cf">in</span> features) {</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\n\t</span><span class="st">Reading tree for feature &quot;</span>, feature)</a>
<a class="sourceLine" id="cb4-12" data-line-number="12">  prefix &lt;-<span class="st"> </span><span class="kw">paste0</span>(feature, <span class="st">&quot;_&quot;</span>, collection)</a>
<a class="sourceLine" id="cb4-13" data-line-number="13">  treeFile &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">    dir[<span class="st">&quot;results&quot;</span>],</a>
<a class="sourceLine" id="cb4-15" data-line-number="15">    prefix, </a>
<a class="sourceLine" id="cb4-16" data-line-number="16">    <span class="kw">paste0</span>(prefix, <span class="st">&quot;_clustalw_gblocks.phy_phyml_tree_GTR.phb&quot;</span>))</a>
<a class="sourceLine" id="cb4-17" data-line-number="17">  treeFiles[feature] &lt;-<span class="st"> </span>treeFile</a>
<a class="sourceLine" id="cb4-18" data-line-number="18">  </a>
<a class="sourceLine" id="cb4-19" data-line-number="19">  </a>
<a class="sourceLine" id="cb4-20" data-line-number="20">  </a>
<a class="sourceLine" id="cb4-21" data-line-number="21">  <span class="co">## Load the  tree</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22">  treeData[[feature]] &lt;-<span class="st"> </span><span class="kw">loadTree</span>(</a>
<a class="sourceLine" id="cb4-23" data-line-number="23">    <span class="dt">treeFile =</span> treeFiles[feature], </a>
<a class="sourceLine" id="cb4-24" data-line-number="24">    <span class="dt">outgroup =</span> outgroups[[collection]], </a>
<a class="sourceLine" id="cb4-25" data-line-number="25">    <span class="dt">rootNode =</span> <span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb4-26" data-line-number="26">    <span class="dt">nodesToRotate =</span> <span class="ot">NULL</span>)</a>
<a class="sourceLine" id="cb4-27" data-line-number="27">  </a>
<a class="sourceLine" id="cb4-28" data-line-number="28">  <span class="kw">plotMyTree</span>(treeData[[feature]], </a>
<a class="sourceLine" id="cb4-29" data-line-number="29">             <span class="dt">main  =</span> <span class="kw">paste0</span>(feature, <span class="st">&quot; tree&quot;</span>), </a>
<a class="sourceLine" id="cb4-30" data-line-number="30">             <span class="dt">scaleLength =</span> <span class="fl">0.02</span>,</a>
<a class="sourceLine" id="cb4-31" data-line-number="31">             <span class="dt">show.node.label =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb4-32" data-line-number="32">  </a>
<a class="sourceLine" id="cb4-33" data-line-number="33">  <span class="co"># ## Identify some clades</span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34">  <span class="co"># cladelabels(genomeTree$tree, &quot;CoV2&quot;, 46, cex = 0.7, orientation = &quot;horizontal&quot;, offset = 5)</span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35">  <span class="co"># cladelabels(genomeTree$tree, &quot;MERS&quot;, 43, cex = 0.7, orientation = &quot;horizontal&quot;, offset = 5)</span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36">  <span class="co"># cladelabels(genomeTree$tree, &quot;SARS&quot;, 69, cex = 0.7, orientation = &quot;horizontal&quot;, offset = 5)</span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37">}</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-1.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-2.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-3.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-4.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-5.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-6.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-7.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-8.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-9.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-10.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-11.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-12.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-13.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-14.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-15.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
<div class="figure" style="text-align: center">
<img src="figures/phylogeny/CoV_feature_tree-16.png" alt="Feature-specific tree.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). " width="100%" />
<p class="caption">
Feature-specific tree. The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw).
</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
