---
title: "Phylogenetic analysis of coronavirus sequences"
author: "Jacques van Helden"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    code_folding: hide
    fig_caption: yes
    highlight: zenburn
    self_contained: no
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  ioslides_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    smaller: yes
    toc: yes
    widescreen: yes
  beamer_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    incremental: no
    keep_tex: no
    slide_level: 2
    theme: Montpellier
    toc: yes
  word_document:
    toc: yes
    toc_depth: '3'
  slidy_presentation:
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    highlight: tango
    incremental: no
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
font-import: http://fonts.googleapis.com/css?family=Risque
font-family: Garamond
transition: linear
---

```{r libraries, echo=FALSE, results=FALSE, warning=FALSE, message=FALSE}
#### Install required packages ####
required.packages <- c("knitr", "ape", "phytools")

for (pkg in required.packages) {
  if (!require(pkg, character.only = TRUE)) {
    message("Installing package ", pkg)
    install.packages(pkg, dependencies = TRUE)
  }
  require(pkg, character.only = TRUE)
}


```



```{r knitr_settings, include=FALSE, echo=FALSE, eval=TRUE}
library(knitr)
options(width = 300)
knitr::opts_chunk$set(
  fig.width = 7, fig.height = 5, 
  fig.path = 'figures/phylogeny/CoV_',
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = TRUE, comment = "")
# knitr::asis_output("\\footnotesize")


## Store original graphic parameters to restore them after chunks
par.ori <- par(no.readonly = TRUE)


```

```{r directories}
dir <- vector()
dir["main"] <- "../"

dir["results"] <- file.path(dir["main"], "results")
dir["genomes"] <- file.path(dir["results"], "genome_phylogeny", "clustalw_alignments")



```






## Phylogeny from full genomes

We inferred a phylogeny of virus strains based ontheir full genomes. 

A multiple alignment of genome sequences was performed with a progressive method (`clustalw`). The tree of virus strains was inferred with a maximum likelihood approach (`phyml` software).

```{r load_genome_tree}
genomeTreeFile <- file.path(
  dir["genomes"],
  "coronavirus_selected-plus-GISAID_genomes_clustalw_gblocks.phy_phyml_tree.phb")

## Load the tree with phytools
genomeTree <- ape::read.tree(file = genomeTreeFile)
# genomeTree <- phytools::read.newick(file = genomeTreeFile)
# plot(genomeTree)
# nodelabels()

## Enroot the tree
# genomeTree$tip.label
# outgroup <- c("HuOC43",
#               "CmMERS",
#               "HuMERS_17",
#               "PiPRCV",
#               "HuTGEV",
#               "PiSADS",
#               "Hu229E",
#               "HuNL63")
# genomeTree <- root(genomeTree, outgroup = outgroup, resolve.root = TRUE)

## Define the root
# nodelabels()
rootNode <- 41
genomeTree <- reroot(genomeTree, node = rootNode)
# #genomeTree <-reroot(genomeTree, interactive=TRUE)
genomeTree <- rotate(genomeTree, 39)
genomeTree <- rotate(genomeTree, 44)
genomeTree <- rotate(genomeTree, 45)
# nodelabels()

if (is.binary(genomeTree)) {
  treeType <- "binary"
} else {
  treeType <- "non-binary"
}
message("\tLoaded ", treeType, " genome tree", 
        " with ", Ntip(genomeTree), " tips",
        " and ", Nnode(genomeTree), " internal nodes")

## Check theinternal structure of the tree
# str(genomeTree)
# genomeTree$node.label
#  dim(genomeTree$edge)
#  genomeTree$tip.label
#  class(genomeTree)

## Color palette per species
speciesPalette <- list(
  Human = "#880000",
  Bat = "#BBBBBB",
  Pangolin = "#00BB00",
  Camel = "#BB8800",
  Pig = "#FFBBBB",
  Civet = "#0000BB"
)


## Tip parametes
tipParam <- data.frame(
  row.names = genomeTree$tip.label,
  color = rep(x = "grey", length.out = length(genomeTree$tip.label)))
tipParam$species <- "NONE"

## Identify species per tip
speciesPrefix <- c("Hu" = "Human",
                   "Bt" = "Bat",
                   "Pn" = "Pangolin",
                   "Cm" = "Camel",
                   "Pi" = "Pig",
                   "Cv" = "Civet")
for (prefix in names(speciesPrefix)) {
  tipParam[grep(pattern = paste0("^", prefix), x = row.names(tipParam), perl = TRUE), "species"] <- speciesPrefix[prefix]
  
}
# table(tipParam$species)
# View(tipParam)

## Assign tip color accoding to species
tipParam$color <- unlist(speciesPalette[tipParam$species])
# table(tipParam$color, tipParam$species)
# table(tipParam$color)

## Identify CoV2 tips
cov2Tips <- grep(pattern = "CoV2", x = genomeTree$tip.label, perl = TRUE, value = TRUE)





```



```{r genome_tree, fig.width=7, fig.height=10,  out.width="80%", fig.cap="Genome tree of selected coronaviruses.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). "}

#### Plot the genome tree ####

genomeTree <- paintSubTree(tree = genomeTree, node = 49, state = "CoV2")

## Plot the tree
ape::plot.phylo(genomeTree, 
                type = "phylogram", 
                show.node.label = FALSE,
                show.tip.label = TRUE,
                edge.color = "grey",
                tip.color = tipParam$color,
                #         ftype = "off",
                edge.width = 2,
                label.offset = 0.1, 
                lwd = 2,
                # fsize = 0.7,
                cex = 0.7,
                color = speciesPalette$Bat, 
                nomargin = FALSE,
                font = 1)
add.scale.bar(cex = 1, font = 2, col = "blue", length = 0.2, 
              lwd = 2, lcol = "blue")
# add.arrow(genomeTree, tip = cov2Tips, 
#           arrl = 0.3, 
#           col = speciesPalette$Human)

# length(genomeTree$node.label)
# nodelabels(cex = 0.5)
# sort(as.numeric(genomeTree$node.label))
# tiplabels()

## Identify some clades
cladelabels(genomeTree, "CoV2", 49, cex = 0.7, orientation = "horizontal", offset = 5)
cladelabels(genomeTree, "MERS",46, cex = 0.7, orientation = "horizontal", offset = 3)
cladelabels(genomeTree, "SARS", 72, cex = 0.7, orientation = "horizontal", offset = 3)



```

