---
title: "Phylogenetic analysis of coronavirus sequences"
author: "Jacques van Helden"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    code_folding: hide
    fig_caption: yes
    highlight: zenburn
    self_contained: no
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  ioslides_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    smaller: yes
    toc: yes
    widescreen: yes
  beamer_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    incremental: no
    keep_tex: no
    slide_level: 2
    theme: Montpellier
    toc: yes
  word_document:
    toc: yes
    toc_depth: '3'
  slidy_presentation:
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    highlight: tango
    incremental: no
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
font-import: http://fonts.googleapis.com/css?family=Risque
font-family: Garamond
transition: linear
---

```{r libraries, echo=FALSE, results=FALSE, warning=FALSE, message=FALSE}
#### Install required packages ####
required.packages <- c("knitr", "ape", "phytools")

for (pkg in required.packages) {
  if (!require(pkg, character.only = TRUE)) {
    message("Installing package ", pkg)
    install.packages(pkg, dependencies = TRUE)
  }
  require(pkg, character.only = TRUE)
}


```



```{r knitr_settings, include=FALSE, echo=FALSE, eval=TRUE}
library(knitr)
options(width = 300)
knitr::opts_chunk$set(
  fig.width = 7, fig.height = 5, 
  fig.path = 'figures/phylogeny/CoV_',
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = TRUE, comment = "")
# knitr::asis_output("\\footnotesize")


## Store original graphic parameters to restore them after chunks
par.ori <- par(no.readonly = TRUE)


```

```{r directories}
dir <- vector()
dir["main"] <- ".."

dir["results"] <- file.path(dir["main"], "results")
dir["genomes"] <- file.path(dir["results"], "genome_phylogeny", "clustalw_alignments")



```

```{r parameters}

## Color palette per species
speciesPalette <- list(
  Human = "#880000",
  Bat = "#BBBBBB",
  Pangolin = "#00BB00",
  Camel = "#BB8800",
  Pig = "#FFBBBB",
  Civet = "#0000BB"
)

## Species prefix in the tip labels
speciesPrefix <- c("Hu" = "Human",
                   "Bt" = "Bat",
                   "Pn" = "Pangolin",
                   "Cm" = "Camel",
                   "Pi" = "Pig",
                   "Cv" = "Civet")
## Outgroup
outgroup <- c("HuOC43",
              "PiPRCV",
              "HuTGEV",
              "PiSADS",
              "Hu229E",
              "HuNL63")


```





## Phylogeny from full genomes

We inferred a phylogeny of virus strains based ontheir full genomes. 

A multiple alignment of genome sequences was performed with a progressive method (`clustalw`). The tree of virus strains was inferred with a maximum likelihood approach (`phyml` software).


```{r load_tree_function}
#### Define a function to load a tree ####

#' @title Load a tree and prepare if for the plot
#' @author Jacques van Helden
#' @param file file containing the tree (must  be  format supported by ape::read.tree)
#' @param outgroup=NULL out group used to define the root. Should be a vector of tip labels.
#' @param rootNode=NULL alternative to outgroup: define the root based on an internal node number
#' @param rotateNodes=NULL a vector of internal node numbers to rotate for the plot
loadTree <- function(file, 
                       outgroup = NULL,
                       rootNode = NULL,
                       rotateNodes = NULL
                       ) {
  
  
  ## Load the tree with phytools
  tree <- ape::read.tree(file = file)
  # tree <- phytools::read.newick(file = treeFile)
  # plot(tree)
  # nodelabels()
  
  ## Define the root
  if (!is.null(outgroup)) {
    tree <- root(tree, outgroup = outgroup, resolve.root = TRUE)
  } else if (!is.null(root)) {
    tree <- reroot(tree, node = rootNode)  
  }
  
  # Rotate user-specified nodes
  if (!is.null(rotateNodes)) {
    for (node in rotateNodes) {
      tree <- rotate(tree, node)
    }
    
  }
  #tree <- rotate(tree, 39)
  #tree <- rotate(tree, 44)
  # tree <- rotate(tree, 75)
  # nodelabels()
  
  if (is.binary(tree)) {
    treeType <- "binary"
  } else {
    treeType <- "non-binary"
  }
  message("\tLoaded ", treeType, " genome tree", 
          " with ", Ntip(tree), " tips",
          " and ", Nnode(tree), " internal nodes")
  
  ## Check theinternal structure of the tree
  # str(tree)
  # tree$node.label
  #  dim(tree$edge)
  #  tree$tip.label
  #  class(tree)
  
  ## Tip parametes
  tipParam <- data.frame(
    row.names = tree$tip.label,
    color = rep(x = "grey", length.out = length(tree$tip.label)))
  tipParam$species <- "NONE"
  
  
  ## Identify species per tip
  for (prefix in names(speciesPrefix)) {
    tipParam[grep(pattern = paste0("^", prefix), x = row.names(tipParam), perl = TRUE), "species"] <- speciesPrefix[prefix]
    
  }
  # table(tipParam$species)
  # View(tipParam)
  
  ## Assign tip color accoding to species
  tipParam$color <- unlist(speciesPalette[tipParam$species])
  # table(tipParam$color, tipParam$species)
  # table(tipParam$color)
  
  # ## Identify CoV2 tips
  # cov2Tips <- grep(pattern = "CoV2", x = tree$tip.label, perl = TRUE, value = TRUE)
  result <- list(tree = tree, 
                 tipParam = tipParam)
  return(result)
}

```


```{r plot_my_tree_function}
plotMyTree <- function(myTree, 
                       show.node.label = FALSE, ...) {
  ## Plot the tree
  ape::plot.phylo(myTree$tree, 
                  type = "phylogram", 
                  show.node.label = show.node.label,
                  show.tip.label = TRUE,
                  edge.color = "grey",
                  tip.color = myTree$tipParam$color,
                  #         ftype = "off",
                  edge.width = 2,
                  label.offset = 0.1, 
                  lwd = 2,
                  # fsize = 0.7,
                  cex = 0.7,
                  color = speciesPalette$Bat, 
                  nomargin = FALSE,
                  font = 1, 
                  ...)
  add.scale.bar(cex = 1, font = 2, col = "blue", length = 0.2, 
                lwd = 2, lcol = "blue")
  # add.arrow(myTree$tree, tip = cov2Tips, 
  #           arrl = 0.3, 
  #           col = speciesPalette$Human)
  
  # length(myTree$tree$node.label)
  # nodelabels(cex = 0.5)
  # sort(as.numeric(myTree$tree$node.label))
  # tiplabels()
  
}

```




```{r genome_tree, fig.width=12, fig.height=7,  out.width="100%", fig.cap="Genome tree of selected coronaviruses.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). "}

#### Load and plot the genome tree ####

genomeTreeFile <- file.path(
  dir["genomes"],
  "coronavirus_selected-plus-GISAID_genomes_clustalw_gblocks.phy_phyml_tree.phb")


genomeTree <- loadTree(
  file = genomeTreeFile, 
  outgroup = outgroup, 
  rootNode = NULL, 
  rotateNodes = c(39, 75, 42))
# genomeTree <- paintSubTree(tree = genomeTree, node = 49, state = "CoV2")
# nodelabels(cex = 0.4)

plotMyTree(genomeTree, main  = "Genome-based virus tree", show.node.label = TRUE)

## Identify some clades
cladelabels(genomeTree$tree, "CoV2", 46, cex = 0.7, orientation = "horizontal", offset = 5)
cladelabels(genomeTree$tree, "MERS", 43, cex = 0.7, orientation = "horizontal", offset = 5)
cladelabels(genomeTree$tree, "SARS", 69, cex = 0.7, orientation = "horizontal", offset = 5)

```



```{r S_gene_tree, fig.width=12, fig.height=7,  out.width="100%", fig.cap="Phylogenetic tree of the S gene for selected coronaviruses.  The tree was inferred by maximum likelihood apprroach (PhyML) based on a progressive multiple alignment (clustalw). "}

treeFiles <- vector()
trees <- vector()

#### Load and plot the genome tree ####
treeFiles["S gene"] <- file.path(
  dir["results"],
  "S-gene", "clustalw_alignments", 
  "S-gene_around-cov-2-plus-GISAID_gblocks.phy_phyml_boot_trees.phb")

trees["S gene"] <- loadTree(
  file = treeFiles["S gene"], 
  outgroup = NULL, 
  rootNode = NULL, 
  rotateNodes = NULL)
# genomeTree <- paintSubTree(tree = genomeTree, node = 49, state = "CoV2")
# nodelabels(cex = 0.4)

plotMyTree(genomeTree, main  = "Genome-based virus tree", show.node.label = FALSE)

## Identify some clades
cladelabels(genomeTree$tree, "CoV2", 46, cex = 0.7, orientation = "horizontal", offset = 5)
cladelabels(genomeTree$tree, "MERS", 43, cex = 0.7, orientation = "horizontal", offset = 5)
cladelabels(genomeTree$tree, "SARS", 69, cex = 0.7, orientation = "horizontal", offset = 5)

```


