<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-05-28" />

<title>Analysis of coronavirus sequences</title>

<script src="pip-profiles_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="pip-profiles_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="pip-profiles_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="pip-profiles_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="pip-profiles_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="pip-profiles_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="pip-profiles_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="pip-profiles_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="pip-profiles_files/navigation-1.1/tabsets.js"></script>
<script src="pip-profiles_files/navigation-1.1/codefolding.js"></script>
<link href="pip-profiles_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="pip-profiles_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of coronavirus sequences</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-05-28</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#### General parameters for the analysis ####</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">## Use (or not) GIDAID sequences</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">## </span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">## A few genoes were not available in NCBI Genbank at the time of </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">## this analysis, and had to be downloaded from GISAID. These sequences</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">##  can however not be redistributed, they should thus be downloaded </span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">##  manually to reproduce the full trees. Alternatively, useGISAID</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">##  can be set to FALSE, whcih will reproduce the analysis with almost </span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">##  all the sequences of the paper.</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11">useGISAID &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="co">#### Define directories and files ####</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14">dir &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">main =</span> <span class="st">&#39;..&#39;</span>)</a>
<a class="sourceLine" id="cb1-15" data-line-number="15">dir<span class="op">$</span>R &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;scripts/R&quot;</span>)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16"></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co">#### Create output directory for sequences ####</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18">dir<span class="op">$</span>outseq &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb1-19" data-line-number="19">  dir<span class="op">$</span>main, <span class="st">&quot;results&quot;</span>, <span class="st">&quot;S-gene&quot;</span>, <span class="st">&quot;Nto1_alignments&quot;</span>)</a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="kw">dir.create</span>(dir<span class="op">$</span>outseq, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-21" data-line-number="21"></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co">## Instantiate a list for output files</span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23">outfiles &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb1-24" data-line-number="24"></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="co">## Input files</span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26">infiles &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-27" data-line-number="27"></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">## Sequence collection</span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29"><span class="co">## Supported: </span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co">## collection &lt;- &quot;around-CoV-2&quot; # ~20 genomes</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="co">## collection &lt;- &quot;selected&quot; # ~60 genomes</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="co">## collection &lt;- &quot;all&quot; # ~60 genomes</span></a>
<a class="sourceLine" id="cb1-33" data-line-number="33">collections &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;around-CoV-2&quot;</span>, <span class="st">&quot;selected&quot;</span>, <span class="st">&quot;all&quot;</span>)</a>
<a class="sourceLine" id="cb1-34" data-line-number="34">collection &lt;-<span class="st"> &quot;around-CoV-2&quot;</span> <span class="co"># ~20 genomes</span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"></a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="co">## Genome dir and files</span></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="cf">if</span> (useGISAID) {</a>
<a class="sourceLine" id="cb1-39" data-line-number="39">  dir<span class="op">$</span>genomes &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;GISAID_genomes&quot;</span>)</a>
<a class="sourceLine" id="cb1-40" data-line-number="40">  infiles<span class="op">$</span>genomes &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb1-41" data-line-number="41">    dir<span class="op">$</span>genomes, </a>
<a class="sourceLine" id="cb1-42" data-line-number="42">    <span class="kw">paste0</span>(<span class="st">&quot;genomes_&quot;</span>, collection, <span class="st">&quot;-plus-GISAID.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb1-43" data-line-number="43">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb1-44" data-line-number="44">  dir<span class="op">$</span>genomes &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;virus_genomes&quot;</span>)</a>
<a class="sourceLine" id="cb1-45" data-line-number="45">  infiles<span class="op">$</span>genomes &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb1-46" data-line-number="46">    dir<span class="op">$</span>genomes, </a>
<a class="sourceLine" id="cb1-47" data-line-number="47">    <span class="kw">paste0</span>(<span class="st">&quot;genomes_&quot;</span>, collection,<span class="st">&quot;.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb1-48" data-line-number="48">}</a>
<a class="sourceLine" id="cb1-49" data-line-number="49"></a>
<a class="sourceLine" id="cb1-50" data-line-number="50"><span class="co">## Genome sequences</span></a>
<a class="sourceLine" id="cb1-51" data-line-number="51"><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(infiles<span class="op">$</span>genomes)) {</a>
<a class="sourceLine" id="cb1-52" data-line-number="52">  <span class="kw">stop</span>(<span class="st">&quot;Genome sequence file is missing&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, infiles<span class="op">$</span>genomes)</a>
<a class="sourceLine" id="cb1-53" data-line-number="53">}</a>
<a class="sourceLine" id="cb1-54" data-line-number="54"></a>
<a class="sourceLine" id="cb1-55" data-line-number="55"></a>
<a class="sourceLine" id="cb1-56" data-line-number="56"><span class="co">## Output tables</span></a>
<a class="sourceLine" id="cb1-57" data-line-number="57"><span class="co"># di$output &lt;- file.path(dir.main, &quot;&quot;)</span></a>
<a class="sourceLine" id="cb1-58" data-line-number="58"><span class="co"># dir$tables &lt;- </span></a>
<a class="sourceLine" id="cb1-59" data-line-number="59"></a>
<a class="sourceLine" id="cb1-60" data-line-number="60"><span class="co">## Load custom functions</span></a>
<a class="sourceLine" id="cb1-61" data-line-number="61"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;align_n_to_one.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-62" data-line-number="62"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;plot_pip_profiles.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-63" data-line-number="63"></a>
<a class="sourceLine" id="cb1-64" data-line-number="64"><span class="co">## Reference genome</span></a>
<a class="sourceLine" id="cb1-65" data-line-number="65">refPattern &lt;-<span class="st"> &quot;HuCoV2_WH01_2019&quot;</span></a>
<a class="sourceLine" id="cb1-66" data-line-number="66"></a>
<a class="sourceLine" id="cb1-67" data-line-number="67"></a>
<a class="sourceLine" id="cb1-68" data-line-number="68"><span class="co">## Features : coordinates of features of interest in the reference genome</span></a>
<a class="sourceLine" id="cb1-69" data-line-number="69">features &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-70" data-line-number="70"><span class="co">## Spike gene  (coding for the spike protein)</span></a>
<a class="sourceLine" id="cb1-71" data-line-number="71">features[[<span class="st">&#39;CDS-ORF1ab&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">266</span>, <span class="dt">end =</span> <span class="dv">21555</span>)</a>
<a class="sourceLine" id="cb1-72" data-line-number="72">features[[<span class="st">&#39;CDS-S&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">21563</span>, <span class="dt">end =</span> <span class="dv">25384</span>)</a>
<a class="sourceLine" id="cb1-73" data-line-number="73">features[[<span class="st">&#39;CDS-ORF3a&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">25393</span>, <span class="dt">end =</span> <span class="dv">26220</span>)</a>
<a class="sourceLine" id="cb1-74" data-line-number="74">features[[<span class="st">&#39;CDS-E&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">26245</span>, <span class="dt">end =</span> <span class="dv">26472</span>)</a>
<a class="sourceLine" id="cb1-75" data-line-number="75">features[[<span class="st">&#39;CDS-M&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">26523</span>, <span class="dt">end =</span> <span class="dv">27191</span>)</a>
<a class="sourceLine" id="cb1-76" data-line-number="76">features[[<span class="st">&#39;CDS-ORF6&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27202</span>, <span class="dt">end =</span> <span class="dv">27387</span>)</a>
<a class="sourceLine" id="cb1-77" data-line-number="77">features[[<span class="st">&#39;CDS-ORF7a&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27394</span>, <span class="dt">end =</span> <span class="dv">27759</span>)</a>
<a class="sourceLine" id="cb1-78" data-line-number="78">features[[<span class="st">&#39;CDS-ORF8&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">27894</span>, <span class="dt">end =</span> <span class="dv">28259</span>)</a>
<a class="sourceLine" id="cb1-79" data-line-number="79">features[[<span class="st">&#39;CDS-N&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">28274</span>, <span class="dt">end =</span> <span class="dv">29533</span>)</a>
<a class="sourceLine" id="cb1-80" data-line-number="80">features[[<span class="st">&#39;CDS-ORF10&#39;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">start =</span> <span class="dv">29558</span>, <span class="dt">end =</span> <span class="dv">29674</span>)                             </a>
<a class="sourceLine" id="cb1-81" data-line-number="81"></a>
<a class="sourceLine" id="cb1-82" data-line-number="82"><span class="co">## Query genomes</span></a>
<a class="sourceLine" id="cb1-83" data-line-number="83">queryPatterns &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-84" data-line-number="84">  <span class="st">&quot;BtRaTG13_2013_Yunnan&quot;</span>, </a>
<a class="sourceLine" id="cb1-85" data-line-number="85">  <span class="st">&quot;BtZC45&quot;</span>,</a>
<a class="sourceLine" id="cb1-86" data-line-number="86">  <span class="st">&quot;BtZXC21&quot;</span>,</a>
<a class="sourceLine" id="cb1-87" data-line-number="87">  <span class="st">&quot;PnMP789&quot;</span>,</a>
<a class="sourceLine" id="cb1-88" data-line-number="88">  <span class="st">&quot;PnGX-P1E_2017&quot;</span>, </a>
<a class="sourceLine" id="cb1-89" data-line-number="89">  <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span></a>
<a class="sourceLine" id="cb1-90" data-line-number="90">)</a>
<a class="sourceLine" id="cb1-91" data-line-number="91"></a>
<a class="sourceLine" id="cb1-92" data-line-number="92"><span class="co">#### Add GISAID IDs to the query pattern ####</span></a>
<a class="sourceLine" id="cb1-93" data-line-number="93"><span class="co">## Note that GISAI genomes are be submitted to the github repo because they cannot be redistributed</span></a>
<a class="sourceLine" id="cb1-94" data-line-number="94"><span class="cf">if</span> (useGISAID) {</a>
<a class="sourceLine" id="cb1-95" data-line-number="95">  queryPatterns &lt;-<span class="st"> </span><span class="kw">append</span>(queryPatterns, </a>
<a class="sourceLine" id="cb1-96" data-line-number="96">                          <span class="kw">c</span>(<span class="st">&quot;BtYu-RmYN02_2019&quot;</span>,</a>
<a class="sourceLine" id="cb1-97" data-line-number="97">                            <span class="st">&quot;PnGu1_2019&quot;</span></a>
<a class="sourceLine" id="cb1-98" data-line-number="98">                          ))</a>
<a class="sourceLine" id="cb1-99" data-line-number="99">}</a>
<a class="sourceLine" id="cb1-100" data-line-number="100"></a>
<a class="sourceLine" id="cb1-101" data-line-number="101"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Reference genomes: &quot;</span>, refPattern)</a>
<a class="sourceLine" id="cb1-102" data-line-number="102"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Number of query patterns: &quot;</span>, <span class="kw">length</span>(queryPatterns))</a>
<a class="sourceLine" id="cb1-103" data-line-number="103"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Query patterns: &quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, queryPatterns))</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">#### Load genome sequences ####</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">genomes &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="dt">filepath =</span> infiles<span class="op">$</span>genome, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">## Shorten sequence names by suppressing the fasta comment (after the space)</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw">names</span>(genomes) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">&quot; .*&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="dt">x =</span> <span class="kw">names</span>(genomes), <span class="dt">perl =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">genomeNames &lt;-<span class="st"> </span><span class="kw">names</span>(genomes)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">nbGenomes &lt;-<span class="st"> </span><span class="kw">length</span>(genomeNames)</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">message</span>(<span class="st">&quot;Loaded &quot;</span>, nbGenomes, <span class="st">&quot; genomes from file &quot;</span>, infiles<span class="op">$</span>genomes)</a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co"># View(genomes)</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#### Define reference and query genomes ####</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">refGenomeName &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> refPattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </a>
<a class="sourceLine" id="cb2-14" data-line-number="14">                      <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="cf">if</span> (<span class="kw">is.null</span>(refGenomeName)) {</a>
<a class="sourceLine" id="cb2-16" data-line-number="16">  <span class="kw">stop</span>(<span class="st">&quot;Could not identify reference genome with pattern &quot;</span>, refPattern)</a>
<a class="sourceLine" id="cb2-17" data-line-number="17">}</a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="kw">message</span>(<span class="st">&quot;Reference genome name: &quot;</span>, refGenomeName)</a>
<a class="sourceLine" id="cb2-19" data-line-number="19"></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">## Query genomes</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21">queryRegExp &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;(&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;)|(&quot;</span>, queryPatterns), <span class="st">&quot;)&quot;</span>)</a>
<a class="sourceLine" id="cb2-22" data-line-number="22">queryGenomeNames &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> queryRegExp, </a>
<a class="sourceLine" id="cb2-23" data-line-number="23">                         <span class="dt">x =</span> genomeNames, </a>
<a class="sourceLine" id="cb2-24" data-line-number="24">                         <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-25" data-line-number="25">nbQueryGenomes &lt;-<span class="st"> </span><span class="kw">length</span>(queryGenomeNames)</a>
<a class="sourceLine" id="cb2-26" data-line-number="26"></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="cf">if</span> (nbQueryGenomes <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) {</a>
<a class="sourceLine" id="cb2-28" data-line-number="28">  <span class="kw">stop</span>(<span class="st">&quot;Could not identify any query genome with query pattern</span><span class="ch">\n</span><span class="st">&quot;</span>, queryRegExp)</a>
<a class="sourceLine" id="cb2-29" data-line-number="29">}</a>
<a class="sourceLine" id="cb2-30" data-line-number="30"></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="cf">if</span> (<span class="kw">length</span>(queryPatterns) <span class="op">!=</span><span class="st"> </span><span class="kw">length</span>(queryGenomeNames)) {</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">  foundPatterns &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> queryRegExp, <span class="dt">x =</span> queryGenomeNames, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-33" data-line-number="33">  missingPatterns &lt;-<span class="st"> </span><span class="kw">setdiff</span>(queryPatterns, queryGenomeNames)</a>
<a class="sourceLine" id="cb2-34" data-line-number="34">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </a>
<a class="sourceLine" id="cb2-35" data-line-number="35">          <span class="kw">length</span>(missingPatterns), <span class="st">&quot; Missing genomes: &quot;</span>, </a>
<a class="sourceLine" id="cb2-36" data-line-number="36">          <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, missingPatterns))</a>
<a class="sourceLine" id="cb2-37" data-line-number="37">}</a>
<a class="sourceLine" id="cb2-38" data-line-number="38"></a>
<a class="sourceLine" id="cb2-39" data-line-number="39"></a>
<a class="sourceLine" id="cb2-40" data-line-number="40"><span class="co">## Compute some statistics about genome sizes</span></a>
<a class="sourceLine" id="cb2-41" data-line-number="41">genomeStat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb2-42" data-line-number="42">  <span class="dt">row.names =</span> <span class="kw">c</span>(refGenomeName, queryGenomeNames),</a>
<a class="sourceLine" id="cb2-43" data-line-number="43">  <span class="dt">status =</span> <span class="kw">c</span>(<span class="st">&quot;Reference&quot;</span>, <span class="kw">rep</span>(<span class="st">&quot;Query&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(queryGenomeNames)))</a>
<a class="sourceLine" id="cb2-44" data-line-number="44">)</a>
<a class="sourceLine" id="cb2-45" data-line-number="45"></a>
<a class="sourceLine" id="cb2-46" data-line-number="46">g &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb2-47" data-line-number="47"><span class="cf">for</span> (g <span class="cf">in</span> <span class="kw">c</span>(refGenomeName, queryGenomeNames)) {</a>
<a class="sourceLine" id="cb2-48" data-line-number="48">  genomeStat[g, <span class="st">&quot;length&quot;</span>] &lt;-<span class="st"> </span><span class="kw">length</span>(genomes[[g]])</a>
<a class="sourceLine" id="cb2-49" data-line-number="49">}</a>
<a class="sourceLine" id="cb2-50" data-line-number="50"></a>
<a class="sourceLine" id="cb2-51" data-line-number="51"><span class="kw">kable</span>(genomeStat, <span class="dt">caption =</span> <span class="st">&quot;Reference and query genomes&quot;</span>)</a></code></pre></div>
<table>
<caption>Reference and query genomes</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">status</th>
<th align="right">length</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HuCoV2_WH01_2019</td>
<td align="left">Reference</td>
<td align="right">29899</td>
</tr>
<tr class="even">
<td>BtRaTG13_2013_Yunnan</td>
<td align="left">Query</td>
<td align="right">29855</td>
</tr>
<tr class="odd">
<td>BtZC45</td>
<td align="left">Query</td>
<td align="right">29802</td>
</tr>
<tr class="even">
<td>BtZXC21</td>
<td align="left">Query</td>
<td align="right">29732</td>
</tr>
<tr class="odd">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="left">Query</td>
<td align="right">29727</td>
</tr>
<tr class="even">
<td>PnGX-P1E_2017</td>
<td align="left">Query</td>
<td align="right">29801</td>
</tr>
<tr class="odd">
<td>PnMP789</td>
<td align="left">Query</td>
<td align="right">29521</td>
</tr>
<tr class="even">
<td>BtYu-RmYN02_2019</td>
<td align="left">Query</td>
<td align="right">29671</td>
</tr>
<tr class="odd">
<td>PnGu1_2019</td>
<td align="left">Query</td>
<td align="right">29825</td>
</tr>
</tbody>
</table>
<p>The collection around-CoV-2 contains 18 virus genome sequences.</p>
<div id="strain-statistics" class="section level2">
<h2>Strain statistics</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">## Report the number of genoomes</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">strainNames &lt;-<span class="st"> </span><span class="kw">names</span>(genomes)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">nbStrains &lt;-<span class="st"> </span><span class="kw">length</span>(strainNames)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Loaded &quot;</span>, nbStrains, <span class="st">&quot; genomes from file &quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co"># View(genomes)</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#### Define reference and query genomes ####</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9">refStrainName &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> refPattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </a>
<a class="sourceLine" id="cb3-10" data-line-number="10">                      <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="cf">if</span> (<span class="kw">is.null</span>(refStrainName)) {</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">  <span class="kw">stop</span>(<span class="st">&quot;Could not identify reference genome with pattern &quot;</span>, refPattern)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13">}</a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Reference genome name: &quot;</span>, refStrainName)</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#### Compute statistics about sequence sizes </span><span class="al">###</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17">strainStats &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">  <span class="dt">n =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genomes),</a>
<a class="sourceLine" id="cb3-19" data-line-number="19">  <span class="dt">row.names =</span> <span class="kw">names</span>(genomes),</a>
<a class="sourceLine" id="cb3-20" data-line-number="20">  <span class="dt">status =</span> <span class="kw">rep</span>(<span class="st">&quot;Query&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(strainNames))</a>
<a class="sourceLine" id="cb3-21" data-line-number="21">)</a>
<a class="sourceLine" id="cb3-22" data-line-number="22">strainStats[,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> </span><span class="kw">as.vector</span>(strainStats[,<span class="st">&quot;status&quot;</span>])</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">strainStats[refStrainName,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> &quot;Reference&quot;</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24">g &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="cf">for</span> (g <span class="cf">in</span> strainNames) {</a>
<a class="sourceLine" id="cb3-26" data-line-number="26">  strainStats[g, <span class="st">&quot;length&quot;</span>] &lt;-<span class="st"> </span><span class="kw">length</span>(genomes[[g]])</a>
<a class="sourceLine" id="cb3-27" data-line-number="27">}</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">#### Define the color associated to each sequence ####</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">## Color palette per species</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5">speciesPalette &lt;-<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">  <span class="dt">Human =</span> <span class="st">&quot;#880000&quot;</span>,</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">  <span class="dt">Bat =</span> <span class="st">&quot;#888888&quot;</span>,</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">  <span class="dt">Pangolin =</span> <span class="st">&quot;#448800&quot;</span>,</a>
<a class="sourceLine" id="cb4-9" data-line-number="9">  <span class="dt">Camel =</span> <span class="st">&quot;#BB8800&quot;</span>,</a>
<a class="sourceLine" id="cb4-10" data-line-number="10">  <span class="dt">Pig =</span> <span class="st">&quot;#FFBBBB&quot;</span>,</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">  <span class="dt">Civet =</span> <span class="st">&quot;#00BBFF&quot;</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12">)</a>
<a class="sourceLine" id="cb4-13" data-line-number="13"></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">## Species prefix in the tip labels</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15">speciesPrefix &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Hu&quot;</span> =<span class="st"> &quot;Human&quot;</span>,</a>
<a class="sourceLine" id="cb4-16" data-line-number="16">                   <span class="st">&quot;Bt&quot;</span> =<span class="st"> &quot;Bat&quot;</span>,</a>
<a class="sourceLine" id="cb4-17" data-line-number="17">                   <span class="st">&quot;Pn&quot;</span> =<span class="st"> &quot;Pangolin&quot;</span>,</a>
<a class="sourceLine" id="cb4-18" data-line-number="18">                   <span class="st">&quot;Cm&quot;</span> =<span class="st"> &quot;Camel&quot;</span>,</a>
<a class="sourceLine" id="cb4-19" data-line-number="19">                   <span class="st">&quot;Pi&quot;</span> =<span class="st"> &quot;Pig&quot;</span>,</a>
<a class="sourceLine" id="cb4-20" data-line-number="20">                   <span class="st">&quot;Cv&quot;</span> =<span class="st"> &quot;Civet&quot;</span>)</a>
<a class="sourceLine" id="cb4-21" data-line-number="21"></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">## Strain-specific colors</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23">strainColor &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb4-24" data-line-number="24">  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span> =<span class="st"> &quot;red&quot;</span>,</a>
<a class="sourceLine" id="cb4-25" data-line-number="25">  <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span> =<span class="st"> &quot;#0044BB&quot;</span>,</a>
<a class="sourceLine" id="cb4-26" data-line-number="26">  <span class="st">&quot;PnGu1_2019&quot;</span> =<span class="st"> &quot;#00BB00&quot;</span>,</a>
<a class="sourceLine" id="cb4-27" data-line-number="27">  <span class="st">&quot;PnMP789&quot;</span> =<span class="st"> &quot;#00FF88&quot;</span>,</a>
<a class="sourceLine" id="cb4-28" data-line-number="28">  <span class="st">&quot;BtRaTG13_&quot;</span> =<span class="st"> &quot;#FF6600&quot;</span>,</a>
<a class="sourceLine" id="cb4-29" data-line-number="29">  <span class="st">&quot;BtYu-RmYN&quot;</span> =<span class="st"> &quot;#FFBB22&quot;</span>,</a>
<a class="sourceLine" id="cb4-30" data-line-number="30">  <span class="st">&quot;BtZXC21&quot;</span> =<span class="st"> &quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb4-31" data-line-number="31">  <span class="st">&quot;BtZC45&quot;</span> =<span class="st"> &quot;black&quot;</span>)</a>
<a class="sourceLine" id="cb4-32" data-line-number="32"></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">## Identify species per tip</span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="cf">for</span> (prefix <span class="cf">in</span> <span class="kw">names</span>(speciesPrefix)) {</a>
<a class="sourceLine" id="cb4-35" data-line-number="35">  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, prefix), <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;species&quot;</span>] &lt;-<span class="st"> </span>speciesPrefix[prefix]</a>
<a class="sourceLine" id="cb4-36" data-line-number="36">  </a>
<a class="sourceLine" id="cb4-37" data-line-number="37">}</a>
<a class="sourceLine" id="cb4-38" data-line-number="38"></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="co">## Assign acolor to each species</span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40">strainStats<span class="op">$</span>color &lt;-<span class="st"> &quot;grey&quot;</span> <span class="co"># default</span></a>
<a class="sourceLine" id="cb4-41" data-line-number="41">strainStats<span class="op">$</span>color &lt;-<span class="st"> </span>speciesPalette[<span class="kw">as.vector</span>(strainStats<span class="op">$</span>species)]</a>
<a class="sourceLine" id="cb4-42" data-line-number="42"></a>
<a class="sourceLine" id="cb4-43" data-line-number="43"><span class="cf">for</span> (strain <span class="cf">in</span> <span class="kw">names</span>(strainColor)) {</a>
<a class="sourceLine" id="cb4-44" data-line-number="44">  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, strain), </a>
<a class="sourceLine" id="cb4-45" data-line-number="45">                  <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;color&quot;</span>] &lt;-<span class="st"> </span>strainColor[strain]</a>
<a class="sourceLine" id="cb4-46" data-line-number="46">  </a>
<a class="sourceLine" id="cb4-47" data-line-number="47">}</a>
<a class="sourceLine" id="cb4-48" data-line-number="48"></a>
<a class="sourceLine" id="cb4-49" data-line-number="49"></a>
<a class="sourceLine" id="cb4-50" data-line-number="50"><span class="co">## Assign specific color to some nodes</span></a>
<a class="sourceLine" id="cb4-51" data-line-number="51"></a>
<a class="sourceLine" id="cb4-52" data-line-number="52"><span class="co">## Define a color for each strain</span></a>
<a class="sourceLine" id="cb4-53" data-line-number="53">strainColors &lt;-<span class="st"> </span>(<span class="kw">unlist</span>(strainStats<span class="op">$</span>color))</a>
<a class="sourceLine" id="cb4-54" data-line-number="54"><span class="kw">names</span>(strainColors) &lt;-<span class="st"> </span><span class="kw">row.names</span>(strainStats)</a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">kable</span>(strainStats, <span class="dt">caption =</span> <span class="st">&quot;Reference and query genomes&quot;</span>)</a></code></pre></div>
<table>
<caption>Reference and query genomes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">n</th>
<th align="left">status</th>
<th align="right">length</th>
<th align="left">species</th>
<th align="left">color</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BtBM48-31</td>
<td align="right">1</td>
<td align="left">Query</td>
<td align="right">29276</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtGX2013</td>
<td align="right">2</td>
<td align="left">Query</td>
<td align="right">29161</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="odd">
<td>BtHKU3-12</td>
<td align="right">3</td>
<td align="left">Query</td>
<td align="right">29704</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">4</td>
<td align="left">Query</td>
<td align="right">29855</td>
<td align="left">Bat</td>
<td align="left">#FF6600</td>
</tr>
<tr class="odd">
<td>BtRs4874</td>
<td align="right">5</td>
<td align="left">Query</td>
<td align="right">30311</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtYN2013</td>
<td align="right">6</td>
<td align="left">Query</td>
<td align="right">29142</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="odd">
<td>BtYN2018B</td>
<td align="right">7</td>
<td align="left">Query</td>
<td align="right">30256</td>
<td align="left">Bat</td>
<td align="left">#888888</td>
</tr>
<tr class="even">
<td>BtZC45</td>
<td align="right">8</td>
<td align="left">Query</td>
<td align="right">29802</td>
<td align="left">Bat</td>
<td align="left">black</td>
</tr>
<tr class="odd">
<td>BtZXC21</td>
<td align="right">9</td>
<td align="left">Query</td>
<td align="right">29732</td>
<td align="left">Bat</td>
<td align="left">black</td>
</tr>
<tr class="even">
<td>Cv007-2004</td>
<td align="right">10</td>
<td align="left">Query</td>
<td align="right">29540</td>
<td align="left">Civet</td>
<td align="left">#00BBFF</td>
</tr>
<tr class="odd">
<td>HuCoV2_WH01_2019</td>
<td align="right">11</td>
<td align="left">Reference</td>
<td align="right">29899</td>
<td align="left">Human</td>
<td align="left">red</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="right">12</td>
<td align="left">Query</td>
<td align="right">29727</td>
<td align="left">Human</td>
<td align="left">#0044BB</td>
</tr>
<tr class="odd">
<td>PnGX-P1E_2017</td>
<td align="right">13</td>
<td align="left">Query</td>
<td align="right">29801</td>
<td align="left">Pangolin</td>
<td align="left">#448800</td>
</tr>
<tr class="even">
<td>PnGX-P2V_2018</td>
<td align="right">14</td>
<td align="left">Query</td>
<td align="right">29795</td>
<td align="left">Pangolin</td>
<td align="left">#448800</td>
</tr>
<tr class="odd">
<td>PnMP789</td>
<td align="right">15</td>
<td align="left">Query</td>
<td align="right">29521</td>
<td align="left">Pangolin</td>
<td align="left">#00FF88</td>
</tr>
<tr class="even">
<td>BtYu-RmYN02_2019</td>
<td align="right">16</td>
<td align="left">Query</td>
<td align="right">29671</td>
<td align="left">Bat</td>
<td align="left">#FFBB22</td>
</tr>
<tr class="odd">
<td>PnGu1_2019</td>
<td align="right">17</td>
<td align="left">Query</td>
<td align="right">29825</td>
<td align="left">Pangolin</td>
<td align="left">#00BB00</td>
</tr>
<tr class="even">
<td>PnGu-P2S_2019</td>
<td align="right">18</td>
<td align="left">Query</td>
<td align="right">29769</td>
<td align="left">Pangolin</td>
<td align="left">#448800</td>
</tr>
</tbody>
</table>
</div>
<div id="n-to-1-full-genome-alignments" class="section level2">
<h2>N-to-1 full genome alignments</h2>
<p>We perform a pairwise lignment between each genome query and the reference genome (HuCoV2_WH01_2019).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co">#### N-to-1 genome alignments ####</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">## Define output file for genome alignments</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4">outfiles[<span class="st">&quot;Genome alignments&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">  dir<span class="op">$</span>outseq, <span class="kw">paste0</span>(<span class="st">&quot;genome_alignments_ref_&quot;</span>, refGenomeName))</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">## Get sequences for reference and query genomes</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8">refGenome &lt;-<span class="st"> </span>genomes[refGenomeName]</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">queryGenomes &lt;-<span class="st"> </span>genomes[queryGenomeNames]</a>
<a class="sourceLine" id="cb6-10" data-line-number="10">genomesNto1 &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb6-11" data-line-number="11">  <span class="dt">refSequence =</span> refGenome, </a>
<a class="sourceLine" id="cb6-12" data-line-number="12">  <span class="dt">querySequences =</span> queryGenomes,</a>
<a class="sourceLine" id="cb6-13" data-line-number="13">  <span class="dt">outfile =</span> outfiles[[<span class="st">&quot;Genome alignments&quot;</span>]] )</a>
<a class="sourceLine" id="cb6-14" data-line-number="14"></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="kw">kable</span>(genomesNto1<span class="op">$</span>stats[<span class="kw">order</span>(genomesNto1<span class="op">$</span>stats<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), ], </a>
<a class="sourceLine" id="cb6-16" data-line-number="16">      <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of full genomes&quot;</span>)</a></code></pre></div>
<table>
<caption>N-to-one alignment of full genomes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">96.07482</td>
<td align="right">29884</td>
<td align="right">6</td>
<td align="right">29</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">49895.67</td>
</tr>
<tr class="even">
<td>BtYu-RmYN02_2019</td>
<td align="right">93.70042</td>
<td align="right">29875</td>
<td align="right">59</td>
<td align="right">204</td>
<td align="right">22</td>
<td align="right">46</td>
<td align="right">43735.11</td>
</tr>
<tr class="odd">
<td>PnGu1_2019</td>
<td align="right">90.25891</td>
<td align="right">29894</td>
<td align="right">16</td>
<td align="right">69</td>
<td align="right">9</td>
<td align="right">23</td>
<td align="right">36100.06</td>
</tr>
<tr class="even">
<td>PnMP789</td>
<td align="right">90.00945</td>
<td align="right">29638</td>
<td align="right">17</td>
<td align="right">117</td>
<td align="right">9</td>
<td align="right">23</td>
<td align="right">34255.32</td>
</tr>
<tr class="odd">
<td>BtZC45</td>
<td align="right">87.87383</td>
<td align="right">29927</td>
<td align="right">33</td>
<td align="right">127</td>
<td align="right">16</td>
<td align="right">28</td>
<td align="right">30517.91</td>
</tr>
<tr class="even">
<td>BtZXC21</td>
<td align="right">87.62115</td>
<td align="right">29922</td>
<td align="right">37</td>
<td align="right">192</td>
<td align="right">13</td>
<td align="right">23</td>
<td align="right">30020.70</td>
</tr>
<tr class="odd">
<td>PnGX-P1E_2017</td>
<td align="right">85.39630</td>
<td align="right">29876</td>
<td align="right">27</td>
<td align="right">77</td>
<td align="right">13</td>
<td align="right">19</td>
<td align="right">24433.32</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="right">79.53068</td>
<td align="right">30001</td>
<td align="right">76</td>
<td align="right">276</td>
<td align="right">63</td>
<td align="right">131</td>
<td align="right">10310.73</td>
</tr>
</tbody>
</table>
</div>
<div id="full-genome-pip-plot" class="section level2">
<h2>Full genome PIP plot</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co">## PIP profile of full genome N-to-1 alignments</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> genomesNto1<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb7-3" data-line-number="3">                <span class="dt">windowSize =</span> <span class="dv">800</span>, </a>
<a class="sourceLine" id="cb7-4" data-line-number="4">                <span class="dt">vgrid1 =</span> <span class="dv">5000</span>, </a>
<a class="sourceLine" id="cb7-5" data-line-number="5">                <span class="dt">vgrid2 =</span> <span class="dv">1000</span>, <span class="dt">colors =</span> strainColors,</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">                <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Full genome PIP profile&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, refGenomeName),</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">                <span class="dt">legendMargin =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb7-8" data-line-number="8">                <span class="dt">legendCorner =</span> <span class="st">&quot;bottom&quot;</span>, </a>
<a class="sourceLine" id="cb7-9" data-line-number="9">                <span class="dt">legendCex =</span> <span class="fl">0.8</span>, </a>
<a class="sourceLine" id="cb7-10" data-line-number="10">                <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">40</span>,<span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/pip_profiles/RNA_genome_pip-1.png" alt="Percent Identical Positions profile over the whole genome of SARS-CoV-2. " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole genome of SARS-CoV-2.
</p>
</div>
</div>
<div id="sars-vs-civet" class="section level2">
<h2>SARS vs Civet</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co">## Define output file for genome alignments</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">outfiles[<span class="st">&quot;Genome alignments - SARS&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">  dir<span class="op">$</span>outseq, <span class="kw">paste0</span>(<span class="st">&quot;genome_alignments_ref_&quot;</span>,</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">                     <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>))</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#### Compare SARS (2002) wih the related Civet genome ####</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7">SARSvsCivetGenome &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb8-8" data-line-number="8">  <span class="dt">refSequence =</span> genomes[<span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>], </a>
<a class="sourceLine" id="cb8-9" data-line-number="9">  <span class="dt">querySequences =</span> genomes[<span class="kw">c</span>(<span class="st">&quot;Cv007-2004&quot;</span>, </a>
<a class="sourceLine" id="cb8-10" data-line-number="10">                             <span class="st">&quot;HuCoV2_WH01_2019&quot;</span>)],</a>
<a class="sourceLine" id="cb8-11" data-line-number="11">  <span class="dt">outfile =</span> outfiles[<span class="st">&quot;Genome alignments - SARS&quot;</span>] )</a>
<a class="sourceLine" id="cb8-12" data-line-number="12"></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="kw">kable</span>(SARSvsCivetGenome<span class="op">$</span>stats[<span class="kw">order</span>(SARSvsCivetGenome<span class="op">$</span>stats<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), ], </a>
<a class="sourceLine" id="cb8-14" data-line-number="14">      <span class="dt">caption =</span> <span class="st">&quot;SARS (2003). N-to-one alignment of full genomes of the closest animal virus (Civet) and of Human SARS-CoV-2. &quot;</span>)</a></code></pre></div>
<table>
<caption>SARS (2003). N-to-one alignment of full genomes of the closest animal virus (Civet) and of Human SARS-CoV-2.</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Cv007-2004</td>
<td align="right">99.51634</td>
<td align="right">29566</td>
<td align="right">1</td>
<td align="right">26</td>
<td align="right">1</td>
<td align="right">29</td>
<td align="right">56768.85</td>
</tr>
<tr class="even">
<td>HuCoV2_WH01_2019</td>
<td align="right">79.52735</td>
<td align="right">30001</td>
<td align="right">63</td>
<td align="right">131</td>
<td align="right">75</td>
<td align="right">274</td>
<td align="right">10434.99</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co">## PIP profile of full genome N-to-1 alignments</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> SARSvsCivetGenome<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb9-3" data-line-number="3">                <span class="dt">windowSize =</span> <span class="dv">500</span>, <span class="dt">colors =</span> strainColors, </a>
<a class="sourceLine" id="cb9-4" data-line-number="4">                <span class="dt">legend =</span> <span class="kw">paste0</span>(<span class="kw">names</span>(SARSvsCivetGenome<span class="op">$</span>alignments), <span class="st">&quot; (&quot;</span>, <span class="kw">round</span>(<span class="dt">digits =</span> <span class="dv">2</span>, SARSvsCivetGenome<span class="op">$</span>stats<span class="op">$</span>pid), <span class="st">&quot;%)&quot;</span>),</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">                <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Percent Identical Positions - Full genome&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, <span class="st">&quot;Human_SARS-CoV_Frankfurt_1&quot;</span>),</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">                <span class="dt">legendMargin =</span> <span class="dv">0</span>, <span class="dt">legendCorner =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legendCex =</span> <span class="fl">0.7</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">40</span>,<span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/pip_profiles/RNA_PIP_SARS_vs_civet_genome-1.png" alt="Percent Identical Positions profile over the whole genome of SARS (2002-2003). " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole genome of SARS (2002-2003).
</p>
</div>
</div>
<div id="n-to-1-alignemnts-of-spike-genes" class="section level2">
<h2>N-to-1 alignemnts of spike genes</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">featureName &lt;-<span class="st"> &quot;CDS-S&quot;</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">featureLimits &lt;-<span class="st"> </span>features[[featureName]]</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#### N-to-1 alignments of spike-coding sequences ####</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6">dir[[featureName]] &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;results&quot;</span>, featureName)</a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="kw">dir.create</span>(dir[[featureName]], <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">outfiles[featureName]  &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb10-9" data-line-number="9">  dir[[featureName]], <span class="kw">paste0</span>(featureName, <span class="st">&quot;_&quot;</span>, collection, <span class="st">&quot;_matches.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb10-10" data-line-number="10">  </a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">## Get sequences for reference and query genomes</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12">refSeq &lt;-<span class="st"> </span><span class="kw">subseq</span>(genomes[refGenomeName], <span class="dt">start =</span> featureLimits[<span class="dv">1</span>], <span class="dt">end =</span> featureLimits[<span class="dv">2</span>])</a>
<a class="sourceLine" id="cb10-13" data-line-number="13"></a>
<a class="sourceLine" id="cb10-14" data-line-number="14">featureAlignmentsNto1 &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb10-15" data-line-number="15">  <span class="dt">refSequence =</span> refSeq, </a>
<a class="sourceLine" id="cb10-16" data-line-number="16">  <span class="dt">querySequences =</span> queryGenomes, </a>
<a class="sourceLine" id="cb10-17" data-line-number="17">  <span class="dt">type =</span> <span class="st">&quot;global-local&quot;</span>,</a>
<a class="sourceLine" id="cb10-18" data-line-number="18">  <span class="dt">outfile =</span> outfiles[featureName])</a>
<a class="sourceLine" id="cb10-19" data-line-number="19"></a>
<a class="sourceLine" id="cb10-20" data-line-number="20"><span class="kw">kable</span>(featureAlignmentsNto1<span class="op">$</span>stats[<span class="kw">order</span>(featureAlignmentsNto1<span class="op">$</span>stat<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),], </a>
<a class="sourceLine" id="cb10-21" data-line-number="21">      <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of S genes&quot;</span>)</a></code></pre></div>
<table>
<caption>N-to-one alignment of S genes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">92.85714</td>
<td align="right">3822</td>
<td align="right">1</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5435.6235</td>
</tr>
<tr class="even">
<td>PnGu1_2019</td>
<td align="right">84.22836</td>
<td align="right">3836</td>
<td align="right">12</td>
<td align="right">38</td>
<td align="right">6</td>
<td align="right">14</td>
<td align="right">2752.7136</td>
</tr>
<tr class="odd">
<td>PnMP789</td>
<td align="right">84.17623</td>
<td align="right">3836</td>
<td align="right">12</td>
<td align="right">38</td>
<td align="right">6</td>
<td align="right">14</td>
<td align="right">2736.9521</td>
</tr>
<tr class="even">
<td>PnGX-P1E_2017</td>
<td align="right">83.42050</td>
<td align="right">3824</td>
<td align="right">8</td>
<td align="right">26</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">2534.8452</td>
</tr>
<tr class="odd">
<td>BtZC45</td>
<td align="right">76.61458</td>
<td align="right">3840</td>
<td align="right">24</td>
<td align="right">99</td>
<td align="right">11</td>
<td align="right">18</td>
<td align="right">405.0663</td>
</tr>
<tr class="even">
<td>BtZXC21</td>
<td align="right">75.87285</td>
<td align="right">3838</td>
<td align="right">24</td>
<td align="right">100</td>
<td align="right">9</td>
<td align="right">16</td>
<td align="right">198.5031</td>
</tr>
<tr class="odd">
<td>BtYu-RmYN02_2019</td>
<td align="right">74.75980</td>
<td align="right">3851</td>
<td align="right">43</td>
<td align="right">167</td>
<td align="right">14</td>
<td align="right">29</td>
<td align="right">-226.3011</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="right">73.76921</td>
<td align="right">3839</td>
<td align="right">21</td>
<td align="right">90</td>
<td align="right">13</td>
<td align="right">17</td>
<td align="right">-465.0419</td>
</tr>
</tbody>
</table>
<div id="spike-gene-pip-plot" class="section level3">
<h3>Spike gene PIP plot</h3>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co">## PIP profile of spike N-to-1 alignments</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> featureAlignmentsNto1<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                <span class="dt">windowSize =</span> <span class="dv">200</span>,  <span class="dt">colors =</span> strainColors,</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#                legend = paste0(names(featureAlignmentsNto1$alignments), &quot; (&quot;, round(digits = 2, featureAlignmentsNto1$stats$pid), &quot;%)&quot;),</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                <span class="dt">main =</span> <span class="kw">paste0</span>(featureName, <span class="st">&quot; - PIP profile&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, refGenomeName),</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">                <span class="dt">legendMargin =</span> <span class="dv">0</span>, <span class="dt">legendCorner =</span> <span class="st">&quot;bottomright&quot;</span>, <span class="dt">legendCex =</span> <span class="fl">0.8</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/pip_profiles/RNA_Sgene_pip-1.png" alt="Percent Identical Positions profile over the whole genome of SARS-CoV-2. " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole genome of SARS-CoV-2.
</p>
</div>
<!-- ### Getting sequences of spike gene in all the selected coronavirus genomes -->
<!-- ```{r  match sequence fragment against all genomes and export the matches to a fasta file} -->
<!-- #' @title Export sequencesquery + subject sequences from N-to-1 alignments -->
<!-- #' @author Jacques.van-Helden@france)bioinformatique.fr -->
<!-- #' @param refsequence  sequence used as reference (DB) for the N-to-1 alignment -->
<!-- #' @param genomes genome sequences -->
<!-- #' @param IDsuffix=NULL if not null, append the suffix to the sequence name for the fasta header -->
<!-- #' @param outfile=NULL output file. If specified, the matches will be exported as fasta-formatted sequences. -->
<!-- #' @export -->
<!-- ExportSegmentMatches <- function(refsequence, -->
<!--                                  genomes, -->
<!--                                  IDsuffix = NULL, -->
<!--                                  outfile=NULL) { -->
<!--   #### N-to-1 alignments of all spike-coding sequences #### -->
<!--   ## Get sequences for reference and query genomes -->
<!--   alignmentsNto1 <- alignNtoOne( -->
<!--     refSequence = refsequence,  -->
<!--     querySequences = genomes,  -->
<!--     type = "global-local") -->
<!--   if (!is.null(outfile)) { -->
<!--     ## Write the reference sequence in the output fle -->
<!--     writeXStringSet(refsequence, filepath = outfile, format = "fasta") -->
<!--     i <- 1 -->
<!--     nbAlignments <- length(alignmentsNto1$alignments) -->
<!--     for (i in 1:nbAlignments) { -->
<!--       genomeName <- names(genomes[i]) -->
<!--       alignment <- alignmentsNto1$alignments[[i]] -->
<!--       subject <- subject(alignment) -->
<!--       ## Suppress the dashes from the alignment to get the raw sequence -->
<!--       sequence <- as.character(subject) -->
<!--       sequenceDesaligned <- gsub(pattern = "-", replacement = "", x = sequence) -->
<!--       seqStringSet <- DNAStringSet(x = sequenceDesaligned) #, start = start(subject), end=end(subject)) -->
<!--       ## Define a sequence ID for the fasta header -->
<!--       sequenceID <- genomeName -->
<!--       if (!is.null(IDsuffix)) { -->
<!--         sequenceID <- paste0(sequenceID, IDsuffix) -->
<!--       }  -->
<!--       sequenceID <- paste0(sequenceID, "_", start(subject), "-", end(subject)) -->
<!--       names(seqStringSet) <- sequenceID -->
<!--       #  -->
<!--       ## Write pairwise alignment (temporarily disaactivated) -->
<!--       # alignmentFile <- paste0("pairwise-alignment_",  -->
<!--       #                         # gsub(pattern = "/", replacement = "-", x = genomeName),  -->
<!--       #                         ".txt") -->
<!--       # writePairwiseAlignments(x = alignment, file = outfile) -->
<!--       ## Append the sequence to the file -->
<!--       message("\tAppending sequence ", i, "/", nbAlignments, "\t", sequenceID) -->
<!--       writeXStringSet(seqStringSet, -->
<!--                       filepath = outfile, format = "fasta", append = TRUE) -->
<!--     } -->
<!--     message("\tExported alignments to\t", outfile) -->
<!--   }  -->
<!--   return(alignmentsNto1) -->
<!-- } -->
<!-- ``` -->
<!-- ```{r Sgene_exportseq} -->
<!-- #### Export S gene sequences found in the reference genomes #### -->
<!-- # toString(refSeq) -->
<!-- writeXStringSet( -->
<!--   refSeq,  -->
<!--   filepath = file.path( -->
<!--     dir$outseq,  -->
<!--     paste0("S-gene_", refGenomeName, ".fasta")),  -->
<!--   format = "fasta") -->
<!-- ## Export matches to S-gene -->
<!-- outfiles["S gene matches"] <- file.path( -->
<!--   dir$outseq,  -->
<!--   paste0("S-gene_", refGenomeName, "_matches.fasta")) -->
<!-- featureAlignmentsNto1All <- ExportSegmentMatches( -->
<!--   refsequence = refSeq, -->
<!--   genomes = genomes,  -->
<!--   IDsuffix = "_S-gene", -->
<!--   outfile = outfiles["S gene matches"]) -->
<!-- kable(featureAlignmentsNto1All$stats[order(featureAlignmentsNto1All$stat$score, decreasing = TRUE),],  -->
<!--         caption = "N-to-one alignment of S genes") -->
<!-- ``` -->
<!-- ## Export matches to the coding sequence of the S1 cleavage product -->
<!-- ## N-to-1 alignemnts of S1 coding sequences -->
<!-- ```{r S1_align_queries} -->
<!-- refS1Limits <- refSpikeLimits[1] + 3*c(13, 685) -->
<!-- #### N-to-1 alignments of S1-coding sequences #### -->
<!-- ## Get sequences for reference and query genomes -->
<!-- refS1 <- subseq(genomes[refGenomeName], start = refS1Limits[1], end = refS1Limits[2]) -->
<!-- S1Nto1 <- alignNtoOne(refSequence = refS1, querySequences = queryGenomes, type = "global-local") -->
<!-- kable(S1Nto1$stats[order(S1Nto1$stat$score, decreasing = TRUE),],  -->
<!--       caption = "N-to-one alignment of S1 coding sequence") -->
<!-- ``` -->
<!-- ### S1 gene PIP plot -->
<!-- ```{r S1_pip, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole genome of SARS-CoV-2. "} -->
<!-- ## PIP profile of S1 N-to-1 alignments -->
<!-- plotPIPprofiles(alignments = S1Nto1$alignments,  -->
<!--                 windowSize = 200, colors = strainColors,  -->
<!--                 legend = paste0(names(S1Nto1$alignments), " (", round(digits = 2, S1Nto1$stats$pid), "%)"), -->
<!--                 main = paste0("S1 coding sequence - PIP profile", "\nRef: ", refGenomeName), -->
<!--                 legendMargin = 0, legendCorner = "bottomleft", legendCex = 0.8, ylim = c(30, 100)) -->
<!-- ``` -->
</div>
</div>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">kable</span>(<span class="kw">t</span>(<span class="kw">as.data.frame</span>(dir)), <span class="dt">col.names =</span> <span class="st">&quot;Dir&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Directories&quot;</span>)</a></code></pre></div>
<table>
<caption>Directories</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">Dir</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>main</td>
<td align="left">..</td>
</tr>
<tr class="even">
<td>R</td>
<td align="left">../scripts/R</td>
</tr>
<tr class="odd">
<td>outseq</td>
<td align="left">../results/S-gene/Nto1_alignments</td>
</tr>
<tr class="even">
<td>genomes</td>
<td align="left">../data/GISAID_genomes</td>
</tr>
<tr class="odd">
<td>CDS.S</td>
<td align="left">../results/CDS-S</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">kable</span>(outfiles, <span class="dt">col.names =</span> <span class="st">&quot;File&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Output files&quot;</span>)</a></code></pre></div>
<table>
<caption>Output files</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">File</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Genome alignments</td>
<td align="left">../results/S-gene/Nto1_alignments/genome_alignments_ref_HuCoV2_WH01_2019</td>
</tr>
<tr class="even">
<td>Genome alignments - SARS</td>
<td align="left">../results/S-gene/Nto1_alignments/genome_alignments_ref_HuSARS-Frankfurt-1_2003</td>
</tr>
<tr class="odd">
<td>CDS-S</td>
<td align="left">../results/CDS-S/CDS-S_around-CoV-2_matches.fasta</td>
</tr>
</tbody>
</table>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Biostrings_2.56.0   XVector_0.28.0      IRanges_2.22.2      S4Vectors_0.26.1    BiocGenerics_0.34.0 knitr_1.28         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6        crayon_1.3.4        digest_0.6.25       magrittr_1.5        evaluate_0.14       highr_0.8           zlibbioc_1.34.0     rlang_0.4.6         stringi_1.4.6       rmarkdown_2.1       tools_4.0.0         stringr_1.4.0       xfun_0.14           yaml_2.2.1         
[15] compiler_4.0.0      BiocManager_1.30.10 htmltools_0.4.0    </code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
