---
title: "Analysis of coronavirus sequences"
author: "Jacques van Helden"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    code_folding: hide
    fig_caption: yes
    highlight: zenburn
    self_contained: no
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  ioslides_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    smaller: yes
    toc: yes
    widescreen: yes
  beamer_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    incremental: no
    keep_tex: no
    slide_level: 2
    theme: Montpellier
    toc: yes
  word_document:
    toc: yes
    toc_depth: '3'
  slidy_presentation:
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    highlight: tango
    incremental: no
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
font-import: http://fonts.googleapis.com/css?family=Risque
subtitle:  Normalization and target selection methods
font-family: Garamond
transition: linear
---

```{r libraries, echo=FALSE, results=FALSE, warning=FALSE, message=FALSE}
#### Install required packages ####
required.packages <- c()

for (pkg in required.packages) {
  if (!require(pkg, character.only = TRUE)) {
    message("Installing package ", pkg)
    install.packages(pkg, dependencies = TRUE)
  }
  require(pkg, character.only = TRUE)
}

#### Load libraries ####
if (!requireNamespace("BiocManager", quietly = TRUE)) {
  install.packages("BiocManager")
}
if (!require("Biostrings", quietly = TRUE)) {
  BiocManager::install("Biostrings")
  require("Biostrings")
}


```



```{r knitr_settings, include=FALSE, echo=FALSE, eval=TRUE}
library(knitr)
options(width = 300)
knitr::opts_chunk$set(
  fig.width = 7, fig.height = 5, 
  fig.path = 'figures/cov-seq_',
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = TRUE, comment = "")
# knitr::asis_output("\\footnotesize")


## Store original graphic parameters to restore them after chunks
par.ori <- par(no.readonly = TRUE)



```


```{r parameters}

#### Define directories and files ####
dir <- list(main = '..')
dir$R <- file.path(dir$main, "scripts/R")

## Input files
infiles <- list()

## Genome dir and files
dir$genomes <- file.path(dir$main, "data/virus_genomes/")
# infiles$genomes <- file.path(dir$genomes, "coronavirus_selected_genomes.fasta")

## Genome sequences
infiles$genomes <- file.path(dir$genomes, "GISAID_genomes", "coronavirus_selected-plus-GISAID_genomes.fasta")
if (!file.exists(infiles$genomes)) {
  stop("Genome sequence file is missing", "\n", infiles$genomes)
}


## Output tables
# di$output <- file.path(dir.main, "")
# dir$tables <- 

## Load custom functions
source(file.path(dir$R, "align_n_to_one.R"))
source(file.path(dir$R, "plot_pip_profiles.R"))

## Reference genome
refPattern <- "Hum_SARS-CoV-2_Whu1"
# refPattern <- "SARS-CoV-2_BetaCoV/Wuhan/IPBCAMS-WH-01/2019"
refSpikeLimits <- c(21563, 25384) ## Coordinates of the S gene in the reference genome (coding for the spike protein)

## Query genomes
queryPatterns <- c(
  "GX-P1E_2017/2020", 
  "Bat_RaTG13", 
  "Bat_SL-CoVZC45",
  "Bat_SL-CoVZXC21",
  "Hum_SARS-CoV_Frankfurt_1"
)

#### Add GISAID IDs to the query pattern ####
## Note that GISAI genomes are be submitted to the github repo because they cannot be redistributed
queryPatterns <- append(queryPatterns, 
                        c("Bat_Yu-RmYN02_2019",
                          "Png_Gu1_2019"
                          ))

message("Number of query genomes: ", length(queryPatterns))

```



```{r load_sequences}

#### Load genome sequences ####
genomes <- readDNAStringSet(filepath = infiles$genome, format = "fasta")
genomeNames <- names(genomes)
nbGenomes <- length(genomeNames)
message("Loaded ", nbGenomes, " genomes from file ", infiles$genomes)

## Shorten sequence names by suppressing the fasta comment (after the space)
names(genomes) <- sub(pattern = " .*", replacement = "", x = names(genomes), perl = TRUE)

# View(genomes)

#### Define reference and query genomes ####
refGenomeName <- grep(pattern = refPattern, x = names(genomes), 
                  ignore.case = TRUE, value = TRUE)
if (is.null(refGenomeName)) {
  stop("Could not identify reference genome with pattern ", refPattern)
}
message("Reference genome name: ", refGenomeName)

## Query genomes
queryRegExp <- paste0("(", paste(collapse = ")|(", queryPatterns), ")")
queryGenomeNames <- grep(pattern = queryRegExp, 
                    x = names(genomes), 
                    ignore.case = TRUE, value = TRUE)
nbQueryGenomes <- length(queryGenomeNames)

if (nbQueryGenomes == 0) {
  stop("Could not identify any query genome with query pattern\n", queryRegExp)
}

if (length(queryPatterns) != length(queryGenomeNames)) {
  foundPatterns <- grep(pattern = queryRegExp, x = queryGenomeNames, value = TRUE)
  missingPatterns <- setdiff(queryPatterns, queryGenomeNames)
  message("\t", length(missingGenomes), " Missing : ", paste(sep = ", ", missingPatterns))
  stop("Missing genomes")
}


## Compute some statistics about genome sizes
genomeStat <- data.frame(
  row.names = c(refGenomeName, queryGenomeNames),
  status = c("Reference", rep("Query", length.out = length(queryGenomeNames)))
)

for (g in c(refGenomeName, queryGenomeNames)) {
  genomeStat[g, "length"] <- length(genomes[[g]])
}

kable(genomeStat, caption = "Reference and query genomes")

```


## N-to-1 full genome alignments

We perform a pairwise lignment between each genome query and the reference genome (`r refGenomeName`).

```{r full-genomes_align}
#### N-to-1 genome alignments ####

## Get sequences for reference and query genomes
refGenome <- genomes[refGenomeName]
queryGenomes <- genomes[queryGenomeNames]
genomesNto1 <- alignNtoOne(refSequence = refGenome, querySequences = queryGenomes)

kable(genomesNto1$stats[order(genomesNto1$stats$score, decreasing = TRUE), ], 
      caption = "N-to-one alignment of full genomes")



```

#### Full genome PIP plot


```{r genome_pip, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole genome of SARS-CoV-2. "}

## PIP profile of full genome N-to-1 alignments
plotPIPprofiles(alignments = genomesNto1$alignments, 
                windowSize = 800, 
                legend = paste0(names(genomesNto1$alignments), " (", round(digits = 2, genomesNto1$stats$pid), "%)"),
                main = paste0("Full genome PIP profile", "\nRef: ", refGenomeName),
                legendMargin = 0, legendCorner = "bottom", legendCex = 0.7, ylim = c(40,100))


```


## SARS vs Civet

```{r SARS_vs_civet_genome, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole genome of SARS (2002-2003). "}
#### Compare SARS (2002) wih the related Civet genome ####
SARSvsCivetGenome <- alignNtoOne(
  refSequence = genomes["Human_SARS-CoV_Frankfurt_1"], 
  querySequences = genomes["Civet_SARS-CoV_007/2004"])

## PIP profile of full genome N-to-1 alignments
plotPIPprofiles(alignments = SARSvsCivetGenome$alignments, 
                windowSize = 500, 
                legend = paste0(names(SARSvsCivetGenome$alignments), " (", round(digits = 2, SARSvsCivetGenome$stats$pid), "%)"),
                main = paste0("Percent Identical Positions - Full genome", "\nRef: ", "Human_SARS-CoV_Frankfurt_1"),
                legendMargin = 0, legendCorner = "bottom", legendCex = 0.7, ylim = c(40,100))


```




## N-to-1 alignemnts of spike genes

```{r S-gene_align_queries}

#### N-to-1 alignments of spike-coding sequences ####

## Get sequences for reference and query genomes
refSpike <- subseq(genomes[refGenomeName], start = refSpikeLimits[1], end = refSpikeLimits[2])

spikesNto1 <- alignNtoOne(refSequence = refSpike, querySequences = queryGenomes, type = "global-local")

  kable(spikesNto1$stats[order(spikesNto1$stat$score, decreasing = TRUE),], 
      caption = "N-to-one alignment of S genes")

```

### Spike protein PIP plot

```{r Sgene_pip, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole genome of SARS-CoV-2. "}

## PIP profile of spike N-to-1 alignments
plotPIPprofiles(alignments = spikesNto1$alignments, 
                windowSize = 200, 
                legend = paste0(names(spikesNto1$alignments), " (", round(digits = 2, spikesNto1$stats$pid), "%)"),
                main = paste0("S gene - PIP profile", "\nRef: ", refGenomeName),
                legendMargin = 0, legendCorner = "bottomright", legendCex = 0.8, ylim = c(30, 100))

```



### Getting spike proteins in all the selected coronavirus genomes


```{r S-gene_align_all}

#### N-to-1 alignments of all spike-coding sequences ####

## Get sequences for reference and query genomes
spikesNto1All <- alignNtoOne(refSequence = refSpike, querySequences = genomes, type = "global-local")


kable(spikesNto1All$stats[order(spikesNto1All$stat$score, decreasing = TRUE),], 
      caption = "N-to-one alignment of S genes")

```

```{r Sgene_exportseq}
#### Export S gene sequences found in the regerence genomes ####
# toString(refSpike)

dir$outseq <- file.path(dir$main, "results", "S-gene", "Nto1_alignments")
dir.create(dir$outseq, showWarnings = FALSE, recursive = TRUE)

outfiles <- list()

writeXStringSet(refSpike, 
                filepath = file.path(dir$outseq, paste0("S-gene_", refGenomeName, ".fasta")), format = "fasta")

outfiles$Smatches <- file.path(dir$outseq, paste0("S-gene_", refGenomeName, "_matches.fasta"))
writeXStringSet(refSpike, 
                filepath = outfiles$Smatches, format = "fasta")

i <- 1
nbAlignments <- length(spikesNto1All$alignments)
for (i in 1:nbAlignments) {
  genomeName <- names(genomes[i])
  alignment <- spikesNto1All$alignments[[i]]
  
  alignmentFile <- paste0("pairwise-alignment_", 
                       # gsub(pattern = "/", replacement = "-", x = genomeName), 
                       ".txt")
  writePairwiseAlignments(x = alignment, file = file.path(dir$outseq, alignmentFile))
  
  subject <- subject(alignment)
  sequence <- as.character(subject)
  sequenceDesaligned <- gsub(pattern = "-", replacement = "", x = sequence)
  sequenceID <- paste(sep = "_", genomeName, "S-gene", start(subject), end(subject))
  seqStringSet <- DNAStringSet(x = sequenceDesaligned) #, start = start(subject), end=end(subject))
  names(seqStringSet) <- sequenceID
  message("\tAppending sequence ", i, "/", nbAlignments, "\t", sequenceID)
  writeXStringSet(seqStringSet,
                  filepath = outfiles$Smatches, format = "fasta", append = TRUE)

}
message("\t", outfiles$Smatches)
# 
```




