<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-05-09" />

<title>Analysis of coronavirus sequences</title>

<script src="pip-profiles_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="pip-profiles_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="pip-profiles_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="pip-profiles_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="pip-profiles_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="pip-profiles_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="pip-profiles_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="pip-profiles_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="pip-profiles_files/navigation-1.1/tabsets.js"></script>
<script src="pip-profiles_files/navigation-1.1/codefolding.js"></script>
<link href="pip-profiles_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="pip-profiles_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of coronavirus sequences</h1>
<h3 class="subtitle">Normalization and target selection methods</h3>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-05-09</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#### Define directories and files ####</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">dir &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">main =</span> <span class="st">&#39;..&#39;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">dir<span class="op">$</span>genomes &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data/virus_genomes/&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">dir<span class="op">$</span>R &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;scripts/R&quot;</span>)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">## Output tables</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co"># di$output &lt;- file.path(dir.main, &quot;&quot;)</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co"># dir$tables &lt;- </span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">## Load custom functions</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;align_n_to_one.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;plot_pip_profiles.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">## Reference genome</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15">refPattern &lt;-<span class="st"> &quot;Human_SARS-CoV-2_Whu1&quot;</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="co"># refPattern &lt;- &quot;SARS-CoV-2_BetaCoV/Wuhan/IPBCAMS-WH-01/2019&quot;</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17">refSpikeLimits &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">21563</span>, <span class="dv">25384</span>) <span class="co">## Coordinates of the S gene in the reference genome (coding for the spike protein)</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="co">## Query genomes</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20">queryPatterns &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">#  &quot;SARS-CoV-2_WIV04&quot;,</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co">#  &quot;Pangolin_MP789&quot;,</span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23">  <span class="st">&quot;Pangolin_PCoV_GX-P1E_2017/2020&quot;</span>, </a>
<a class="sourceLine" id="cb1-24" data-line-number="24">  <span class="st">&quot;Bat_RaTG13&quot;</span>, </a>
<a class="sourceLine" id="cb1-25" data-line-number="25">  <span class="st">&quot;Bat_bat-SL-CoVZC45&quot;</span>,</a>
<a class="sourceLine" id="cb1-26" data-line-number="26">  <span class="st">&quot;Bat_bat-SL-CoVZXC21&quot;</span>,</a>
<a class="sourceLine" id="cb1-27" data-line-number="27">  <span class="st">&quot;Human_SARS-CoV_Frankfurt_1&quot;</span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28">)</a>
<a class="sourceLine" id="cb1-29" data-line-number="29"></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co">#### Add GISAID IDs to the query pattern ####</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="co">## Note that GISAI genomes are be submitted to the github repo because they cannot be redistributed</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32">queryPatterns &lt;-<span class="st"> </span><span class="kw">append</span>(queryPatterns, </a>
<a class="sourceLine" id="cb1-33" data-line-number="33">                        <span class="kw">c</span>(<span class="st">&quot;Bat_Yu-RmYN02_2019&quot;</span>,</a>
<a class="sourceLine" id="cb1-34" data-line-number="34"><span class="co">#                          &quot;Pangolin_Gu-P2S_2019&quot;,</span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35">                          <span class="st">&quot;Pangolin_Gu1_2019&quot;</span></a>
<a class="sourceLine" id="cb1-36" data-line-number="36">                          ))</a>
<a class="sourceLine" id="cb1-37" data-line-number="37"></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="co">#                   &quot;Civet_SARS-CoV_007/2004&quot;,</span></a>
<a class="sourceLine" id="cb1-39" data-line-number="39"><span class="co"># &quot;SARS-CoV-2_BetaCoV/Wuhan/IPBCAMS-WH-01/2019&quot;,</span></a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="co">#                   &quot;SARS-CoV-2_Whu1&quot;,</span></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="co">#                   &quot;Camel_MERS&quot;,</span></a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">#### Load genome sequences ####</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">infiles &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co"># infiles$genomes &lt;- file.path(dir$genomes, &quot;coronavirus_selected_genomes.fasta&quot;)</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">infiles<span class="op">$</span>genomes &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>genomes, <span class="st">&quot;sequences_from_GISAID&quot;</span>, <span class="st">&quot;coronavirus_selected-plus-GISAID_genomes.fasta&quot;</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">genomes &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="dt">filepath =</span> infiles<span class="op">$</span>genome, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">genomeNames &lt;-<span class="st"> </span><span class="kw">names</span>(genomes)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">nbGenomes &lt;-<span class="st"> </span><span class="kw">length</span>(genomeNames)</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">message</span>(<span class="st">&quot;Loaded &quot;</span>, nbGenomes, <span class="st">&quot; genomes from file &quot;</span>, infiles<span class="op">$</span>genomes)</a>
<a class="sourceLine" id="cb2-10" data-line-number="10"></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">## Shorten sequence names by suppressing the fasta comment (after the space)</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="kw">names</span>(genomes) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">&quot; .*&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="dt">x =</span> <span class="kw">names</span>(genomes), <span class="dt">perl =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-13" data-line-number="13"></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co"># View(genomes)</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#### Define reference and query genomes ####</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17">refGenomeName &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> refPattern, <span class="dt">x =</span> <span class="kw">names</span>(genomes), </a>
<a class="sourceLine" id="cb2-18" data-line-number="18">                  <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="cf">if</span> (<span class="kw">is.null</span>(refGenomeName)) {</a>
<a class="sourceLine" id="cb2-20" data-line-number="20">  <span class="kw">stop</span>(<span class="st">&quot;Could not identify reference genome with pattern &quot;</span>, refPattern)</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">}</a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="kw">message</span>(<span class="st">&quot;Reference genome name: &quot;</span>, refGenomeName)</a>
<a class="sourceLine" id="cb2-23" data-line-number="23"></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co">## Query genomes</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25">queryRegExp &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;(&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;)|(&quot;</span>, queryPatterns), <span class="st">&quot;)&quot;</span>)</a>
<a class="sourceLine" id="cb2-26" data-line-number="26">queryGenomeNames &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> queryRegExp, </a>
<a class="sourceLine" id="cb2-27" data-line-number="27">                    <span class="dt">x =</span> <span class="kw">names</span>(genomes), </a>
<a class="sourceLine" id="cb2-28" data-line-number="28">                    <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-29" data-line-number="29">nbQueryGenomes &lt;-<span class="st"> </span><span class="kw">length</span>(queryGenomeNames)</a>
<a class="sourceLine" id="cb2-30" data-line-number="30"></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="cf">if</span> (nbQueryGenomes <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) {</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">  <span class="kw">stop</span>(<span class="st">&quot;Could not identify any query genome with query pattern</span><span class="ch">\n</span><span class="st">&quot;</span>, queryRegExp)</a>
<a class="sourceLine" id="cb2-33" data-line-number="33">}</a>
<a class="sourceLine" id="cb2-34" data-line-number="34"></a>
<a class="sourceLine" id="cb2-35" data-line-number="35"></a>
<a class="sourceLine" id="cb2-36" data-line-number="36"><span class="co">## Compute some statistics about genome sizes</span></a>
<a class="sourceLine" id="cb2-37" data-line-number="37">genomeStat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb2-38" data-line-number="38">  <span class="dt">row.names =</span> <span class="kw">c</span>(refGenomeName, queryGenomeNames),</a>
<a class="sourceLine" id="cb2-39" data-line-number="39">  <span class="dt">status =</span> <span class="kw">c</span>(<span class="st">&quot;Reference&quot;</span>, <span class="kw">rep</span>(<span class="st">&quot;Query&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(queryGenomeNames)))</a>
<a class="sourceLine" id="cb2-40" data-line-number="40">)</a>
<a class="sourceLine" id="cb2-41" data-line-number="41"></a>
<a class="sourceLine" id="cb2-42" data-line-number="42"><span class="cf">for</span> (g <span class="cf">in</span> <span class="kw">c</span>(refGenomeName, queryGenomeNames)) {</a>
<a class="sourceLine" id="cb2-43" data-line-number="43">  genomeStat[g, <span class="st">&quot;length&quot;</span>] &lt;-<span class="st"> </span><span class="kw">length</span>(genomes[[g]])</a>
<a class="sourceLine" id="cb2-44" data-line-number="44">}</a>
<a class="sourceLine" id="cb2-45" data-line-number="45"></a>
<a class="sourceLine" id="cb2-46" data-line-number="46"><span class="kw">kable</span>(genomeStat, <span class="dt">caption =</span> <span class="st">&quot;Reference and query genomes&quot;</span>)</a></code></pre></div>
<table>
<caption>Reference and query genomes</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">status</th>
<th align="right">length</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Human_SARS-CoV-2_Whu1</td>
<td align="left">Reference</td>
<td align="right">29903</td>
</tr>
<tr class="even">
<td>Bat_bat-SL-CoVZC45</td>
<td align="left">Query</td>
<td align="right">29802</td>
</tr>
<tr class="odd">
<td>Bat_bat-SL-CoVZXC21</td>
<td align="left">Query</td>
<td align="right">29732</td>
</tr>
<tr class="even">
<td>Bat_RaTG13</td>
<td align="left">Query</td>
<td align="right">29855</td>
</tr>
<tr class="odd">
<td>Human_SARS-CoV_Frankfurt_1</td>
<td align="left">Query</td>
<td align="right">29727</td>
</tr>
<tr class="even">
<td>Pangolin_PCoV_GX-P1E_2017/2020</td>
<td align="left">Query</td>
<td align="right">29801</td>
</tr>
<tr class="odd">
<td>Bat_Yu-RmYN02_2019</td>
<td align="left">Query</td>
<td align="right">29671</td>
</tr>
<tr class="even">
<td>Pangolin_Gu1_2019</td>
<td align="left">Query</td>
<td align="right">29825</td>
</tr>
</tbody>
</table>
<div id="n-to-1-full-genome-alignments" class="section level2">
<h2>N-to-1 full genome alignments</h2>
<p>We perform a pairwise lignment between each genome query and the reference genome (Human_SARS-CoV-2_Whu1).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#### N-to-1 genome alignments ####</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">## Get sequences for reference and query genomes</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">refGenome &lt;-<span class="st"> </span>genomes[refGenomeName]</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">queryGenomes &lt;-<span class="st"> </span>genomes[queryGenomeNames]</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">genomesNto1 &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(<span class="dt">refSequence =</span> refGenome, <span class="dt">querySequences =</span> queryGenomes)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7"></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="kw">kable</span>(genomesNto1<span class="op">$</span>stats[<span class="kw">order</span>(genomesNto1<span class="op">$</span>stats<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), ], </a>
<a class="sourceLine" id="cb3-9" data-line-number="9">      <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of full genomes&quot;</span>)</a></code></pre></div>
<table>
<caption>N-to-one alignment of full genomes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Bat_RaTG13</td>
<td align="right">96.07200</td>
<td align="right">29888</td>
<td align="right">6</td>
<td align="right">33</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">49903.32</td>
</tr>
<tr class="even">
<td>Bat_Yu-RmYN02_2019</td>
<td align="right">93.71046</td>
<td align="right">29875</td>
<td align="right">59</td>
<td align="right">204</td>
<td align="right">22</td>
<td align="right">46</td>
<td align="right">43742.79</td>
</tr>
<tr class="odd">
<td>Pangolin_Gu1_2019</td>
<td align="right">90.26895</td>
<td align="right">29894</td>
<td align="right">16</td>
<td align="right">69</td>
<td align="right">9</td>
<td align="right">23</td>
<td align="right">36107.67</td>
</tr>
<tr class="even">
<td>Bat_bat-SL-CoVZC45</td>
<td align="right">87.87211</td>
<td align="right">29931</td>
<td align="right">33</td>
<td align="right">132</td>
<td align="right">16</td>
<td align="right">28</td>
<td align="right">30527.45</td>
</tr>
<tr class="odd">
<td>Bat_bat-SL-CoVZXC21</td>
<td align="right">87.61946</td>
<td align="right">29926</td>
<td align="right">37</td>
<td align="right">197</td>
<td align="right">13</td>
<td align="right">23</td>
<td align="right">30030.24</td>
</tr>
<tr class="even">
<td>Pangolin_PCoV_GX-P1E_2017/2020</td>
<td align="right">85.40635</td>
<td align="right">29876</td>
<td align="right">27</td>
<td align="right">77</td>
<td align="right">13</td>
<td align="right">19</td>
<td align="right">24440.96</td>
</tr>
<tr class="odd">
<td>Human_SARS-CoV_Frankfurt_1</td>
<td align="right">79.54068</td>
<td align="right">30001</td>
<td align="right">76</td>
<td align="right">276</td>
<td align="right">63</td>
<td align="right">131</td>
<td align="right">10318.37</td>
</tr>
</tbody>
</table>
<div id="full-genome-pip-plot" class="section level4">
<h4>Full genome PIP plot</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">## PIP profile of full genome N-to-1 alignments</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> genomesNto1<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb4-3" data-line-number="3">                <span class="dt">windowSize =</span> <span class="dv">800</span>, </a>
<a class="sourceLine" id="cb4-4" data-line-number="4">                <span class="dt">legend =</span> <span class="kw">paste0</span>(<span class="kw">names</span>(genomesNto1<span class="op">$</span>alignments), <span class="st">&quot; (&quot;</span>, <span class="kw">round</span>(<span class="dt">digits =</span> <span class="dv">2</span>, genomesNto1<span class="op">$</span>stats<span class="op">$</span>pid), <span class="st">&quot;%)&quot;</span>),</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Full genome PIP profile&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, refGenomeName),</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">                <span class="dt">legendMargin =</span> <span class="dv">0</span>, <span class="dt">legendCorner =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legendCex =</span> <span class="fl">0.7</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">40</span>,<span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/cov-seq_genome_pip-1.png" alt="Percent Identical Positions profile over the whole genome of SARS-CoV-2. " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole genome of SARS-CoV-2.
</p>
</div>
</div>
</div>
<div id="sars-vs-civet" class="section level2">
<h2>SARS vs Civet</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co">#### Compare SARS (2002) wih the related Civet genome ####</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">SARSvsCivetGenome &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">  <span class="dt">refSequence =</span> genomes[<span class="st">&quot;Human_SARS-CoV_Frankfurt_1&quot;</span>], </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">  <span class="dt">querySequences =</span> genomes[<span class="st">&quot;Civet_SARS-CoV_007/2004&quot;</span>])</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">## PIP profile of full genome N-to-1 alignments</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> SARSvsCivetGenome<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb5-8" data-line-number="8">                <span class="dt">windowSize =</span> <span class="dv">500</span>, </a>
<a class="sourceLine" id="cb5-9" data-line-number="9">                <span class="dt">legend =</span> <span class="kw">paste0</span>(<span class="kw">names</span>(SARSvsCivetGenome<span class="op">$</span>alignments), <span class="st">&quot; (&quot;</span>, <span class="kw">round</span>(<span class="dt">digits =</span> <span class="dv">2</span>, SARSvsCivetGenome<span class="op">$</span>stats<span class="op">$</span>pid), <span class="st">&quot;%)&quot;</span>),</a>
<a class="sourceLine" id="cb5-10" data-line-number="10">                <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Percent Identical Positions - Full genome&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, <span class="st">&quot;Human_SARS-CoV_Frankfurt_1&quot;</span>),</a>
<a class="sourceLine" id="cb5-11" data-line-number="11">                <span class="dt">legendMargin =</span> <span class="dv">0</span>, <span class="dt">legendCorner =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legendCex =</span> <span class="fl">0.7</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">40</span>,<span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/cov-seq_SARS_vs_civet_genome-1.png" alt="Percent Identical Positions profile over the whole genome of SARS (2002-2003). " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole genome of SARS (2002-2003).
</p>
</div>
</div>
<div id="n-to-1-alignemnts-of-spike-genes" class="section level2">
<h2>N-to-1 alignemnts of spike genes</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co">#### N-to-1 alignments of spike-coding sequences ####</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">## Get sequences for reference and query genomes</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4">refSpike &lt;-<span class="st"> </span><span class="kw">subseq</span>(genomes[refGenomeName], <span class="dt">start =</span> refSpikeLimits[<span class="dv">1</span>], <span class="dt">end =</span> refSpikeLimits[<span class="dv">2</span>])</a>
<a class="sourceLine" id="cb6-5" data-line-number="5"></a>
<a class="sourceLine" id="cb6-6" data-line-number="6">spikesNto1 &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(<span class="dt">refSequence =</span> refSpike, <span class="dt">querySequences =</span> queryGenomes, <span class="dt">type =</span> <span class="st">&quot;global-local&quot;</span>)</a>
<a class="sourceLine" id="cb6-7" data-line-number="7"></a>
<a class="sourceLine" id="cb6-8" data-line-number="8">  <span class="kw">kable</span>(spikesNto1<span class="op">$</span>stats[<span class="kw">order</span>(spikesNto1<span class="op">$</span>stat<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),], </a>
<a class="sourceLine" id="cb6-9" data-line-number="9">      <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of S genes&quot;</span>)</a></code></pre></div>
<table>
<caption>N-to-one alignment of S genes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Bat_RaTG13</td>
<td align="right">92.85714</td>
<td align="right">3822</td>
<td align="right">1</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5435.6235</td>
</tr>
<tr class="even">
<td>Pangolin_Gu1_2019</td>
<td align="right">84.22836</td>
<td align="right">3836</td>
<td align="right">12</td>
<td align="right">38</td>
<td align="right">6</td>
<td align="right">14</td>
<td align="right">2752.7136</td>
</tr>
<tr class="odd">
<td>Pangolin_PCoV_GX-P1E_2017/2020</td>
<td align="right">83.42050</td>
<td align="right">3824</td>
<td align="right">8</td>
<td align="right">26</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">2534.8452</td>
</tr>
<tr class="even">
<td>Bat_bat-SL-CoVZC45</td>
<td align="right">76.61458</td>
<td align="right">3840</td>
<td align="right">24</td>
<td align="right">99</td>
<td align="right">11</td>
<td align="right">18</td>
<td align="right">405.0663</td>
</tr>
<tr class="odd">
<td>Bat_bat-SL-CoVZXC21</td>
<td align="right">75.87285</td>
<td align="right">3838</td>
<td align="right">24</td>
<td align="right">100</td>
<td align="right">9</td>
<td align="right">16</td>
<td align="right">198.5031</td>
</tr>
<tr class="even">
<td>Bat_Yu-RmYN02_2019</td>
<td align="right">74.75980</td>
<td align="right">3851</td>
<td align="right">43</td>
<td align="right">167</td>
<td align="right">14</td>
<td align="right">29</td>
<td align="right">-226.3011</td>
</tr>
<tr class="odd">
<td>Human_SARS-CoV_Frankfurt_1</td>
<td align="right">73.76921</td>
<td align="right">3839</td>
<td align="right">21</td>
<td align="right">90</td>
<td align="right">13</td>
<td align="right">17</td>
<td align="right">-465.0419</td>
</tr>
</tbody>
</table>
<div id="spike-protein-pip-plot" class="section level3">
<h3>Spike protein PIP plot</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co">## PIP profile of spike N-to-1 alignments</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> spikesNto1<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb7-3" data-line-number="3">                <span class="dt">windowSize =</span> <span class="dv">200</span>, </a>
<a class="sourceLine" id="cb7-4" data-line-number="4">                <span class="dt">legend =</span> <span class="kw">paste0</span>(<span class="kw">names</span>(spikesNto1<span class="op">$</span>alignments), <span class="st">&quot; (&quot;</span>, <span class="kw">round</span>(<span class="dt">digits =</span> <span class="dv">2</span>, spikesNto1<span class="op">$</span>stats<span class="op">$</span>pid), <span class="st">&quot;%)&quot;</span>),</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">                <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;S gene - PIP profile&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, refGenomeName),</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">                <span class="dt">legendMargin =</span> <span class="dv">0</span>, <span class="dt">legendCorner =</span> <span class="st">&quot;bottomright&quot;</span>, <span class="dt">legendCex =</span> <span class="fl">0.8</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/cov-seq_Sgene_pip-1.png" alt="Percent Identical Positions profile over the whole genome of SARS-CoV-2. " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole genome of SARS-CoV-2.
</p>
</div>
</div>
<div id="getting-spike-proteins-in-all-the-selected-coronavirus-genomes" class="section level3">
<h3>Getting spike proteins in all the selected coronavirus genomes</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co">#### N-to-1 alignments of all spike-coding sequences ####</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">## Get sequences for reference and query genomes</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4">spikesNto1All &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(<span class="dt">refSequence =</span> refSpike, <span class="dt">querySequences =</span> genomes, <span class="dt">type =</span> <span class="st">&quot;global-local&quot;</span>)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="kw">kable</span>(spikesNto1All<span class="op">$</span>stats[<span class="kw">order</span>(spikesNto1All<span class="op">$</span>stat<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),], </a>
<a class="sourceLine" id="cb8-8" data-line-number="8">      <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of S genes&quot;</span>)</a></code></pre></div>
<table>
<caption>N-to-one alignment of S genes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Human_SARS-CoV-2_Whu1</td>
<td align="right">100.00000</td>
<td align="right">3822</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">7574.5405</td>
</tr>
<tr class="even">
<td>Bat_RaTG13</td>
<td align="right">92.85714</td>
<td align="right">3822</td>
<td align="right">1</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5435.6235</td>
</tr>
<tr class="odd">
<td>Pangolin_Gu1_2019</td>
<td align="right">84.22836</td>
<td align="right">3836</td>
<td align="right">12</td>
<td align="right">38</td>
<td align="right">6</td>
<td align="right">14</td>
<td align="right">2752.7136</td>
</tr>
<tr class="even">
<td>Pangolin_PCoV_GX-P1E_2017/2020</td>
<td align="right">83.42050</td>
<td align="right">3824</td>
<td align="right">8</td>
<td align="right">26</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">2534.8452</td>
</tr>
<tr class="odd">
<td>Bat_bat-SL-CoVZC45</td>
<td align="right">76.61458</td>
<td align="right">3840</td>
<td align="right">24</td>
<td align="right">99</td>
<td align="right">11</td>
<td align="right">18</td>
<td align="right">405.0663</td>
</tr>
<tr class="even">
<td>Bat_bat-SL-CoVZXC21</td>
<td align="right">75.87285</td>
<td align="right">3838</td>
<td align="right">24</td>
<td align="right">100</td>
<td align="right">9</td>
<td align="right">16</td>
<td align="right">198.5031</td>
</tr>
<tr class="odd">
<td>Bat_Yu-RmYN02_2019</td>
<td align="right">74.75980</td>
<td align="right">3851</td>
<td align="right">43</td>
<td align="right">167</td>
<td align="right">14</td>
<td align="right">29</td>
<td align="right">-226.3011</td>
</tr>
<tr class="even">
<td>Bat_LYRa11</td>
<td align="right">74.17926</td>
<td align="right">3838</td>
<td align="right">25</td>
<td align="right">94</td>
<td align="right">9</td>
<td align="right">16</td>
<td align="right">-335.2238</td>
</tr>
<tr class="odd">
<td>Human_SARS-CoV_Frankfurt_1</td>
<td align="right">73.76921</td>
<td align="right">3839</td>
<td align="right">21</td>
<td align="right">90</td>
<td align="right">13</td>
<td align="right">17</td>
<td align="right">-465.0419</td>
</tr>
<tr class="even">
<td>Bat_SZ3</td>
<td align="right">73.96619</td>
<td align="right">3845</td>
<td align="right">26</td>
<td align="right">83</td>
<td align="right">15</td>
<td align="right">23</td>
<td align="right">-477.7650</td>
</tr>
<tr class="odd">
<td>Bat_BtRs-BetaCoV/YN2018B</td>
<td align="right">73.90739</td>
<td align="right">3844</td>
<td align="right">25</td>
<td align="right">85</td>
<td align="right">16</td>
<td align="right">22</td>
<td align="right">-493.6092</td>
</tr>
<tr class="even">
<td>Civet_SARS-CoV_007/2004</td>
<td align="right">73.57999</td>
<td align="right">3838</td>
<td align="right">20</td>
<td align="right">89</td>
<td align="right">12</td>
<td align="right">16</td>
<td align="right">-505.9898</td>
</tr>
<tr class="odd">
<td>Bat_HKU3-12</td>
<td align="right">72.94975</td>
<td align="right">3841</td>
<td align="right">29</td>
<td align="right">124</td>
<td align="right">12</td>
<td align="right">19</td>
<td align="right">-714.8980</td>
</tr>
<tr class="even">
<td>Bat-recombinant_Bat-SRBD</td>
<td align="right">72.87165</td>
<td align="right">3841</td>
<td align="right">29</td>
<td align="right">124</td>
<td align="right">12</td>
<td align="right">19</td>
<td align="right">-738.5422</td>
</tr>
<tr class="odd">
<td>Bat_Bat_SARS_CoV_Rm1/2004</td>
<td align="right">72.73201</td>
<td align="right">3836</td>
<td align="right">27</td>
<td align="right">122</td>
<td align="right">9</td>
<td align="right">14</td>
<td align="right">-743.2681</td>
</tr>
<tr class="even">
<td>Bat_HKU3-8</td>
<td align="right">72.90692</td>
<td align="right">3846</td>
<td align="right">30</td>
<td align="right">129</td>
<td align="right">15</td>
<td align="right">24</td>
<td align="right">-749.6403</td>
</tr>
<tr class="odd">
<td>Bat_BtRs-BetaCoV/YN2013</td>
<td align="right">72.76518</td>
<td align="right">3837</td>
<td align="right">33</td>
<td align="right">144</td>
<td align="right">10</td>
<td align="right">15</td>
<td align="right">-759.7230</td>
</tr>
<tr class="even">
<td>Bat_Rp3/2004</td>
<td align="right">72.55668</td>
<td align="right">3837</td>
<td align="right">25</td>
<td align="right">123</td>
<td align="right">13</td>
<td align="right">15</td>
<td align="right">-812.6312</td>
</tr>
<tr class="odd">
<td>Bat_BtRs-BetaCoV/GX2013</td>
<td align="right">72.41919</td>
<td align="right">3836</td>
<td align="right">27</td>
<td align="right">123</td>
<td align="right">8</td>
<td align="right">14</td>
<td align="right">-825.9421</td>
</tr>
<tr class="even">
<td>Bat_Rp/Shaanxi2011</td>
<td align="right">72.33212</td>
<td align="right">3842</td>
<td align="right">26</td>
<td align="right">122</td>
<td align="right">13</td>
<td align="right">20</td>
<td align="right">-883.9606</td>
</tr>
<tr class="odd">
<td>Bat_Cp/Yunnan2011</td>
<td align="right">72.14695</td>
<td align="right">3838</td>
<td align="right">25</td>
<td align="right">124</td>
<td align="right">10</td>
<td align="right">16</td>
<td align="right">-902.9724</td>
</tr>
<tr class="even">
<td>Bat_Rs_672/2006</td>
<td align="right">72.08637</td>
<td align="right">3844</td>
<td align="right">27</td>
<td align="right">122</td>
<td align="right">9</td>
<td align="right">22</td>
<td align="right">-925.0091</td>
</tr>
<tr class="odd">
<td>Bat_BtRl-BetaCoV/SC2018</td>
<td align="right">72.22512</td>
<td align="right">3838</td>
<td align="right">30</td>
<td align="right">125</td>
<td align="right">10</td>
<td align="right">16</td>
<td align="right">-927.4298</td>
</tr>
<tr class="even">
<td>Bat_BtRs-BetaCoV/YN2018C</td>
<td align="right">72.03125</td>
<td align="right">3840</td>
<td align="right">28</td>
<td align="right">126</td>
<td align="right">11</td>
<td align="right">18</td>
<td align="right">-970.8171</td>
</tr>
<tr class="odd">
<td>Pangolin_MP789</td>
<td align="right">70.19281</td>
<td align="right">3838</td>
<td align="right">27</td>
<td align="right">341</td>
<td align="right">8</td>
<td align="right">16</td>
<td align="right">-1081.8977</td>
</tr>
<tr class="even">
<td>Bat_YNLF_31C</td>
<td align="right">71.16085</td>
<td align="right">3842</td>
<td align="right">32</td>
<td align="right">128</td>
<td align="right">11</td>
<td align="right">20</td>
<td align="right">-1267.2054</td>
</tr>
<tr class="odd">
<td>Bat_JL2012</td>
<td align="right">70.55167</td>
<td align="right">3861</td>
<td align="right">34</td>
<td align="right">150</td>
<td align="right">16</td>
<td align="right">39</td>
<td align="right">-1450.2301</td>
</tr>
<tr class="even">
<td>Bat_BM48-31</td>
<td align="right">69.85485</td>
<td align="right">3858</td>
<td align="right">22</td>
<td align="right">83</td>
<td align="right">15</td>
<td align="right">36</td>
<td align="right">-1664.0186</td>
</tr>
<tr class="odd">
<td>Bat_BtKY72</td>
<td align="right">69.87296</td>
<td align="right">3857</td>
<td align="right">28</td>
<td align="right">95</td>
<td align="right">17</td>
<td align="right">35</td>
<td align="right">-1717.2273</td>
</tr>
<tr class="even">
<td>Pangolin_Gu-P2S_2019</td>
<td align="right">57.71847</td>
<td align="right">3822</td>
<td align="right">2</td>
<td align="right">968</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-3342.9629</td>
</tr>
<tr class="odd">
<td>Human_h-MERS_KOR/CNUH_SNU/172_06_2015</td>
<td align="right">51.83897</td>
<td align="right">4024</td>
<td align="right">114</td>
<td align="right">385</td>
<td align="right">72</td>
<td align="right">202</td>
<td align="right">-8043.8848</td>
</tr>
<tr class="even">
<td>Camel_MERS</td>
<td align="right">51.83988</td>
<td align="right">4022</td>
<td align="right">116</td>
<td align="right">383</td>
<td align="right">71</td>
<td align="right">200</td>
<td align="right">-8057.5645</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co">#### Export S gene sequences found in the regerence genomes ####</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co"># toString(refSpike)</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"></a>
<a class="sourceLine" id="cb9-4" data-line-number="4">dir<span class="op">$</span>outseq &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;results&quot;</span>, <span class="st">&quot;S-gene&quot;</span>, <span class="st">&quot;Nto1_alignments&quot;</span>)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="kw">dir.create</span>(dir<span class="op">$</span>outseq, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"></a>
<a class="sourceLine" id="cb9-7" data-line-number="7">outfiles &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb9-8" data-line-number="8"></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="kw">writeXStringSet</span>(refSpike, </a>
<a class="sourceLine" id="cb9-10" data-line-number="10">                <span class="dt">filepath =</span> <span class="kw">file.path</span>(dir<span class="op">$</span>outseq, <span class="kw">paste0</span>(<span class="st">&quot;S-gene_&quot;</span>, refGenomeName, <span class="st">&quot;.fasta&quot;</span>)), <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</a>
<a class="sourceLine" id="cb9-11" data-line-number="11"></a>
<a class="sourceLine" id="cb9-12" data-line-number="12">outfiles<span class="op">$</span>Smatches &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>outseq, <span class="kw">paste0</span>(<span class="st">&quot;S-gene_&quot;</span>, refGenomeName, <span class="st">&quot;_matches.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="kw">writeXStringSet</span>(refSpike, </a>
<a class="sourceLine" id="cb9-14" data-line-number="14">                <span class="dt">filepath =</span> outfiles<span class="op">$</span>Smatches, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</a>
<a class="sourceLine" id="cb9-15" data-line-number="15"></a>
<a class="sourceLine" id="cb9-16" data-line-number="16">i &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17">nbAlignments &lt;-<span class="st"> </span><span class="kw">length</span>(spikesNto1All<span class="op">$</span>alignments)</a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>nbAlignments) {</a>
<a class="sourceLine" id="cb9-19" data-line-number="19">  genomeName &lt;-<span class="st"> </span><span class="kw">names</span>(genomes[i])</a>
<a class="sourceLine" id="cb9-20" data-line-number="20">  alignment &lt;-<span class="st"> </span>spikesNto1All<span class="op">$</span>alignments[[i]]</a>
<a class="sourceLine" id="cb9-21" data-line-number="21">  </a>
<a class="sourceLine" id="cb9-22" data-line-number="22">  alignmentFile &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;pairwise-alignment_&quot;</span>, </a>
<a class="sourceLine" id="cb9-23" data-line-number="23">                       <span class="co"># gsub(pattern = &quot;/&quot;, replacement = &quot;-&quot;, x = genomeName), </span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24">                       <span class="st">&quot;.txt&quot;</span>)</a>
<a class="sourceLine" id="cb9-25" data-line-number="25">  <span class="kw">writePairwiseAlignments</span>(<span class="dt">x =</span> alignment, <span class="dt">file =</span> <span class="kw">file.path</span>(dir<span class="op">$</span>outseq, alignmentFile))</a>
<a class="sourceLine" id="cb9-26" data-line-number="26">  </a>
<a class="sourceLine" id="cb9-27" data-line-number="27">  subject &lt;-<span class="st"> </span><span class="kw">subject</span>(alignment)</a>
<a class="sourceLine" id="cb9-28" data-line-number="28">  sequence &lt;-<span class="st"> </span><span class="kw">as.character</span>(subject)</a>
<a class="sourceLine" id="cb9-29" data-line-number="29">  sequenceDesaligned &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="dt">pattern =</span> <span class="st">&quot;-&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="dt">x =</span> sequence)</a>
<a class="sourceLine" id="cb9-30" data-line-number="30">  sequenceID &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>, genomeName, <span class="st">&quot;S-gene&quot;</span>, <span class="kw">start</span>(subject), <span class="kw">end</span>(subject))</a>
<a class="sourceLine" id="cb9-31" data-line-number="31">  seqStringSet &lt;-<span class="st"> </span><span class="kw">DNAStringSet</span>(<span class="dt">x =</span> sequenceDesaligned) <span class="co">#, start = start(subject), end=end(subject))</span></a>
<a class="sourceLine" id="cb9-32" data-line-number="32">  <span class="kw">names</span>(seqStringSet) &lt;-<span class="st"> </span>sequenceID</a>
<a class="sourceLine" id="cb9-33" data-line-number="33">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Appending sequence &quot;</span>, i, <span class="st">&quot;/&quot;</span>, nbAlignments, <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, sequenceID)</a>
<a class="sourceLine" id="cb9-34" data-line-number="34">  <span class="kw">writeXStringSet</span>(seqStringSet,</a>
<a class="sourceLine" id="cb9-35" data-line-number="35">                  <span class="dt">filepath =</span> outfiles<span class="op">$</span>Smatches, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>, <span class="dt">append =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-36" data-line-number="36"></a>
<a class="sourceLine" id="cb9-37" data-line-number="37">}</a>
<a class="sourceLine" id="cb9-38" data-line-number="38"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, outfiles<span class="op">$</span>Smatches)</a>
<a class="sourceLine" id="cb9-39" data-line-number="39"><span class="co"># </span></a></code></pre></div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
