<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-05-24" />

<title>PIP profiles of coronavirus spike proteins - protein and RNA sequences</title>

<script src="PIP_spike-proteins_selected-plus-GISAID_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="PIP_spike-proteins_selected-plus-GISAID_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="PIP_spike-proteins_selected-plus-GISAID_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="PIP_spike-proteins_selected-plus-GISAID_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="PIP_spike-proteins_selected-plus-GISAID_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="PIP_spike-proteins_selected-plus-GISAID_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="PIP_spike-proteins_selected-plus-GISAID_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="PIP_spike-proteins_selected-plus-GISAID_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="PIP_spike-proteins_selected-plus-GISAID_files/navigation-1.1/tabsets.js"></script>
<script src="PIP_spike-proteins_selected-plus-GISAID_files/navigation-1.1/codefolding.js"></script>
<link href="PIP_spike-proteins_selected-plus-GISAID_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="PIP_spike-proteins_selected-plus-GISAID_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">PIP profiles of coronavirus spike proteins - protein and RNA sequences</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-05-24</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#### General parameters for the analysis ####</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">## Use (or not) GIDAID sequences</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">## </span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">## A few genoes were not available in NCBI Genbank at the time of </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">## this analysis, and had to be downloaded from GISAID. These sequences</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">##  can however not be redistributed, they should thus be downloaded </span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">##  manually to reproduce the full trees. Alternatively, useGISAID</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">##  can be set to FALSE, whcih will reproduce the analysis with almost </span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">##  all the sequences of the paper.</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11">useGISAID &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">#### Define directories and files ####</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15">dir &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">main =</span> <span class="st">&#39;..&#39;</span>)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">dir<span class="op">$</span>R &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;scripts/R&quot;</span>)</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">#### Create output directory for sequences ####</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19">seqPrefix &lt;-<span class="st"> &quot;spike_proteins&quot;</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20">dir<span class="op">$</span>outseq &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb1-21" data-line-number="21">  dir<span class="op">$</span>main, <span class="st">&quot;results&quot;</span>, seqPrefix, <span class="st">&quot;PIP_profiles&quot;</span>)</a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="kw">dir.create</span>(dir<span class="op">$</span>outseq, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-23" data-line-number="23"><span class="co"># list.files(dir$outseq)</span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24"></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="co">## Instantiate a list for output files</span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26">outfiles &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb1-27" data-line-number="27"></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">## Input files</span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29">infiles &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-30" data-line-number="30"></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="co">#### Sequence collection ####</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"></a>
<a class="sourceLine" id="cb1-33" data-line-number="33"><span class="co">## Supported: </span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34">collections &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-35" data-line-number="35">  <span class="st">&quot;around-CoV-2&quot;</span>, </a>
<a class="sourceLine" id="cb1-36" data-line-number="36">  <span class="st">&quot;selected&quot;</span>,</a>
<a class="sourceLine" id="cb1-37" data-line-number="37">  <span class="st">&quot;around-CoV-2-plus-GISAID&quot;</span>, </a>
<a class="sourceLine" id="cb1-38" data-line-number="38">  <span class="st">&quot;selected-plus-GISAID&quot;</span></a>
<a class="sourceLine" id="cb1-39" data-line-number="39">  )</a>
<a class="sourceLine" id="cb1-40" data-line-number="40"></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="co">## Selected collection</span></a>
<a class="sourceLine" id="cb1-42" data-line-number="42"><span class="co"># collection &lt;- &quot;around-CoV-2&quot; # 14 strains</span></a>
<a class="sourceLine" id="cb1-43" data-line-number="43"><span class="co"># collection &lt;- &quot;all&quot; # 16 strains</span></a>
<a class="sourceLine" id="cb1-44" data-line-number="44"><span class="co"># collection &lt;- &quot;selected&quot; # ~60 strains</span></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"><span class="co"># collection &lt;- &quot;around-CoV-2-plus-GISAID&quot; # 16 strains</span></a>
<a class="sourceLine" id="cb1-46" data-line-number="46">collection &lt;-<span class="st"> &quot;selected-plus-GISAID&quot;</span> <span class="co"># ~40 strains</span></a>
<a class="sourceLine" id="cb1-47" data-line-number="47"><span class="co"># collection &lt;- &quot;all-plus-GISAID&quot; # ~60 strains</span></a>
<a class="sourceLine" id="cb1-48" data-line-number="48"></a>
<a class="sourceLine" id="cb1-49" data-line-number="49"><span class="co">## Use a collection-specific path for the figures</span></a>
<a class="sourceLine" id="cb1-50" data-line-number="50">knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(</a>
<a class="sourceLine" id="cb1-51" data-line-number="51">  <span class="dt">fig.path =</span> <span class="kw">paste0</span>(<span class="st">&#39;figures/spike-protein_PIP_&#39;</span>, collection, <span class="st">&#39;/&#39;</span>, collection, <span class="st">&quot;_&quot;</span>))</a>
<a class="sourceLine" id="cb1-52" data-line-number="52"></a>
<a class="sourceLine" id="cb1-53" data-line-number="53"></a>
<a class="sourceLine" id="cb1-54" data-line-number="54"><span class="co">## Genome dir and files</span></a>
<a class="sourceLine" id="cb1-55" data-line-number="55"><span class="cf">if</span> (<span class="kw">length</span>(<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="st">&quot;GISAID&quot;</span>, <span class="dt">x =</span> collection)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) {</a>
<a class="sourceLine" id="cb1-56" data-line-number="56">  useGISAID &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb1-57" data-line-number="57">  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;GISAID_genomes&quot;</span>)</a>
<a class="sourceLine" id="cb1-58" data-line-number="58">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb1-59" data-line-number="59">  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;genomes&quot;</span>)</a>
<a class="sourceLine" id="cb1-60" data-line-number="60">}</a>
<a class="sourceLine" id="cb1-61" data-line-number="61"></a>
<a class="sourceLine" id="cb1-62" data-line-number="62">seqPrefix &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;spike_proteins_&quot;</span>, collection)</a>
<a class="sourceLine" id="cb1-63" data-line-number="63"></a>
<a class="sourceLine" id="cb1-64" data-line-number="64">infiles<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>sequences, <span class="kw">paste0</span>(seqPrefix,<span class="st">&quot;.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb1-65" data-line-number="65"></a>
<a class="sourceLine" id="cb1-66" data-line-number="66"><span class="co">## Sequence sequences</span></a>
<a class="sourceLine" id="cb1-67" data-line-number="67"><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(infiles<span class="op">$</span>sequences)) {</a>
<a class="sourceLine" id="cb1-68" data-line-number="68">  <span class="kw">stop</span>(<span class="st">&quot;¨Protein sequence file is missing&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb1-69" data-line-number="69">}</a>
<a class="sourceLine" id="cb1-70" data-line-number="70"></a>
<a class="sourceLine" id="cb1-71" data-line-number="71"></a>
<a class="sourceLine" id="cb1-72" data-line-number="72"><span class="co">## Load custom functions</span></a>
<a class="sourceLine" id="cb1-73" data-line-number="73"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;align_n_to_one.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-74" data-line-number="74"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;plot_pip_profiles.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-75" data-line-number="75"></a>
<a class="sourceLine" id="cb1-76" data-line-number="76"></a>
<a class="sourceLine" id="cb1-77" data-line-number="77"><span class="co">## Query patterns for SARS-CoV-2</span></a>
<a class="sourceLine" id="cb1-78" data-line-number="78">queryPatterns &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-79" data-line-number="79">queryPatterns[[<span class="st">&quot;HuCoV2_WH01_2019&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-80" data-line-number="80">  <span class="st">&quot;PnGX-P1E_2017&quot;</span>, </a>
<a class="sourceLine" id="cb1-81" data-line-number="81">  <span class="st">&quot;PnGu1_2019&quot;</span>,</a>
<a class="sourceLine" id="cb1-82" data-line-number="82">  <span class="st">&quot;BtRaTG13_2013_Yunnan&quot;</span>, </a>
<a class="sourceLine" id="cb1-83" data-line-number="83">  <span class="st">&quot;BtZC45&quot;</span>,</a>
<a class="sourceLine" id="cb1-84" data-line-number="84">  <span class="st">&quot;BtZXC21&quot;</span>,</a>
<a class="sourceLine" id="cb1-85" data-line-number="85">  <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>,</a>
<a class="sourceLine" id="cb1-86" data-line-number="86">  <span class="st">&quot;CmMERS&quot;</span>,</a>
<a class="sourceLine" id="cb1-87" data-line-number="87">  <span class="st">&quot;HuMERS_172-06_2015&quot;</span></a>
<a class="sourceLine" id="cb1-88" data-line-number="88">)</a>
<a class="sourceLine" id="cb1-89" data-line-number="89"></a>
<a class="sourceLine" id="cb1-90" data-line-number="90">queryPatterns[[<span class="st">&quot;PnGu1_2019&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-91" data-line-number="91">  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span>,</a>
<a class="sourceLine" id="cb1-92" data-line-number="92">  <span class="st">&quot;PnGX-P1E_2017&quot;</span>,</a>
<a class="sourceLine" id="cb1-93" data-line-number="93">  <span class="st">&quot;BtRaTG13_2013_Yunnan&quot;</span></a>
<a class="sourceLine" id="cb1-94" data-line-number="94">)</a>
<a class="sourceLine" id="cb1-95" data-line-number="95"></a>
<a class="sourceLine" id="cb1-96" data-line-number="96">queryPatterns[[<span class="st">&quot;BtRaTG13_2013_Yunnan&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-97" data-line-number="97">  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span>,</a>
<a class="sourceLine" id="cb1-98" data-line-number="98">  <span class="st">&quot;PnGX-P1E_2017&quot;</span>,</a>
<a class="sourceLine" id="cb1-99" data-line-number="99">  <span class="st">&quot;PnGu1_2019&quot;</span></a>
<a class="sourceLine" id="cb1-100" data-line-number="100">)</a>
<a class="sourceLine" id="cb1-101" data-line-number="101"></a>
<a class="sourceLine" id="cb1-102" data-line-number="102">queryPatterns[[<span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb1-103" data-line-number="103">  <span class="st">&quot;CvSZ3&quot;</span>,</a>
<a class="sourceLine" id="cb1-104" data-line-number="104">  <span class="st">&quot;BtRs4874&quot;</span>,</a>
<a class="sourceLine" id="cb1-105" data-line-number="105">  <span class="st">&quot;BtWIV16_2013&quot;</span>,</a>
<a class="sourceLine" id="cb1-106" data-line-number="106">  <span class="st">&quot;BtRs4231&quot;</span>,</a>
<a class="sourceLine" id="cb1-107" data-line-number="107">  <span class="st">&quot;BtRs7327&quot;</span>,</a>
<a class="sourceLine" id="cb1-108" data-line-number="108">  <span class="st">&quot;BtRsSHC014&quot;</span>,</a>
<a class="sourceLine" id="cb1-109" data-line-number="109">  <span class="st">&quot;Btrec-SARSp_2008&quot;</span>,</a>
<a class="sourceLine" id="cb1-110" data-line-number="110">  <span class="st">&quot;PnGX-P1E_2017&quot;</span>,</a>
<a class="sourceLine" id="cb1-111" data-line-number="111">  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span>,</a>
<a class="sourceLine" id="cb1-112" data-line-number="112">  <span class="st">&quot;BtRaTG13_2013_Yunnan&quot;</span>, </a>
<a class="sourceLine" id="cb1-113" data-line-number="113">  <span class="st">&quot;BtZC45&quot;</span>,</a>
<a class="sourceLine" id="cb1-114" data-line-number="114">  <span class="st">&quot;BtZXC21&quot;</span>,</a>
<a class="sourceLine" id="cb1-115" data-line-number="115">  <span class="st">&quot;CmMERS&quot;</span>,</a>
<a class="sourceLine" id="cb1-116" data-line-number="116">  <span class="st">&quot;HuMERS_172-06_2015&quot;</span></a>
<a class="sourceLine" id="cb1-117" data-line-number="117">)</a>
<a class="sourceLine" id="cb1-118" data-line-number="118"></a>
<a class="sourceLine" id="cb1-119" data-line-number="119"><span class="co">#### Add GISAID IDs to the query pattern ####</span></a>
<a class="sourceLine" id="cb1-120" data-line-number="120"><span class="co">## Note that GISAID sequences are be submitted to the github repo because they cannot be redistributed</span></a>
<a class="sourceLine" id="cb1-121" data-line-number="121"><span class="cf">if</span> (useGISAID) {</a>
<a class="sourceLine" id="cb1-122" data-line-number="122">  <span class="cf">for</span> (ref <span class="cf">in</span> <span class="kw">names</span>(queryPatterns))</a>
<a class="sourceLine" id="cb1-123" data-line-number="123">  queryPatterns[[ref]] &lt;-<span class="st"> </span><span class="kw">append</span>(queryPatterns[[ref]], </a>
<a class="sourceLine" id="cb1-124" data-line-number="124">                          <span class="kw">c</span>(<span class="st">&quot;BtYu-RmYN02_2019&quot;</span>,</a>
<a class="sourceLine" id="cb1-125" data-line-number="125">                            <span class="st">&quot;PnGu1_2019&quot;</span></a>
<a class="sourceLine" id="cb1-126" data-line-number="126">                          ))</a>
<a class="sourceLine" id="cb1-127" data-line-number="127">}</a>
<a class="sourceLine" id="cb1-128" data-line-number="128"></a>
<a class="sourceLine" id="cb1-129" data-line-number="129"><span class="co"># message(&quot;\tReference strain: &quot;, refPattern)</span></a>
<a class="sourceLine" id="cb1-130" data-line-number="130"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Query patterns&quot;</span>)</a>
<a class="sourceLine" id="cb1-131" data-line-number="131"><span class="cf">for</span> (ref <span class="cf">in</span> <span class="kw">names</span>(queryPatterns)) {</a>
<a class="sourceLine" id="cb1-132" data-line-number="132">  queryPatterns[[ref]] &lt;-<span class="st"> </span><span class="kw">unique</span>(queryPatterns[[ref]])</a>
<a class="sourceLine" id="cb1-133" data-line-number="133">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, ref, <span class="st">&quot;</span><span class="ch">\t\t</span><span class="st">&quot;</span>, <span class="kw">length</span>(queryPatterns[[ref]]),</a>
<a class="sourceLine" id="cb1-134" data-line-number="134">          <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, queryPatterns[[ref]]))</a>
<a class="sourceLine" id="cb1-135" data-line-number="135">}</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">#### Load sequences ####</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">sequences &lt;-<span class="st"> </span><span class="kw">readAAStringSet</span>(<span class="dt">filepath =</span> infiles<span class="op">$</span>sequences, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">## Shorten sequence names by suppressing the fasta comment (after the space)</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw">names</span>(sequences) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">&quot; .*&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="dt">x =</span> <span class="kw">names</span>(sequences), <span class="dt">perl =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">sequencesNames &lt;-<span class="st"> </span><span class="kw">names</span>(sequences)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">nbsequences &lt;-<span class="st"> </span><span class="kw">length</span>(sequencesNames)</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">message</span>(<span class="st">&quot;Loaded &quot;</span>, nbsequences, <span class="st">&quot; sequences from file &quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co"># View(sequences)</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#### Define reference and query sequences ####</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">refSequenceNames &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="cf">for</span> (ref <span class="cf">in</span> <span class="kw">names</span>(queryPatterns)) {</a>
<a class="sourceLine" id="cb2-15" data-line-number="15">  refSequenceNames[ref] &lt;-<span class="st"> </span><span class="kw">unique</span>(</a>
<a class="sourceLine" id="cb2-16" data-line-number="16">    <span class="kw">grep</span>(<span class="dt">pattern =</span> ref, <span class="dt">x =</span> <span class="kw">names</span>(sequences), </a>
<a class="sourceLine" id="cb2-17" data-line-number="17">         <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb2-18" data-line-number="18">    <span class="cf">if</span> (<span class="kw">is.null</span>(refSequenceNames[ref])) {</a>
<a class="sourceLine" id="cb2-19" data-line-number="19">    <span class="kw">stop</span>(<span class="st">&quot;Could not identify reference sequences with pattern &quot;</span>, ref)</a>
<a class="sourceLine" id="cb2-20" data-line-number="20">  }</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">}</a>
<a class="sourceLine" id="cb2-22" data-line-number="22"></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co">## Query sequences</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co"># querySequenceNames &lt;- list()</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="co"># for (ref in names(queryPatterns)) {</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26"><span class="co"># #  message(&quot;Identifying query sequences for reference: &quot;, ref)</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">#   queryRegExp &lt;- paste0(&quot;(&quot;, paste(collapse = &quot;)|(&quot;, queryPatterns[[ref]]), &quot;)&quot;)</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28"><span class="co">#   querySequenceNames[[ref]] &lt;- grep(pattern = queryRegExp, </span></a>
<a class="sourceLine" id="cb2-29" data-line-number="29"><span class="co">#                              x = sequencesNames, </span></a>
<a class="sourceLine" id="cb2-30" data-line-number="30"><span class="co">#                              ignore.case = TRUE, value = TRUE)</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="co">#   nbquerySeq &lt;- length(querySequenceNames[[ref]])</span></a>
<a class="sourceLine" id="cb2-32" data-line-number="32"><span class="co">#   </span></a>
<a class="sourceLine" id="cb2-33" data-line-number="33"><span class="co">#   if (nbquerySeq == 0) {</span></a>
<a class="sourceLine" id="cb2-34" data-line-number="34"><span class="co">#     stop(&quot;Could not identify any query sequences with query pattern\n&quot;, queryRegExp)</span></a>
<a class="sourceLine" id="cb2-35" data-line-number="35"><span class="co">#   }</span></a>
<a class="sourceLine" id="cb2-36" data-line-number="36"><span class="co">#   </span></a>
<a class="sourceLine" id="cb2-37" data-line-number="37"><span class="co">#   if (length(unlist(queryPatterns[ref[]])) != length(querySequenceNames[[ref]])) {</span></a>
<a class="sourceLine" id="cb2-38" data-line-number="38"><span class="co">#     foundPatterns &lt;- grep(pattern = queryRegExp, x = querySequenceNames[[ref]], value = TRUE)</span></a>
<a class="sourceLine" id="cb2-39" data-line-number="39"><span class="co">#     missingPatterns &lt;- setdiff(foundPatterns, queryPatterns[[ref]])</span></a>
<a class="sourceLine" id="cb2-40" data-line-number="40"><span class="co">#     message(&quot;\t&quot;, </span></a>
<a class="sourceLine" id="cb2-41" data-line-number="41"><span class="co">#             length(missingPatterns), &quot; Missing sequences: &quot;, </span></a>
<a class="sourceLine" id="cb2-42" data-line-number="42"><span class="co">#             paste(collapse = &quot;, &quot;, missingPatterns))</span></a>
<a class="sourceLine" id="cb2-43" data-line-number="43"><span class="co">#   }</span></a>
<a class="sourceLine" id="cb2-44" data-line-number="44"><span class="co">#   </span></a>
<a class="sourceLine" id="cb2-45" data-line-number="45"><span class="co">#   message(&quot;\tFound &quot;, </span></a>
<a class="sourceLine" id="cb2-46" data-line-number="46"><span class="co">#           &quot;\t&quot;, length(querySequenceNames[[ref]]), </span></a>
<a class="sourceLine" id="cb2-47" data-line-number="47"><span class="co">#           &quot; queries for\t&quot;, ref)</span></a>
<a class="sourceLine" id="cb2-48" data-line-number="48"><span class="co"># </span></a>
<a class="sourceLine" id="cb2-49" data-line-number="49"><span class="co">#   ## Compute some statistics about sequences sizes</span></a>
<a class="sourceLine" id="cb2-50" data-line-number="50"><span class="co">#   sequencestat &lt;- data.frame(</span></a>
<a class="sourceLine" id="cb2-51" data-line-number="51"><span class="co">#     row.names = c(ref, querySequenceNames[[ref]]),</span></a>
<a class="sourceLine" id="cb2-52" data-line-number="52"><span class="co">#     status = c(&quot;Reference&quot;, </span></a>
<a class="sourceLine" id="cb2-53" data-line-number="53"><span class="co">#                rep(&quot;Query&quot;, length.out = length(querySequenceNames[[ref]])))</span></a>
<a class="sourceLine" id="cb2-54" data-line-number="54"><span class="co">#   )</span></a>
<a class="sourceLine" id="cb2-55" data-line-number="55"><span class="co">#   </span></a>
<a class="sourceLine" id="cb2-56" data-line-number="56"><span class="co">#   g &lt;- 1</span></a>
<a class="sourceLine" id="cb2-57" data-line-number="57"><span class="co">#   for (g in c(ref, querySequenceNames[[ref]])) {</span></a>
<a class="sourceLine" id="cb2-58" data-line-number="58"><span class="co">#     sequencestat[g, &quot;length&quot;] &lt;- length(sequences[[g]])</span></a>
<a class="sourceLine" id="cb2-59" data-line-number="59"><span class="co">#   }</span></a>
<a class="sourceLine" id="cb2-60" data-line-number="60"><span class="co">#   kable(sequencestat, caption = paste0(&quot;Query sequences for reference &quot;, ref))</span></a>
<a class="sourceLine" id="cb2-61" data-line-number="61"><span class="co"># }</span></a></code></pre></div>
<div id="strain-collection-selected-plus-gisaid" class="section level2">
<h2>Strain collection: selected-plus-GISAID</h2>
<p>The collection selected-plus-GISAID contains 38 virus sequences sequences.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">## Report the number of strains</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">strainNames &lt;-<span class="st"> </span><span class="kw">names</span>(sequences)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">nbStrains &lt;-<span class="st"> </span><span class="kw">length</span>(strainNames)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Loaded &quot;</span>, nbStrains, <span class="st">&quot; sequences from file &quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co"># View(genomes)</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#### Compute sequence sizes ####</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">strainStats &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">  <span class="dt">n =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(sequences),</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">  <span class="dt">row.names =</span> <span class="kw">names</span>(sequences),</a>
<a class="sourceLine" id="cb3-11" data-line-number="11">  <span class="dt">status =</span> <span class="kw">rep</span>(<span class="st">&quot;Query&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(strainNames))</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13">strainStats[,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> </span><span class="kw">as.vector</span>(strainStats[,<span class="st">&quot;status&quot;</span>])</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">strainStats[refSequenceNames,<span class="st">&quot;status&quot;</span>] &lt;-<span class="st"> &quot;Reference&quot;</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15">g &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="cf">for</span> (g <span class="cf">in</span> strainNames) {</a>
<a class="sourceLine" id="cb3-17" data-line-number="17">  strainStats[g, <span class="st">&quot;length&quot;</span>] &lt;-<span class="st"> </span><span class="kw">length</span>(sequences[[g]])</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">}</a>
<a class="sourceLine" id="cb3-19" data-line-number="19"></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#### Define the color associated to each sequence ####</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">## Color palette per species</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23">speciesPalette &lt;-<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb3-24" data-line-number="24">  <span class="dt">Human =</span> <span class="st">&quot;#880000&quot;</span>,</a>
<a class="sourceLine" id="cb3-25" data-line-number="25">  <span class="dt">Bat =</span> <span class="st">&quot;#888888&quot;</span>,</a>
<a class="sourceLine" id="cb3-26" data-line-number="26">  <span class="dt">Pangolin =</span> <span class="st">&quot;#448800&quot;</span>,</a>
<a class="sourceLine" id="cb3-27" data-line-number="27">  <span class="dt">Camel =</span> <span class="st">&quot;#BB8800&quot;</span>,</a>
<a class="sourceLine" id="cb3-28" data-line-number="28">  <span class="dt">Pig =</span> <span class="st">&quot;#FFBBBB&quot;</span>,</a>
<a class="sourceLine" id="cb3-29" data-line-number="29">  <span class="dt">Civet =</span> <span class="st">&quot;#00BBFF&quot;</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30">)</a>
<a class="sourceLine" id="cb3-31" data-line-number="31"></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">## Species prefix in the tip labels</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33">speciesPrefix &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Hu&quot;</span> =<span class="st"> &quot;Human&quot;</span>,</a>
<a class="sourceLine" id="cb3-34" data-line-number="34">                   <span class="st">&quot;Bt&quot;</span> =<span class="st"> &quot;Bat&quot;</span>,</a>
<a class="sourceLine" id="cb3-35" data-line-number="35">                   <span class="st">&quot;Pn&quot;</span> =<span class="st"> &quot;Pangolin&quot;</span>,</a>
<a class="sourceLine" id="cb3-36" data-line-number="36">                   <span class="st">&quot;Cm&quot;</span> =<span class="st"> &quot;Camel&quot;</span>,</a>
<a class="sourceLine" id="cb3-37" data-line-number="37">                   <span class="st">&quot;Pi&quot;</span> =<span class="st"> &quot;Pig&quot;</span>,</a>
<a class="sourceLine" id="cb3-38" data-line-number="38">                   <span class="st">&quot;Cv&quot;</span> =<span class="st"> &quot;Civet&quot;</span>)</a>
<a class="sourceLine" id="cb3-39" data-line-number="39"></a>
<a class="sourceLine" id="cb3-40" data-line-number="40"><span class="co">## Strain-specific colors</span></a>
<a class="sourceLine" id="cb3-41" data-line-number="41">strainColor &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb3-42" data-line-number="42">  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span> =<span class="st"> &quot;red&quot;</span>,</a>
<a class="sourceLine" id="cb3-43" data-line-number="43">  <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span> =<span class="st"> &quot;#0044BB&quot;</span>,</a>
<a class="sourceLine" id="cb3-44" data-line-number="44">  <span class="st">&quot;BtRs4874&quot;</span> =<span class="st"> &quot;#BB00BB&quot;</span>,</a>
<a class="sourceLine" id="cb3-45" data-line-number="45">  <span class="st">&quot;PnGu1_2019&quot;</span> =<span class="st"> &quot;#00BB00&quot;</span>,</a>
<a class="sourceLine" id="cb3-46" data-line-number="46">  <span class="st">&quot;BtRaTG13_&quot;</span> =<span class="st"> &quot;#FF6600&quot;</span>,</a>
<a class="sourceLine" id="cb3-47" data-line-number="47">  <span class="st">&quot;BtYu-RmYN&quot;</span> =<span class="st"> &quot;#FFBB22&quot;</span>,</a>
<a class="sourceLine" id="cb3-48" data-line-number="48">  <span class="st">&quot;BtZXC21&quot;</span> =<span class="st"> &quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb3-49" data-line-number="49">  <span class="st">&quot;BtZC45&quot;</span> =<span class="st"> &quot;black&quot;</span>)</a>
<a class="sourceLine" id="cb3-50" data-line-number="50"></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="co">## Identify species per tip</span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="cf">for</span> (prefix <span class="cf">in</span> <span class="kw">names</span>(speciesPrefix)) {</a>
<a class="sourceLine" id="cb3-53" data-line-number="53">  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, prefix), <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;species&quot;</span>] &lt;-<span class="st"> </span>speciesPrefix[prefix]</a>
<a class="sourceLine" id="cb3-54" data-line-number="54">  </a>
<a class="sourceLine" id="cb3-55" data-line-number="55">}</a>
<a class="sourceLine" id="cb3-56" data-line-number="56"></a>
<a class="sourceLine" id="cb3-57" data-line-number="57"><span class="co">## Assign acolor to each species</span></a>
<a class="sourceLine" id="cb3-58" data-line-number="58">strainStats<span class="op">$</span>color &lt;-<span class="st"> &quot;grey&quot;</span> <span class="co"># default</span></a>
<a class="sourceLine" id="cb3-59" data-line-number="59">strainStats<span class="op">$</span>color &lt;-<span class="st"> </span>speciesPalette[<span class="kw">as.vector</span>(strainStats<span class="op">$</span>species)]</a>
<a class="sourceLine" id="cb3-60" data-line-number="60"></a>
<a class="sourceLine" id="cb3-61" data-line-number="61"><span class="cf">for</span> (strain <span class="cf">in</span> <span class="kw">names</span>(strainColor)) {</a>
<a class="sourceLine" id="cb3-62" data-line-number="62">  strainStats[<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="kw">paste0</span>(<span class="st">&quot;^&quot;</span>, strain), </a>
<a class="sourceLine" id="cb3-63" data-line-number="63">                  <span class="dt">x =</span> <span class="kw">row.names</span>(strainStats), <span class="dt">perl =</span> <span class="ot">TRUE</span>), <span class="st">&quot;color&quot;</span>] &lt;-<span class="st"> </span>strainColor[strain]</a>
<a class="sourceLine" id="cb3-64" data-line-number="64">  </a>
<a class="sourceLine" id="cb3-65" data-line-number="65">}</a>
<a class="sourceLine" id="cb3-66" data-line-number="66"></a>
<a class="sourceLine" id="cb3-67" data-line-number="67"></a>
<a class="sourceLine" id="cb3-68" data-line-number="68"><span class="co">## Assign specific color to some nodes</span></a>
<a class="sourceLine" id="cb3-69" data-line-number="69"></a>
<a class="sourceLine" id="cb3-70" data-line-number="70"><span class="co">## Define a color for each strain</span></a>
<a class="sourceLine" id="cb3-71" data-line-number="71">strainColors &lt;-<span class="st"> </span>(<span class="kw">unlist</span>(strainStats<span class="op">$</span>color))</a>
<a class="sourceLine" id="cb3-72" data-line-number="72"><span class="kw">names</span>(strainColors) &lt;-<span class="st"> </span><span class="kw">row.names</span>(strainStats)</a></code></pre></div>
</div>
<div id="n-to-1-full-sequence-alignments" class="section level2">
<h2>N-to-1 full sequence alignments</h2>
<p>We perform a pairwise lignment between each sequences query and the reference sequences (HuCoV2_WH01_2019).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">sequencesNto1 &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">refSequenceName &lt;-<span class="st"> &quot;HuCoV2_WH01_2019&quot;</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="cf">for</span> (refSequenceName <span class="cf">in</span> refSequenceNames) {</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">  <span class="co">## Define output file for sequences alignments</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5">  outfile &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">    dir<span class="op">$</span>outseq, <span class="kw">paste0</span>(<span class="st">&quot;one-to-n_alignments_ref_&quot;</span>, refSequenceName))</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">  outfiles[<span class="kw">paste0</span>(refSequenceName, <span class="st">&quot; alignments&quot;</span>)] &lt;-<span class="st"> </span>outfile</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">  </a>
<a class="sourceLine" id="cb4-9" data-line-number="9">  <span class="co">## Get sequences for reference and query sequences</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10">  refSequence &lt;-<span class="st"> </span>sequences[refSequenceName]</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">  <span class="co"># querySeq &lt;- sequences[querySequenceNames[[refSequenceName]]]</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12">  </a>
<a class="sourceLine" id="cb4-13" data-line-number="13">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Aligning &quot;</span>, <span class="kw">length</span>(sequences), <span class="st">&quot; sequences&quot;</span>,</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">          <span class="st">&quot; to reference</span><span class="ch">\t</span><span class="st">&quot;</span>, refSequenceName)</a>
<a class="sourceLine" id="cb4-15" data-line-number="15"></a>
<a class="sourceLine" id="cb4-16" data-line-number="16">  sequencesNto1[[refSequenceName]] &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb4-17" data-line-number="17">    <span class="dt">refSequence =</span> refSequence,  </a>
<a class="sourceLine" id="cb4-18" data-line-number="18">    <span class="dt">querySequences =</span> sequences, </a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#    querySequences = querySeq, </span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20">    <span class="dt">sortByPIP =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb4-21" data-line-number="21">    <span class="co">#  querySequences = sequences, </span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22">    <span class="dt">outfile =</span> outfile)</a>
<a class="sourceLine" id="cb4-23" data-line-number="23">  </a>
<a class="sourceLine" id="cb4-24" data-line-number="24">}</a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">refSequenceName &lt;-<span class="st"> &quot;HuCoV2_WH01_2019&quot;</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="cf">for</span> (refSequenceName <span class="cf">in</span> refSequenceNames) {</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">  <span class="kw">cat</span>(<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">## Spike proteins: &quot;</span>, collection, <span class="st">&quot; vs reference &quot;</span>, refSequenceName)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">  </a>
<a class="sourceLine" id="cb5-6" data-line-number="6">  knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">    <span class="dt">fig.path =</span> <span class="kw">paste0</span>(<span class="st">&#39;figures/spike-protein_PIP/&#39;</span>, collection, <span class="st">&#39;_vs_&#39;</span>, refSequenceName))</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"></a>
<a class="sourceLine" id="cb5-9" data-line-number="9">  <span class="kw">kable</span>(sequencesNto1[[refSequenceName]]<span class="op">$</span>stats[<span class="kw">order</span>(sequencesNto1[[refSequenceName]]<span class="op">$</span>stats<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), ], </a>
<a class="sourceLine" id="cb5-10" data-line-number="10">        <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of full sequences&quot;</span>)</a>
<a class="sourceLine" id="cb5-11" data-line-number="11">  </a>
<a class="sourceLine" id="cb5-12" data-line-number="12">  </a>
<a class="sourceLine" id="cb5-13" data-line-number="13">  <span class="co">## PIP profile of full sequences N-to-1 alignments</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14">   <span class="cf">if</span> (<span class="kw">length</span>(<span class="kw">grep</span>(<span class="dt">pattern =</span> <span class="st">&quot;around-CoV-2&quot;</span>, <span class="dt">x =</span> collection))) {</a>
<a class="sourceLine" id="cb5-15" data-line-number="15">     minPIP &lt;-<span class="st"> </span><span class="dv">30</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16">     legendCex &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17">     legendCorner &lt;-<span class="st"> &quot;bottomright&quot;</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18">     legendMargin &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19">   } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb5-20" data-line-number="20">     minPIP &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21">     legendCex &lt;-<span class="st"> </span><span class="fl">0.65</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22">     legendCorner &lt;-<span class="st"> &quot;topright&quot;</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23">     legendMargin &lt;-<span class="st"> </span><span class="fl">0.25</span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24">   }</a>
<a class="sourceLine" id="cb5-25" data-line-number="25">   </a>
<a class="sourceLine" id="cb5-26" data-line-number="26">  <span class="kw">plotPIPprofiles</span>(</a>
<a class="sourceLine" id="cb5-27" data-line-number="27">    <span class="dt">alignments =</span> sequencesNto1[[refSequenceName]]<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb5-28" data-line-number="28">    <span class="dt">windowSize =</span> <span class="dv">100</span>, </a>
<a class="sourceLine" id="cb5-29" data-line-number="29">    <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Spike protein: collection &quot;</span>, collection, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">vs strain &quot;</span>, refSequenceName),</a>
<a class="sourceLine" id="cb5-30" data-line-number="30">    <span class="dt">colors =</span> strainColors,</a>
<a class="sourceLine" id="cb5-31" data-line-number="31">    <span class="dt">legendMargin =</span> legendMargin, </a>
<a class="sourceLine" id="cb5-32" data-line-number="32">    <span class="dt">legendCorner =</span> legendCorner, </a>
<a class="sourceLine" id="cb5-33" data-line-number="33">    <span class="dt">lwd =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb5-34" data-line-number="34">    <span class="dt">legendCex =</span> legendCex, </a>
<a class="sourceLine" id="cb5-35" data-line-number="35">    <span class="dt">ylim =</span> <span class="kw">c</span>(minPIP,<span class="dv">100</span>))</a>
<a class="sourceLine" id="cb5-36" data-line-number="36">  <span class="co"># legendMargin = 0.25, </span></a>
<a class="sourceLine" id="cb5-37" data-line-number="37">  <span class="co"># legendCorner = &quot;topright&quot;, lwd = 1,</span></a>
<a class="sourceLine" id="cb5-38" data-line-number="38">  <span class="co"># legendCex = 0.5, ylim = c(30, 100))</span></a>
<a class="sourceLine" id="cb5-39" data-line-number="39">}</a></code></pre></div>
## Spike proteins: selected-plus-GISAID vs reference HuCoV2_WH01_2019
<div class="figure" style="text-align: center">
<img src="figures/spike-protein_PIP_selected-plus-GISAID/selected-plus-GISAID_spike-protein_PIP-1.png" alt="Percent Identical Positions (PIP) profiles of spike protein sequences. " width="100%" />
<p class="caption">
Percent Identical Positions (PIP) profiles of spike protein sequences.
</p>
</div>
## Spike proteins: selected-plus-GISAID vs reference PnGu1_2019
<div class="figure" style="text-align: center">
<img src="figures/spike-protein_PIP_selected-plus-GISAID/selected-plus-GISAID_spike-protein_PIP-2.png" alt="Percent Identical Positions (PIP) profiles of spike protein sequences. " width="100%" />
<p class="caption">
Percent Identical Positions (PIP) profiles of spike protein sequences.
</p>
</div>
## Spike proteins: selected-plus-GISAID vs reference BtRaTG13_2013_Yunnan
<div class="figure" style="text-align: center">
<img src="figures/spike-protein_PIP_selected-plus-GISAID/selected-plus-GISAID_spike-protein_PIP-3.png" alt="Percent Identical Positions (PIP) profiles of spike protein sequences. " width="100%" />
<p class="caption">
Percent Identical Positions (PIP) profiles of spike protein sequences.
</p>
</div>
## Spike proteins: selected-plus-GISAID vs reference HuSARS-Frankfurt-1_2003
<div class="figure" style="text-align: center">
<img src="figures/spike-protein_PIP_selected-plus-GISAID/selected-plus-GISAID_spike-protein_PIP-4.png" alt="Percent Identical Positions (PIP) profiles of spike protein sequences. " width="100%" />
<p class="caption">
Percent Identical Positions (PIP) profiles of spike protein sequences.
</p>
</div>
<!-- ## Aligning all sequences -->
<!-- ```{r SARS_vs_all} -->
<!-- ## Define output file for sequences alignments -->
<!-- outfiles["Sequence alignments - SARS"] <- file.path( -->
<!--   dir$outseq, paste0("spike-protein_alignments_ref_", -->
<!--                      "HuSARS-Frankfurt-1_2003")) -->
<!-- #### Compare SARS (2002) wih the related Civet sequences #### -->
<!-- SARSvsAll <- alignNtoOne( -->
<!--   refSequence = sequences["HuSARS-Frankfurt-1_2003"],  -->
<!--   querySeq = sequences, sortByPIP = TRUE, -->
<!--   outfile = outfiles["Sequence alignments - SARS"] ) -->
<!-- kable(SARSvsAll$stats,  -->
<!--       caption = "SARS (2003). N-to-one alignment of full sequences of the closest animal virus (Civet) and of Human SARS-CoV-2. ") -->
<!-- ``` -->
<!-- ```{r PIP_SARS_vs_all, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole sequences of SARS (2002-2003). "} -->
<!-- ## PIP profile of full sequences N-to-1 alignments -->
<!-- plotPIPprofiles(alignments = SARSvsAll$alignments,  -->
<!--                 windowSize = 500,  -->
<!--                 legend = paste0(names(SARSvsAll$alignments), " (", round(digits = 2, SARSvsAll$stats$pid), "%)"), -->
<!--                 main = paste0("Spike protein PIP", "\nRef: ", "Human_SARS-CoV_Frankfurt_1"), -->
<!--                 legendMargin = 0.3,  -->
<!--                 legendCorner = "bottomright",  -->
<!--                 legendCex = 0.5,  -->
<!--                 ylim = c(0,100)) -->
<!-- ``` -->
<!-- ## Spike protein against 2003 SARS-CoV (Frankfurt strain) -->
<!-- ```{r S-gene_align_queries} -->
<!-- featureName <- "Spike-protein_SARS-CoV" -->
<!-- featureLimits <- features[[featureName]] -->
<!-- #### N-to-1 alignments of spike-coding sequences #### -->
<!-- dir[[featureName]] <- file.path(dir$main, "results", featureName) -->
<!-- dir.create(featureName, showWarnings = FALSE, recursive = TRUE) -->
<!-- outfiles[featureName]  <- file.path( -->
<!--   dir[[featureName]], paste0(featureName, "_", collection, "_alignments")) -->
<!-- ## Get sequences for reference and query sequences -->
<!-- refSeq <- subseq(sequences[refSequenceName], start = featureLimits[1], end = featureLimits[2]) -->
<!-- featureAlignmentsNto1 <- alignNtoOne( -->
<!--   refSequence = refSeq,  -->
<!--   querySeq = querySeq,  -->
<!--   type = "global-local", sortByPIP = TRUE, -->
<!--   outfile = outfiles[featureName]) -->
<!-- kable(featureAlignmentsNto1$stats[order(featureAlignmentsNto1$stat$score, decreasing = TRUE),],  -->
<!--       caption = "N-to-one alignment of S genes") -->
<!-- ``` -->
<!-- ### Spike gene PIP plot -->
<!-- ```{r Sgene_pip, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole sequences of SARS-CoV-2. "} -->
<!-- ## PIP profile of spike N-to-1 alignments -->
<!-- plotPIPprofiles(alignments = featureAlignmentsNto1$alignments,  -->
<!--                 windowSize = 200,  -->
<!-- #                legend = paste0(names(featureAlignmentsNto1$alignments), " (", round(digits = 2, featureAlignmentsNto1$stats$pid), "%)"), -->
<!--                 main = paste0(featureName, " - PIP profile", "\nRef: ", refSequenceName), -->
<!--                 legendMargin = 0, legendCorner = "bottomright", legendCex = 0.8, ylim = c(30, 100)) -->
<!-- ``` -->
<!-- ### Getting sequences of spike gene in all the selected coronavirus sequences -->
<!-- ```{r  match sequence fragment against all sequences and export the matches to a fasta file} -->
<!-- #' @title Export sequencesquery + subject sequences from N-to-1 alignments -->
<!-- #' @author Jacques.van-Helden@france)bioinformatique.fr -->
<!-- #' @param refSequence  sequence used as reference (DB) for the N-to-1 alignment -->
<!-- #' @param sequences sequences sequences -->
<!-- #' @param IDsuffix=NULL if not null, append the suffix to the sequence name for the fasta header -->
<!-- #' @param outfile=NULL output file. If specified, the matches will be exported as fasta-formatted sequences. -->
<!-- #' @export -->
<!-- ExportSegmentMatches <- function(refSequence, -->
<!--                                  sequences, -->
<!--                                  IDsuffix = NULL, -->
<!--                                  outfile=NULL) { -->
<!--   #### N-to-1 alignments of all spike-coding sequences #### -->
<!--   ## Get sequences for reference and query sequences -->
<!--   alignmentsNto1 <- alignNtoOne( -->
<!--     refSequence = refSequence,  -->
<!--     querySeq = sequences,  -->
<!--     type = "global-local") -->
<!--   if (!is.null(outfile)) { -->
<!--     ## Write the reference sequence in the output fle -->
<!--     writeXStringSet(refSequence, filepath = outfile, format = "fasta") -->
<!--     i <- 1 -->
<!--     nbAlignments <- length(alignmentsNto1$alignments) -->
<!--     for (i in 1:nbAlignments) { -->
<!--       sequencesName <- names(sequences[i]) -->
<!--       alignment <- alignmentsNto1$alignments[[i]] -->
<!--       subject <- subject(alignment) -->
<!--       ## Suppress the dashes from the alignment to get the raw sequence -->
<!--       sequence <- as.character(subject) -->
<!--       sequenceDesaligned <- gsub(pattern = "-", replacement = "", x = sequence) -->
<!--       seqStringSet <- DNAStringSet(x = sequenceDesaligned) #, start = start(subject), end=end(subject)) -->
<!--       ## Define a sequence ID for the fasta header -->
<!--       sequenceID <- sequencesName -->
<!--       if (!is.null(IDsuffix)) { -->
<!--         sequenceID <- paste0(sequenceID, IDsuffix) -->
<!--       }  -->
<!--       sequenceID <- paste0(sequenceID, "_", start(subject), "-", end(subject)) -->
<!--       names(seqStringSet) <- sequenceID -->
<!--       #  -->
<!--       ## Write pairwise alignment (temporarily disaactivated) -->
<!--       # alignmentFile <- paste0("pairwise-alignment_",  -->
<!--       #                         # gsub(pattern = "/", replacement = "-", x = sequencesName),  -->
<!--       #                         ".txt") -->
<!--       # writePairwiseAlignments(x = alignment, file = outfile) -->
<!--       ## Append the sequence to the file -->
<!--       message("\tAppending sequence ", i, "/", nbAlignments, "\t", sequenceID) -->
<!--       writeXStringSet(seqStringSet, -->
<!--                       filepath = outfile, format = "fasta", append = TRUE) -->
<!--     } -->
<!--     message("\tExported alignments to\t", outfile) -->
<!--   }  -->
<!--   return(alignmentsNto1) -->
<!-- } -->
<!-- ``` -->
<!-- ```{r Sgene_exportseq} -->
<!-- #### Export S gene sequences found in the reference sequences #### -->
<!-- # toString(refSeq) -->
<!-- writeXStringSet( -->
<!--   refSeq,  -->
<!--   filepath = file.path( -->
<!--     dir$outseq,  -->
<!--     paste0("S-gene_", refSequenceName, ".fasta")),  -->
<!--   format = "fasta") -->
<!-- ## Export matches to S-gene -->
<!-- outfiles["S gene matches"] <- file.path( -->
<!--   dir$outseq,  -->
<!--   paste0("S-gene_", refSequenceName, "_matches.fasta")) -->
<!-- featureAlignmentsNto1All <- ExportSegmentMatches( -->
<!--   refSequence = refSeq, -->
<!--   sequences = sequences,  -->
<!--   IDsuffix = "_S-gene", -->
<!--   outfile = outfiles["S gene matches"]) -->
<!-- kable(featureAlignmentsNto1All$stats[order(featureAlignmentsNto1All$stat$score, decreasing = TRUE),],  -->
<!--         caption = "N-to-one alignment of S genes") -->
<!-- ``` -->
<!-- ## Export matches to the coding sequence of the S1 cleavage produc -->
<!-- ## N-to-1 alignemnts of S1 coding sequences -->
<!-- ```{r S1_align_queries} -->
<!-- refS1Limits <- refSpikeLimits[1] + 3*c(13, 685) -->
<!-- #### N-to-1 alignments of S1-coding sequences #### -->
<!-- ## Get sequences for reference and query sequences -->
<!-- refS1 <- subseq(sequences[refSequenceName], start = refS1Limits[1], end = refS1Limits[2]) -->
<!-- S1Nto1 <- alignNtoOne(refSequence = refS1, querySeq = querySeq, type = "global-local") -->
<!-- kable(S1Nto1$stats[order(S1Nto1$stat$score, decreasing = TRUE),],  -->
<!--       caption = "N-to-one alignment of S1 coding sequence") -->
<!-- ``` -->
<!-- ### S1 gene PIP plot -->
<!-- ```{r S1_pip, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole sequences of SARS-CoV-2. "} -->
<!-- ## PIP profile of S1 N-to-1 alignments -->
<!-- plotPIPprofiles(alignments = S1Nto1$alignments,  -->
<!--                 windowSize = 200,  -->
<!--                 legend = paste0(names(S1Nto1$alignments), " (", round(digits = 2, S1Nto1$stats$pid), "%)"), -->
<!--                 main = paste0("S1 coding sequence - PIP profile", "\nRef: ", refSequenceName), -->
<!--                 legendMargin = 0, legendCorner = "bottomleft", legendCex = 0.8, ylim = c(30, 100)) -->
<!-- ``` -->
</div>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">kable</span>(<span class="kw">t</span>(<span class="kw">as.data.frame</span>(dir)), <span class="dt">col.names =</span> <span class="st">&quot;Dir&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Directories&quot;</span>)</a></code></pre></div>
<table>
<caption>Directories</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">Dir</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>main</td>
<td align="left">..</td>
</tr>
<tr class="even">
<td>R</td>
<td align="left">../scripts/R</td>
</tr>
<tr class="odd">
<td>outseq</td>
<td align="left">../results/spike_proteins/PIP_profiles</td>
</tr>
<tr class="even">
<td>sequences</td>
<td align="left">../data/GISAID_genomes</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">kable</span>(outfiles, <span class="dt">col.names =</span> <span class="st">&quot;File&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Output files&quot;</span>)</a></code></pre></div>
<table>
<caption>Output files</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">File</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HuCoV2_WH01_2019 alignments</td>
<td align="left">../results/spike_proteins/PIP_profiles/one-to-n_alignments_ref_HuCoV2_WH01_2019</td>
</tr>
<tr class="even">
<td>PnGu1_2019 alignments</td>
<td align="left">../results/spike_proteins/PIP_profiles/one-to-n_alignments_ref_PnGu1_2019</td>
</tr>
<tr class="odd">
<td>BtRaTG13_2013_Yunnan alignments</td>
<td align="left">../results/spike_proteins/PIP_profiles/one-to-n_alignments_ref_BtRaTG13_2013_Yunnan</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003 alignments</td>
<td align="left">../results/spike_proteins/PIP_profiles/one-to-n_alignments_ref_HuSARS-Frankfurt-1_2003</td>
</tr>
</tbody>
</table>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Biostrings_2.52.0   XVector_0.24.0      IRanges_2.18.3      S4Vectors_0.22.1    BiocGenerics_0.30.0 knitr_1.28         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4          digest_0.6.25       magrittr_1.5        evaluate_0.14       highr_0.8           zlibbioc_1.30.0     rlang_0.4.5         stringi_1.4.6       rmarkdown_2.1       tools_3.6.1         stringr_1.4.0       xfun_0.12           yaml_2.2.1          compiler_3.6.1     
[15] BiocManager_1.30.10 htmltools_0.4.0    </code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
