<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-05-22" />

<title>PIP profiles of coronavirus spike proteins - DNA and protein sequences</title>

<script src="spike_PIP_RNA-vs-proteins_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="spike_PIP_RNA-vs-proteins_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="spike_PIP_RNA-vs-proteins_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="spike_PIP_RNA-vs-proteins_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="spike_PIP_RNA-vs-proteins_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="spike_PIP_RNA-vs-proteins_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="spike_PIP_RNA-vs-proteins_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="spike_PIP_RNA-vs-proteins_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="spike_PIP_RNA-vs-proteins_files/navigation-1.1/tabsets.js"></script>
<script src="spike_PIP_RNA-vs-proteins_files/navigation-1.1/codefolding.js"></script>
<link href="spike_PIP_RNA-vs-proteins_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="spike_PIP_RNA-vs-proteins_files/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">PIP profiles of coronavirus spike proteins - DNA and protein sequences</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-05-22</h4>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#### General parameters for the analysis ####</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">## Use (or not) GIDAID sequences</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">## </span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">## A few genoes were not available in NCBI Genbank at the time of </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">## this analysis, and had to be downloaded from GISAID. These sequences</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">##  can however not be redistributed, they should thus be downloaded </span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">##  manually to reproduce the full trees. Alternatively, useGISAID</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">##  can be set to FALSE, whcih will reproduce the analysis with almost </span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">##  all the sequences of the paper.</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11">useGISAID &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">#### Define directories and files ####</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15">dir &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">main =</span> <span class="st">&#39;..&#39;</span>)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">dir<span class="op">$</span>R &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;scripts/R&quot;</span>)</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">#### Create output directory for sequences ####</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19">seqPrefix &lt;-<span class="st"> &quot;spike_proteins&quot;</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20">dir<span class="op">$</span>outseq &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb1-21" data-line-number="21">  dir<span class="op">$</span>main, <span class="st">&quot;results&quot;</span>, seqPrefix, <span class="st">&quot;PIP_profiles&quot;</span>)</a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="kw">dir.create</span>(dir<span class="op">$</span>outseq, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-23" data-line-number="23"><span class="kw">list.files</span>(dir<span class="op">$</span>outseq)</a></code></pre></div>
<pre><code>[1] &quot;Icon\r&quot;                                               &quot;one-to-n_alignments_ref_HuCoV2_WH01_2019&quot;             &quot;one-to-n_alignments_ref_HuSARS-Frankfurt-1_2003&quot;      &quot;spike-protein_alignments_ref_HuSARS-Frankfurt-1_2003&quot;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">## Instantiate a list for output files</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">outfiles &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">## Input files</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5">infiles &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">## Sequence collection</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">## Supported: </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">## collection &lt;- &quot;around-CoV-2&quot; # ~20 sequences</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">## collection &lt;- &quot;selected&quot; # ~60 sequences</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">## collection &lt;- &quot;all&quot; # ~60 sequences</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12">collections &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;around-CoV-2&quot;</span>, <span class="st">&quot;selected&quot;</span>, <span class="st">&quot;all&quot;</span>)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co"># collection &lt;- &quot;around-CoV-2&quot; # ~20 sequences</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co"># collection &lt;- &quot;selected&quot; # ~40 sequences</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15">collection &lt;-<span class="st"> &quot;all&quot;</span> <span class="co"># ~60 sequences</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">## Sequence dir and files</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="cf">if</span> (useGISAID) {</a>
<a class="sourceLine" id="cb3-20" data-line-number="20">  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;GISAID_genomes&quot;</span>)</a>
<a class="sourceLine" id="cb3-21" data-line-number="21">  collection &lt;-<span class="st"> </span><span class="kw">paste0</span>(collection, <span class="st">&quot;-plus-GISAID&quot;</span>)</a>
<a class="sourceLine" id="cb3-22" data-line-number="22">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">  dir<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>main, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;spike_proteins&quot;</span>)</a>
<a class="sourceLine" id="cb3-24" data-line-number="24">}</a>
<a class="sourceLine" id="cb3-25" data-line-number="25"></a>
<a class="sourceLine" id="cb3-26" data-line-number="26">seqPrefix &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;spike_proteins_&quot;</span>, collection)</a>
<a class="sourceLine" id="cb3-27" data-line-number="27"></a>
<a class="sourceLine" id="cb3-28" data-line-number="28">infiles<span class="op">$</span>sequences &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir<span class="op">$</span>sequences, <span class="kw">paste0</span>(seqPrefix,<span class="st">&quot;.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb3-29" data-line-number="29"></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">## Sequence sequences</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(infiles<span class="op">$</span>sequences)) {</a>
<a class="sourceLine" id="cb3-32" data-line-number="32">  <span class="kw">stop</span>(<span class="st">&quot;¨Protein sequence file is missing&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb3-33" data-line-number="33">}</a>
<a class="sourceLine" id="cb3-34" data-line-number="34"></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"><span class="co">## Load custom functions</span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;align_n_to_one.R&quot;</span>))</a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="kw">source</span>(<span class="kw">file.path</span>(dir<span class="op">$</span>R, <span class="st">&quot;plot_pip_profiles.R&quot;</span>))</a>
<a class="sourceLine" id="cb3-39" data-line-number="39"></a>
<a class="sourceLine" id="cb3-40" data-line-number="40"></a>
<a class="sourceLine" id="cb3-41" data-line-number="41"><span class="co">## Query patterns for SARS-CoV-2</span></a>
<a class="sourceLine" id="cb3-42" data-line-number="42">queryPatterns &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb3-43" data-line-number="43">queryPatterns[[<span class="st">&quot;HuCoV2_WH01_2019&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb3-44" data-line-number="44">  <span class="st">&quot;PnGX-P1E_2017&quot;</span>, </a>
<a class="sourceLine" id="cb3-45" data-line-number="45">  <span class="st">&quot;PnGu1_2019&quot;</span>,</a>
<a class="sourceLine" id="cb3-46" data-line-number="46">  <span class="st">&quot;BtRaTG13_2013_Yunnan&quot;</span>, </a>
<a class="sourceLine" id="cb3-47" data-line-number="47">  <span class="st">&quot;BtZC45&quot;</span>,</a>
<a class="sourceLine" id="cb3-48" data-line-number="48">  <span class="st">&quot;BtZXC21&quot;</span>,</a>
<a class="sourceLine" id="cb3-49" data-line-number="49">  <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>,</a>
<a class="sourceLine" id="cb3-50" data-line-number="50">  <span class="st">&quot;CmMERS&quot;</span>,</a>
<a class="sourceLine" id="cb3-51" data-line-number="51">  <span class="st">&quot;HuMERS_172-06_2015&quot;</span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52">)</a>
<a class="sourceLine" id="cb3-53" data-line-number="53"></a>
<a class="sourceLine" id="cb3-54" data-line-number="54">queryPatterns[[<span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb3-55" data-line-number="55">  <span class="st">&quot;CvSZ3&quot;</span>,</a>
<a class="sourceLine" id="cb3-56" data-line-number="56">  <span class="st">&quot;BtRs4874&quot;</span>,</a>
<a class="sourceLine" id="cb3-57" data-line-number="57">  <span class="st">&quot;BtWIV16_2013&quot;</span>,</a>
<a class="sourceLine" id="cb3-58" data-line-number="58">  <span class="st">&quot;BtRs4231&quot;</span>,</a>
<a class="sourceLine" id="cb3-59" data-line-number="59">  <span class="st">&quot;BtRs7327&quot;</span>,</a>
<a class="sourceLine" id="cb3-60" data-line-number="60">  <span class="st">&quot;BtRsSHC014&quot;</span>,</a>
<a class="sourceLine" id="cb3-61" data-line-number="61">  <span class="st">&quot;Btrec-SARSp_2008&quot;</span>,</a>
<a class="sourceLine" id="cb3-62" data-line-number="62">  <span class="st">&quot;PnGX-P1E_2017&quot;</span>,</a>
<a class="sourceLine" id="cb3-63" data-line-number="63">  <span class="st">&quot;HuCoV2_WH01_2019&quot;</span>,</a>
<a class="sourceLine" id="cb3-64" data-line-number="64">  <span class="st">&quot;BtRaTG13_2013_Yunnan&quot;</span>, </a>
<a class="sourceLine" id="cb3-65" data-line-number="65">  <span class="st">&quot;BtZC45&quot;</span>,</a>
<a class="sourceLine" id="cb3-66" data-line-number="66">  <span class="st">&quot;BtZXC21&quot;</span>,</a>
<a class="sourceLine" id="cb3-67" data-line-number="67">  <span class="st">&quot;CmMERS&quot;</span>,</a>
<a class="sourceLine" id="cb3-68" data-line-number="68">  <span class="st">&quot;HuMERS_172-06_2015&quot;</span></a>
<a class="sourceLine" id="cb3-69" data-line-number="69">)</a>
<a class="sourceLine" id="cb3-70" data-line-number="70"></a>
<a class="sourceLine" id="cb3-71" data-line-number="71"><span class="co">#### Add GISAID IDs to the query pattern ####</span></a>
<a class="sourceLine" id="cb3-72" data-line-number="72"><span class="co">## Note that GISAID sequences are be submitted to the github repo because they cannot be redistributed</span></a>
<a class="sourceLine" id="cb3-73" data-line-number="73"><span class="cf">if</span> (useGISAID) {</a>
<a class="sourceLine" id="cb3-74" data-line-number="74">  <span class="cf">for</span> (ref <span class="cf">in</span> <span class="kw">names</span>(queryPatterns))</a>
<a class="sourceLine" id="cb3-75" data-line-number="75">  queryPatterns[[ref]] &lt;-<span class="st"> </span><span class="kw">append</span>(queryPatterns[[ref]], </a>
<a class="sourceLine" id="cb3-76" data-line-number="76">                          <span class="kw">c</span>(<span class="st">&quot;BtYu-RmYN02_2019&quot;</span>,</a>
<a class="sourceLine" id="cb3-77" data-line-number="77">                            <span class="st">&quot;PnGu1_2019&quot;</span></a>
<a class="sourceLine" id="cb3-78" data-line-number="78">                          ))</a>
<a class="sourceLine" id="cb3-79" data-line-number="79">}</a>
<a class="sourceLine" id="cb3-80" data-line-number="80"></a>
<a class="sourceLine" id="cb3-81" data-line-number="81"><span class="co"># message(&quot;\tReference strain: &quot;, refPattern)</span></a>
<a class="sourceLine" id="cb3-82" data-line-number="82"><span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Query patterns&quot;</span>)</a>
<a class="sourceLine" id="cb3-83" data-line-number="83"><span class="cf">for</span> (ref <span class="cf">in</span> <span class="kw">names</span>(queryPatterns)) {</a>
<a class="sourceLine" id="cb3-84" data-line-number="84">  queryPatterns[[ref]] &lt;-<span class="st"> </span><span class="kw">unique</span>(queryPatterns[[ref]])</a>
<a class="sourceLine" id="cb3-85" data-line-number="85">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, ref, <span class="st">&quot;</span><span class="ch">\t\t</span><span class="st">&quot;</span>, <span class="kw">length</span>(queryPatterns[[ref]]),</a>
<a class="sourceLine" id="cb3-86" data-line-number="86">          <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, queryPatterns[[ref]]))</a>
<a class="sourceLine" id="cb3-87" data-line-number="87">}</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">#### Load sequences ####</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">sequences &lt;-<span class="st"> </span><span class="kw">readAAStringSet</span>(<span class="dt">filepath =</span> infiles<span class="op">$</span>sequences, <span class="dt">format =</span> <span class="st">&quot;fasta&quot;</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">## Shorten sequence names by suppressing the fasta comment (after the space)</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="kw">names</span>(sequences) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">&quot; .*&quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="dt">x =</span> <span class="kw">names</span>(sequences), <span class="dt">perl =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-6" data-line-number="6"></a>
<a class="sourceLine" id="cb4-7" data-line-number="7">sequencesNames &lt;-<span class="st"> </span><span class="kw">names</span>(sequences)</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">nbsequences &lt;-<span class="st"> </span><span class="kw">length</span>(sequencesNames)</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="kw">message</span>(<span class="st">&quot;Loaded &quot;</span>, nbsequences, <span class="st">&quot; sequences from file &quot;</span>, infiles<span class="op">$</span>sequences)</a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co"># View(sequences)</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#### Define reference and query sequences ####</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13">refSequenceNames &lt;-<span class="st"> </span><span class="kw">vector</span>()</a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="cf">for</span> (ref <span class="cf">in</span> <span class="kw">names</span>(queryPatterns)) {</a>
<a class="sourceLine" id="cb4-15" data-line-number="15">  refSequenceNames[ref] &lt;-<span class="st"> </span><span class="kw">unique</span>(</a>
<a class="sourceLine" id="cb4-16" data-line-number="16">    <span class="kw">grep</span>(<span class="dt">pattern =</span> ref, <span class="dt">x =</span> <span class="kw">names</span>(sequences), </a>
<a class="sourceLine" id="cb4-17" data-line-number="17">         <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb4-18" data-line-number="18">    <span class="cf">if</span> (<span class="kw">is.null</span>(refSequenceNames[ref])) {</a>
<a class="sourceLine" id="cb4-19" data-line-number="19">    <span class="kw">stop</span>(<span class="st">&quot;Could not identify reference sequences with pattern &quot;</span>, ref)</a>
<a class="sourceLine" id="cb4-20" data-line-number="20">  }</a>
<a class="sourceLine" id="cb4-21" data-line-number="21">}</a>
<a class="sourceLine" id="cb4-22" data-line-number="22"></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">## Query sequences</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24">querySequenceNames &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="cf">for</span> (ref <span class="cf">in</span> <span class="kw">names</span>(queryPatterns)) {</a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#  message(&quot;Identifying query sequences for reference: &quot;, ref)</span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27">  queryRegExp &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;(&quot;</span>, <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;)|(&quot;</span>, queryPatterns[[ref]]), <span class="st">&quot;)&quot;</span>)</a>
<a class="sourceLine" id="cb4-28" data-line-number="28">  querySequenceNames[[ref]] &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> queryRegExp, </a>
<a class="sourceLine" id="cb4-29" data-line-number="29">                             <span class="dt">x =</span> sequencesNames, </a>
<a class="sourceLine" id="cb4-30" data-line-number="30">                             <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-31" data-line-number="31">  nbquerySeq &lt;-<span class="st"> </span><span class="kw">length</span>(querySequenceNames[[ref]])</a>
<a class="sourceLine" id="cb4-32" data-line-number="32">  </a>
<a class="sourceLine" id="cb4-33" data-line-number="33">  <span class="cf">if</span> (nbquerySeq <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) {</a>
<a class="sourceLine" id="cb4-34" data-line-number="34">    <span class="kw">stop</span>(<span class="st">&quot;Could not identify any query sequences with query pattern</span><span class="ch">\n</span><span class="st">&quot;</span>, queryRegExp)</a>
<a class="sourceLine" id="cb4-35" data-line-number="35">  }</a>
<a class="sourceLine" id="cb4-36" data-line-number="36">  </a>
<a class="sourceLine" id="cb4-37" data-line-number="37">  <span class="cf">if</span> (<span class="kw">length</span>(<span class="kw">unlist</span>(queryPatterns[ref[]])) <span class="op">!=</span><span class="st"> </span><span class="kw">length</span>(querySequenceNames[[ref]])) {</a>
<a class="sourceLine" id="cb4-38" data-line-number="38">    foundPatterns &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="dt">pattern =</span> queryRegExp, <span class="dt">x =</span> querySequenceNames[[ref]], <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-39" data-line-number="39">    missingPatterns &lt;-<span class="st"> </span><span class="kw">setdiff</span>(foundPatterns, queryPatterns[[ref]])</a>
<a class="sourceLine" id="cb4-40" data-line-number="40">    <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </a>
<a class="sourceLine" id="cb4-41" data-line-number="41">            <span class="kw">length</span>(missingPatterns), <span class="st">&quot; Missing sequences: &quot;</span>, </a>
<a class="sourceLine" id="cb4-42" data-line-number="42">            <span class="kw">paste</span>(<span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>, missingPatterns))</a>
<a class="sourceLine" id="cb4-43" data-line-number="43">  }</a>
<a class="sourceLine" id="cb4-44" data-line-number="44">  </a>
<a class="sourceLine" id="cb4-45" data-line-number="45">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Found &quot;</span>, </a>
<a class="sourceLine" id="cb4-46" data-line-number="46">          <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="kw">length</span>(querySequenceNames[[ref]]), </a>
<a class="sourceLine" id="cb4-47" data-line-number="47">          <span class="st">&quot; queries for</span><span class="ch">\t</span><span class="st">&quot;</span>, ref)</a>
<a class="sourceLine" id="cb4-48" data-line-number="48"></a>
<a class="sourceLine" id="cb4-49" data-line-number="49">  <span class="co">## Compute some statistics about sequences sizes</span></a>
<a class="sourceLine" id="cb4-50" data-line-number="50">  sequencestat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb4-51" data-line-number="51">    <span class="dt">row.names =</span> <span class="kw">c</span>(ref, querySequenceNames[[ref]]),</a>
<a class="sourceLine" id="cb4-52" data-line-number="52">    <span class="dt">status =</span> <span class="kw">c</span>(<span class="st">&quot;Reference&quot;</span>, </a>
<a class="sourceLine" id="cb4-53" data-line-number="53">               <span class="kw">rep</span>(<span class="st">&quot;Query&quot;</span>, <span class="dt">length.out =</span> <span class="kw">length</span>(querySequenceNames[[ref]])))</a>
<a class="sourceLine" id="cb4-54" data-line-number="54">  )</a>
<a class="sourceLine" id="cb4-55" data-line-number="55">  </a>
<a class="sourceLine" id="cb4-56" data-line-number="56">  g &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb4-57" data-line-number="57">  <span class="cf">for</span> (g <span class="cf">in</span> <span class="kw">c</span>(ref, querySequenceNames[[ref]])) {</a>
<a class="sourceLine" id="cb4-58" data-line-number="58">    sequencestat[g, <span class="st">&quot;length&quot;</span>] &lt;-<span class="st"> </span><span class="kw">length</span>(sequences[[g]])</a>
<a class="sourceLine" id="cb4-59" data-line-number="59">  }</a>
<a class="sourceLine" id="cb4-60" data-line-number="60">  <span class="kw">kable</span>(sequencestat, <span class="dt">caption =</span> <span class="kw">paste0</span>(<span class="st">&quot;Query sequences for reference &quot;</span>, ref))</a>
<a class="sourceLine" id="cb4-61" data-line-number="61">}</a></code></pre></div>
<p>The collection all-plus-GISAID contains 67 virus sequences sequences.</p>
<div id="n-to-1-full-sequence-alignments" class="section level2">
<h2>N-to-1 full sequence alignments</h2>
<p>We perform a pairwise lignment between each sequences query and the reference sequences (HuCoV2_WH01_2019).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">sequencesNto1 &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">refSequenceName &lt;-<span class="st"> &quot;HuCoV2_WH01_2019&quot;</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="cf">for</span> (refSequenceName <span class="cf">in</span> refSequenceNames) {</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">  <span class="co">## Define output file for sequences alignments</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5">  outfile &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">    dir<span class="op">$</span>outseq, <span class="kw">paste0</span>(<span class="st">&quot;one-to-n_alignments_ref_&quot;</span>, refSequenceName))</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">  outfiles[<span class="kw">paste0</span>(refSequenceName, <span class="st">&quot; alignments&quot;</span>)] &lt;-<span class="st"> </span>outfile</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">  </a>
<a class="sourceLine" id="cb5-9" data-line-number="9">  <span class="co">## Get sequences for reference and query sequences</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10">  refSequence &lt;-<span class="st"> </span>sequences[refSequenceName]</a>
<a class="sourceLine" id="cb5-11" data-line-number="11">  querySeq &lt;-<span class="st"> </span>sequences[querySequenceNames[[refSequenceName]]]</a>
<a class="sourceLine" id="cb5-12" data-line-number="12">  </a>
<a class="sourceLine" id="cb5-13" data-line-number="13">  <span class="kw">message</span>(<span class="st">&quot;</span><span class="ch">\t</span><span class="st">Aligning &quot;</span>, <span class="kw">length</span>(querySeq), <span class="st">&quot; query sequences&quot;</span>,</a>
<a class="sourceLine" id="cb5-14" data-line-number="14">          <span class="st">&quot; to reference</span><span class="ch">\t</span><span class="st">&quot;</span>, refSequenceName)</a>
<a class="sourceLine" id="cb5-15" data-line-number="15"></a>
<a class="sourceLine" id="cb5-16" data-line-number="16">  sequencesNto1[[refSequenceName]] &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb5-17" data-line-number="17">    <span class="dt">refSequence =</span> refSequence,  </a>
<a class="sourceLine" id="cb5-18" data-line-number="18">    <span class="dt">querySequences =</span> querySeq, </a>
<a class="sourceLine" id="cb5-19" data-line-number="19">    <span class="dt">sortByPIP =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb5-20" data-line-number="20">    <span class="co">#  querySequences = sequences, </span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21">    <span class="dt">outfile =</span> outfile)</a>
<a class="sourceLine" id="cb5-22" data-line-number="22">  </a>
<a class="sourceLine" id="cb5-23" data-line-number="23">}</a></code></pre></div>
</div>
<div id="alignment-with-sars-cov-2ref-strain-hucov2_wh01_2019" class="section level2">
<h2>Alignment with SARS-CoV-2(ref strain HuCoV2_WH01_2019)</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">refSequenceName &lt;-<span class="st"> &quot;HuCoV2_WH01_2019&quot;</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">kable</span>(sequencesNto1[[refSequenceName]]<span class="op">$</span>stats[<span class="kw">order</span>(sequencesNto1[[refSequenceName]]<span class="op">$</span>stats<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), ], </a>
<a class="sourceLine" id="cb6-3" data-line-number="3">        <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of full sequences&quot;</span>)</a></code></pre></div>
<table>
<caption>N-to-one alignment of full sequences</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">97.40770</td>
<td align="right">1273</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5129.657</td>
</tr>
<tr class="even">
<td>PnGX-P1E_2017</td>
<td align="right">92.30165</td>
<td align="right">1273</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4433.333</td>
</tr>
<tr class="odd">
<td>PnGu1_2019</td>
<td align="right">90.18838</td>
<td align="right">1274</td>
<td align="right">5</td>
<td align="right">11</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">4131.933</td>
</tr>
<tr class="even">
<td>BtZC45</td>
<td align="right">80.87774</td>
<td align="right">1276</td>
<td align="right">10</td>
<td align="right">30</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">2872.985</td>
</tr>
<tr class="odd">
<td>BtZXC21</td>
<td align="right">80.28169</td>
<td align="right">1278</td>
<td align="right">11</td>
<td align="right">33</td>
<td align="right">3</td>
<td align="right">5</td>
<td align="right">2788.446</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="right">76.44757</td>
<td align="right">1278</td>
<td align="right">8</td>
<td align="right">23</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">2269.462</td>
</tr>
<tr class="odd">
<td>BtYu-RmYN02_2019</td>
<td align="right">73.53172</td>
<td align="right">1277</td>
<td align="right">16</td>
<td align="right">50</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">1883.293</td>
</tr>
<tr class="even">
<td>CmMERS</td>
<td align="right">27.35703</td>
<td align="right">1294</td>
<td align="right">45</td>
<td align="right">548</td>
<td align="right">10</td>
<td align="right">21</td>
<td align="right">-3617.178</td>
</tr>
<tr class="odd">
<td>HuMERS_172-06_2015</td>
<td align="right">27.35703</td>
<td align="right">1294</td>
<td align="right">45</td>
<td align="right">548</td>
<td align="right">10</td>
<td align="right">21</td>
<td align="right">-3617.178</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co">## PIP profile of full sequences N-to-1 alignments</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">  <span class="dt">alignments =</span> sequencesNto1[[refSequenceName]]<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb7-4" data-line-number="4">  <span class="dt">windowSize =</span> <span class="dv">100</span>, </a>
<a class="sourceLine" id="cb7-5" data-line-number="5">  <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Full sequences PIP profile&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, refSequenceName),</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">  <span class="dt">legendMargin =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb7-7" data-line-number="7">  <span class="dt">legendCorner =</span> <span class="st">&quot;bottomright&quot;</span>, <span class="dt">legendCex =</span> <span class="fl">0.6</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/pip_profiles/prot_PIP_SARS-CoV-2-1.png" alt="Percent Identical Positions profile over the whole sequences of SARS-CoV-2. " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole sequences of SARS-CoV-2.
</p>
</div>
</div>
<div id="alignment-with-sars-cov-ref-strain-husars-frankfurt-1_2003" class="section level2">
<h2>Alignment with SARS-CoV (ref strain: HuSARS-Frankfurt-1_2003)</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">refSequenceName &lt;-<span class="st"> &quot;HuSARS-Frankfurt-1_2003&quot;</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">kable</span>(sequencesNto1[[refSequenceName]]<span class="op">$</span>stats[<span class="kw">order</span>(sequencesNto1[[refSequenceName]]<span class="op">$</span>stats<span class="op">$</span>score, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), ], </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">        <span class="dt">caption =</span> <span class="st">&quot;N-to-one alignment of full sequences&quot;</span>)</a></code></pre></div>
<table>
<caption>N-to-one alignment of full sequences</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CvSZ3</td>
<td align="right">99.92032</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5390.603</td>
</tr>
<tr class="even">
<td>BtRs4874</td>
<td align="right">96.97211</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5000.518</td>
</tr>
<tr class="odd">
<td>BtWIV16_2013</td>
<td align="right">96.89243</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4989.975</td>
</tr>
<tr class="even">
<td>BtRs4231</td>
<td align="right">94.58167</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4684.237</td>
</tr>
<tr class="odd">
<td>BtRs7327</td>
<td align="right">92.43028</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4399.568</td>
</tr>
<tr class="even">
<td>BtRsSHC014</td>
<td align="right">89.96016</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4072.753</td>
</tr>
<tr class="odd">
<td>Btrec-SARSp_2008</td>
<td align="right">83.82236</td>
<td align="right">1261</td>
<td align="right">3</td>
<td align="right">31</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">3309.095</td>
</tr>
<tr class="even">
<td>PnGX-P1E_2017</td>
<td align="right">77.20126</td>
<td align="right">1272</td>
<td align="right">6</td>
<td align="right">7</td>
<td align="right">5</td>
<td align="right">17</td>
<td align="right">2360.653</td>
</tr>
<tr class="odd">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">76.92308</td>
<td align="right">1274</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">19</td>
<td align="right">2337.090</td>
</tr>
<tr class="even">
<td>BtZXC21</td>
<td align="right">76.59574</td>
<td align="right">1269</td>
<td align="right">8</td>
<td align="right">26</td>
<td align="right">7</td>
<td align="right">14</td>
<td align="right">2269.767</td>
</tr>
<tr class="odd">
<td>HuCoV2_WH01_2019</td>
<td align="right">76.44757</td>
<td align="right">1278</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">8</td>
<td align="right">23</td>
<td align="right">2269.462</td>
</tr>
<tr class="even">
<td>BtZC45</td>
<td align="right">76.51694</td>
<td align="right">1269</td>
<td align="right">8</td>
<td align="right">25</td>
<td align="right">6</td>
<td align="right">14</td>
<td align="right">2266.986</td>
</tr>
<tr class="odd">
<td>PnGu1_2019</td>
<td align="right">76.18297</td>
<td align="right">1268</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">13</td>
<td align="right">2223.492</td>
</tr>
<tr class="even">
<td>BtYu-RmYN02_2019</td>
<td align="right">72.53968</td>
<td align="right">1260</td>
<td align="right">9</td>
<td align="right">33</td>
<td align="right">3</td>
<td align="right">5</td>
<td align="right">1739.962</td>
</tr>
<tr class="odd">
<td>CmMERS</td>
<td align="right">29.54545</td>
<td align="right">1276</td>
<td align="right">61</td>
<td align="right">479</td>
<td align="right">9</td>
<td align="right">21</td>
<td align="right">-3566.885</td>
</tr>
<tr class="even">
<td>HuMERS_172-06_2015</td>
<td align="right">29.54545</td>
<td align="right">1276</td>
<td align="right">61</td>
<td align="right">479</td>
<td align="right">9</td>
<td align="right">21</td>
<td align="right">-3566.885</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co">## PIP profile of full sequences N-to-1 alignments</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> sequencesNto1[[refSequenceName]]<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb9-3" data-line-number="3">                <span class="dt">windowSize =</span> <span class="dv">100</span>, </a>
<a class="sourceLine" id="cb9-4" data-line-number="4">                <span class="co">#                legend = paste0(names(sequencesNto1[[refSequenceName]]$alignments), &quot; (&quot;, round(digits = 2, sequencesNto1[[refSequenceName]]$stats$pid), &quot;%)&quot;),</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5">                <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Full sequences PIP profile&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, refSequenceName),</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">                <span class="dt">legendMargin =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb9-7" data-line-number="7">                <span class="dt">legendCorner =</span> <span class="st">&quot;bottomright&quot;</span>, <span class="dt">legendCex =</span> <span class="fl">0.6</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/pip_profiles/prot_PIP_SARS-CoV-1.png" alt="Percent Identical Positions profile over the whole sequences of SARS-CoV. " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole sequences of SARS-CoV.
</p>
</div>
</div>
<div id="aligning-all-sequences" class="section level2">
<h2>Aligning all sequences</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co">## Define output file for sequences alignments</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">outfiles[<span class="st">&quot;Sequence alignments - SARS&quot;</span>] &lt;-<span class="st"> </span><span class="kw">file.path</span>(</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">  dir<span class="op">$</span>outseq, <span class="kw">paste0</span>(<span class="st">&quot;spike-protein_alignments_ref_&quot;</span>,</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">                     <span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>))</a>
<a class="sourceLine" id="cb10-5" data-line-number="5"></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#### Compare SARS (2002) wih the related Civet sequences ####</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7">SARSvsAll &lt;-<span class="st"> </span><span class="kw">alignNtoOne</span>(</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">  <span class="dt">refSequence =</span> sequences[<span class="st">&quot;HuSARS-Frankfurt-1_2003&quot;</span>], </a>
<a class="sourceLine" id="cb10-9" data-line-number="9">  <span class="dt">querySeq =</span> sequences, <span class="dt">sortByPIP =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb10-10" data-line-number="10">  <span class="dt">outfile =</span> outfiles[<span class="st">&quot;Sequence alignments - SARS&quot;</span>] )</a>
<a class="sourceLine" id="cb10-11" data-line-number="11"></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="kw">kable</span>(SARSvsAll<span class="op">$</span>stats, </a>
<a class="sourceLine" id="cb10-13" data-line-number="13">      <span class="dt">caption =</span> <span class="st">&quot;SARS (2003). N-to-one alignment of full sequences of the closest animal virus (Civet) and of Human SARS-CoV-2. &quot;</span>)</a></code></pre></div>
<table>
<caption>SARS (2003). N-to-one alignment of full sequences of the closest animal virus (Civet) and of Human SARS-CoV-2.</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pid</th>
<th align="right">nchar</th>
<th align="right">insertNb</th>
<th align="right">insertLen</th>
<th align="right">delNb</th>
<th align="right">delLen</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HuSARS-Frankfurt-1_2003</td>
<td align="right">100.00000</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5401.146</td>
</tr>
<tr class="even">
<td>CvSZ3</td>
<td align="right">99.92032</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5390.603</td>
</tr>
<tr class="odd">
<td>HuSARS-CoV_Tor2-FP1-10895</td>
<td align="right">99.92032</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5390.603</td>
</tr>
<tr class="even">
<td>HuTW11-SARS</td>
<td align="right">99.92032</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5390.603</td>
</tr>
<tr class="odd">
<td>Cv007-2004</td>
<td align="right">98.32669</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5179.747</td>
</tr>
<tr class="even">
<td>BtRs4874</td>
<td align="right">96.97211</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5000.518</td>
</tr>
<tr class="odd">
<td>BtWIV16_2013</td>
<td align="right">96.89243</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4989.975</td>
</tr>
<tr class="even">
<td>BtRs4231</td>
<td align="right">94.58167</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4684.237</td>
</tr>
<tr class="odd">
<td>BtRs7327</td>
<td align="right">92.43028</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4399.568</td>
</tr>
<tr class="even">
<td>BtYN2018B</td>
<td align="right">92.35060</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4389.024</td>
</tr>
<tr class="odd">
<td>BtRs3367</td>
<td align="right">92.19124</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4367.940</td>
</tr>
<tr class="even">
<td>BtWIV1</td>
<td align="right">92.19124</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4367.940</td>
</tr>
<tr class="odd">
<td>BtRsSHC014</td>
<td align="right">89.96016</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4072.753</td>
</tr>
<tr class="even">
<td>BtRs4084</td>
<td align="right">89.88048</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4062.212</td>
</tr>
<tr class="odd">
<td>BtLYRa11</td>
<td align="right">89.56175</td>
<td align="right">1255</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4020.048</td>
</tr>
<tr class="even">
<td>Btrec-SARSp_2008</td>
<td align="right">83.82236</td>
<td align="right">1261</td>
<td align="right">3</td>
<td align="right">31</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">3309.095</td>
</tr>
<tr class="odd">
<td>BtYN2013</td>
<td align="right">79.84064</td>
<td align="right">1255</td>
<td align="right">6</td>
<td align="right">25</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2729.811</td>
</tr>
<tr class="even">
<td>BtSC2018</td>
<td align="right">79.73291</td>
<td align="right">1273</td>
<td align="right">8</td>
<td align="right">32</td>
<td align="right">5</td>
<td align="right">18</td>
<td align="right">2740.540</td>
</tr>
<tr class="odd">
<td>BtCp-Yun_2011</td>
<td align="right">79.69865</td>
<td align="right">1261</td>
<td align="right">7</td>
<td align="right">20</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">2706.244</td>
</tr>
<tr class="even">
<td>BtRs-HuB-2013</td>
<td align="right">79.40016</td>
<td align="right">1267</td>
<td align="right">9</td>
<td align="right">52</td>
<td align="right">4</td>
<td align="right">12</td>
<td align="right">2714.435</td>
</tr>
<tr class="odd">
<td>BtRp-Shaanxi2011</td>
<td align="right">79.24230</td>
<td align="right">1267</td>
<td align="right">10</td>
<td align="right">53</td>
<td align="right">4</td>
<td align="right">12</td>
<td align="right">2685.589</td>
</tr>
<tr class="even">
<td>BtRs4081</td>
<td align="right">79.14692</td>
<td align="right">1266</td>
<td align="right">8</td>
<td align="right">27</td>
<td align="right">3</td>
<td align="right">11</td>
<td align="right">2640.289</td>
</tr>
<tr class="odd">
<td>BtGX2013</td>
<td align="right">79.14354</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">48</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2675.137</td>
</tr>
<tr class="even">
<td>BtRm1/2004</td>
<td align="right">78.98493</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">54</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2667.486</td>
</tr>
<tr class="odd">
<td>BtRs_672-2006</td>
<td align="right">78.83096</td>
<td align="right">1266</td>
<td align="right">8</td>
<td align="right">27</td>
<td align="right">3</td>
<td align="right">11</td>
<td align="right">2598.118</td>
</tr>
<tr class="even">
<td>BtRs4237</td>
<td align="right">78.82633</td>
<td align="right">1261</td>
<td align="right">9</td>
<td align="right">49</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2625.205</td>
</tr>
<tr class="odd">
<td>BtRs4247</td>
<td align="right">78.82633</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">48</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2632.966</td>
</tr>
<tr class="even">
<td>BtYN2018C</td>
<td align="right">78.82633</td>
<td align="right">1261</td>
<td align="right">9</td>
<td align="right">49</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2625.205</td>
</tr>
<tr class="odd">
<td>BtRp3-2004</td>
<td align="right">78.74703</td>
<td align="right">1261</td>
<td align="right">9</td>
<td align="right">49</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2614.663</td>
</tr>
<tr class="even">
<td>BtHKU3-8</td>
<td align="right">78.66772</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">48</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2611.881</td>
</tr>
<tr class="odd">
<td>BtHKU3-12</td>
<td align="right">78.58842</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">48</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2601.338</td>
</tr>
<tr class="even">
<td>BtHKU3-6</td>
<td align="right">78.42982</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">48</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2580.253</td>
</tr>
<tr class="odd">
<td>BtLongquan-140</td>
<td align="right">78.35052</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">48</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">2569.710</td>
</tr>
<tr class="even">
<td>BtYNLF_31C</td>
<td align="right">77.87470</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">39</td>
<td align="right">3</td>
<td align="right">6</td>
<td align="right">2476.302</td>
</tr>
<tr class="odd">
<td>PnGX-P1E_2017</td>
<td align="right">77.20126</td>
<td align="right">1272</td>
<td align="right">6</td>
<td align="right">7</td>
<td align="right">5</td>
<td align="right">17</td>
<td align="right">2360.653</td>
</tr>
<tr class="even">
<td>PnGX-P4L</td>
<td align="right">77.04403</td>
<td align="right">1272</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">17</td>
<td align="right">2355.090</td>
</tr>
<tr class="odd">
<td>PnGX-P5L</td>
<td align="right">77.04403</td>
<td align="right">1272</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">17</td>
<td align="right">2355.090</td>
</tr>
<tr class="even">
<td>BtCoV-273-2005</td>
<td align="right">77.00238</td>
<td align="right">1261</td>
<td align="right">9</td>
<td align="right">39</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">2370.331</td>
</tr>
<tr class="odd">
<td>BtRf-HeB-2013</td>
<td align="right">77.00238</td>
<td align="right">1261</td>
<td align="right">7</td>
<td align="right">39</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">2390.331</td>
</tr>
<tr class="even">
<td>BtRaTG13_2013_Yunnan</td>
<td align="right">76.92308</td>
<td align="right">1274</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">19</td>
<td align="right">2337.090</td>
</tr>
<tr class="odd">
<td>BtRf1-2004</td>
<td align="right">76.92308</td>
<td align="right">1261</td>
<td align="right">8</td>
<td align="right">39</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">2369.789</td>
</tr>
<tr class="even">
<td>PnGX-P5E</td>
<td align="right">76.88679</td>
<td align="right">1272</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">17</td>
<td align="right">2334.004</td>
</tr>
<tr class="odd">
<td>PnGX-P2V_2018</td>
<td align="right">76.68760</td>
<td align="right">1274</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">19</td>
<td align="right">2305.462</td>
</tr>
<tr class="even">
<td>BtZXC21</td>
<td align="right">76.59574</td>
<td align="right">1269</td>
<td align="right">8</td>
<td align="right">26</td>
<td align="right">7</td>
<td align="right">14</td>
<td align="right">2269.767</td>
</tr>
<tr class="odd">
<td>BtZC45</td>
<td align="right">76.51694</td>
<td align="right">1269</td>
<td align="right">8</td>
<td align="right">25</td>
<td align="right">6</td>
<td align="right">14</td>
<td align="right">2266.986</td>
</tr>
<tr class="even">
<td>HuCoV2_WH01_2019</td>
<td align="right">76.44757</td>
<td align="right">1278</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">8</td>
<td align="right">23</td>
<td align="right">2269.462</td>
</tr>
<tr class="odd">
<td>HuCoV2_Whu1_2019</td>
<td align="right">76.44757</td>
<td align="right">1278</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">8</td>
<td align="right">23</td>
<td align="right">2269.462</td>
</tr>
<tr class="even">
<td>HuCoV2_WIV04°2019</td>
<td align="right">76.44757</td>
<td align="right">1278</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">8</td>
<td align="right">23</td>
<td align="right">2269.462</td>
</tr>
<tr class="odd">
<td>PnGu1_2019</td>
<td align="right">76.18297</td>
<td align="right">1268</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">13</td>
<td align="right">2223.492</td>
</tr>
<tr class="even">
<td>BtBtKY72</td>
<td align="right">75.59242</td>
<td align="right">1266</td>
<td align="right">9</td>
<td align="right">11</td>
<td align="right">5</td>
<td align="right">11</td>
<td align="right">2100.040</td>
</tr>
<tr class="odd">
<td>BtJTMC15</td>
<td align="right">75.15873</td>
<td align="right">1260</td>
<td align="right">5</td>
<td align="right">24</td>
<td align="right">2</td>
<td align="right">5</td>
<td align="right">2117.721</td>
</tr>
<tr class="even">
<td>BtBM48-31</td>
<td align="right">75.11848</td>
<td align="right">1266</td>
<td align="right">8</td>
<td align="right">29</td>
<td align="right">5</td>
<td align="right">11</td>
<td align="right">2087.086</td>
</tr>
<tr class="odd">
<td>BtJL2012</td>
<td align="right">75.07937</td>
<td align="right">1260</td>
<td align="right">5</td>
<td align="right">24</td>
<td align="right">2</td>
<td align="right">5</td>
<td align="right">2107.178</td>
</tr>
<tr class="even">
<td>BtYu-RmYN02_2019</td>
<td align="right">72.53968</td>
<td align="right">1260</td>
<td align="right">9</td>
<td align="right">33</td>
<td align="right">3</td>
<td align="right">5</td>
<td align="right">1739.962</td>
</tr>
<tr class="odd">
<td>BtHKU5</td>
<td align="right">34.96183</td>
<td align="right">1310</td>
<td align="right">32</td>
<td align="right">191</td>
<td align="right">27</td>
<td align="right">55</td>
<td align="right">-3383.760</td>
</tr>
<tr class="even">
<td>BtHKU9-1</td>
<td align="right">33.35895</td>
<td align="right">1301</td>
<td align="right">43</td>
<td align="right">249</td>
<td align="right">14</td>
<td align="right">46</td>
<td align="right">-3450.922</td>
</tr>
<tr class="odd">
<td>CmMERS</td>
<td align="right">29.54545</td>
<td align="right">1276</td>
<td align="right">61</td>
<td align="right">479</td>
<td align="right">9</td>
<td align="right">21</td>
<td align="right">-3566.885</td>
</tr>
<tr class="even">
<td>HuMERS_172-06_2015</td>
<td align="right">29.54545</td>
<td align="right">1276</td>
<td align="right">61</td>
<td align="right">479</td>
<td align="right">9</td>
<td align="right">21</td>
<td align="right">-3566.885</td>
</tr>
<tr class="odd">
<td>HuHKU1_N23</td>
<td align="right">28.19107</td>
<td align="right">1277</td>
<td align="right">55</td>
<td align="right">440</td>
<td align="right">13</td>
<td align="right">22</td>
<td align="right">-3817.429</td>
</tr>
<tr class="even">
<td>HuTGEV</td>
<td align="right">27.44648</td>
<td align="right">1308</td>
<td align="right">69</td>
<td align="right">464</td>
<td align="right">18</td>
<td align="right">53</td>
<td align="right">-4088.233</td>
</tr>
<tr class="odd">
<td>HuHK04-02</td>
<td align="right">27.30829</td>
<td align="right">1278</td>
<td align="right">55</td>
<td align="right">539</td>
<td align="right">10</td>
<td align="right">23</td>
<td align="right">-3685.739</td>
</tr>
<tr class="even">
<td>Btrec-SARSg_2008</td>
<td align="right">27.23005</td>
<td align="right">1278</td>
<td align="right">55</td>
<td align="right">539</td>
<td align="right">10</td>
<td align="right">23</td>
<td align="right">-3696.281</td>
</tr>
<tr class="odd">
<td>HuOC43</td>
<td align="right">27.23005</td>
<td align="right">1278</td>
<td align="right">55</td>
<td align="right">539</td>
<td align="right">10</td>
<td align="right">23</td>
<td align="right">-3696.281</td>
</tr>
<tr class="even">
<td>PiSADS</td>
<td align="right">26.09034</td>
<td align="right">1284</td>
<td align="right">63</td>
<td align="right">477</td>
<td align="right">17</td>
<td align="right">29</td>
<td align="right">-4146.150</td>
</tr>
<tr class="odd">
<td>PiPRCV</td>
<td align="right">25.90038</td>
<td align="right">1305</td>
<td align="right">62</td>
<td align="right">503</td>
<td align="right">15</td>
<td align="right">50</td>
<td align="right">-4110.309</td>
</tr>
<tr class="even">
<td>Hu229E</td>
<td align="right">25.64886</td>
<td align="right">1310</td>
<td align="right">57</td>
<td align="right">554</td>
<td align="right">16</td>
<td align="right">55</td>
<td align="right">-3997.205</td>
</tr>
<tr class="odd">
<td>HuNL63</td>
<td align="right">24.29907</td>
<td align="right">1284</td>
<td align="right">74</td>
<td align="right">607</td>
<td align="right">13</td>
<td align="right">29</td>
<td align="right">-4167.559</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co">## PIP profile of full sequences N-to-1 alignments</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">plotPIPprofiles</span>(<span class="dt">alignments =</span> SARSvsAll<span class="op">$</span>alignments, </a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                <span class="dt">windowSize =</span> <span class="dv">500</span>, </a>
<a class="sourceLine" id="cb11-4" data-line-number="4">                <span class="dt">legend =</span> <span class="kw">paste0</span>(<span class="kw">names</span>(SARSvsAll<span class="op">$</span>alignments), <span class="st">&quot; (&quot;</span>, <span class="kw">round</span>(<span class="dt">digits =</span> <span class="dv">2</span>, SARSvsAll<span class="op">$</span>stats<span class="op">$</span>pid), <span class="st">&quot;%)&quot;</span>),</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Spike protein PIP&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">Ref: &quot;</span>, <span class="st">&quot;Human_SARS-CoV_Frankfurt_1&quot;</span>),</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">                <span class="dt">legendMargin =</span> <span class="fl">0.3</span>, </a>
<a class="sourceLine" id="cb11-7" data-line-number="7">                <span class="dt">legendCorner =</span> <span class="st">&quot;bottomright&quot;</span>, </a>
<a class="sourceLine" id="cb11-8" data-line-number="8">                <span class="dt">legendCex =</span> <span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb11-9" data-line-number="9">                <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">100</span>))</a></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/pip_profiles/prot_PIP_SARS_vs_all-1.png" alt="Percent Identical Positions profile over the whole sequences of SARS (2002-2003). " width="100%" />
<p class="caption">
Percent Identical Positions profile over the whole sequences of SARS (2002-2003).
</p>
</div>
<!-- ## Spike protein against 2003 SARS-CoV (Frankfurt strain) -->
<!-- ```{r S-gene_align_queries} -->
<!-- featureName <- "Spike-protein_SARS-CoV" -->
<!-- featureLimits <- features[[featureName]] -->
<!-- #### N-to-1 alignments of spike-coding sequences #### -->
<!-- dir[[featureName]] <- file.path(dir$main, "results", featureName) -->
<!-- dir.create(featureName, showWarnings = FALSE, recursive = TRUE) -->
<!-- outfiles[featureName]  <- file.path( -->
<!--   dir[[featureName]], paste0(featureName, "_", collection, "_alignments")) -->
<!-- ## Get sequences for reference and query sequences -->
<!-- refSeq <- subseq(sequences[refSequenceName], start = featureLimits[1], end = featureLimits[2]) -->
<!-- featureAlignmentsNto1 <- alignNtoOne( -->
<!--   refSequence = refSeq,  -->
<!--   querySeq = querySeq,  -->
<!--   type = "global-local", sortByPIP = TRUE, -->
<!--   outfile = outfiles[featureName]) -->
<!-- kable(featureAlignmentsNto1$stats[order(featureAlignmentsNto1$stat$score, decreasing = TRUE),],  -->
<!--       caption = "N-to-one alignment of S genes") -->
<!-- ``` -->
<!-- ### Spike gene PIP plot -->
<!-- ```{r Sgene_pip, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole sequences of SARS-CoV-2. "} -->
<!-- ## PIP profile of spike N-to-1 alignments -->
<!-- plotPIPprofiles(alignments = featureAlignmentsNto1$alignments,  -->
<!--                 windowSize = 200,  -->
<!-- #                legend = paste0(names(featureAlignmentsNto1$alignments), " (", round(digits = 2, featureAlignmentsNto1$stats$pid), "%)"), -->
<!--                 main = paste0(featureName, " - PIP profile", "\nRef: ", refSequenceName), -->
<!--                 legendMargin = 0, legendCorner = "bottomright", legendCex = 0.8, ylim = c(30, 100)) -->
<!-- ``` -->
<!-- ### Getting sequences of spike gene in all the selected coronavirus sequences -->
<!-- ```{r  match sequence fragment against all sequences and export the matches to a fasta file} -->
<!-- #' @title Export sequencesquery + subject sequences from N-to-1 alignments -->
<!-- #' @author Jacques.van-Helden@france)bioinformatique.fr -->
<!-- #' @param refsequence  sequence used as reference (DB) for the N-to-1 alignment -->
<!-- #' @param sequences sequences sequences -->
<!-- #' @param IDsuffix=NULL if not null, append the suffix to the sequence name for the fasta header -->
<!-- #' @param outfile=NULL output file. If specified, the matches will be exported as fasta-formatted sequences. -->
<!-- #' @export -->
<!-- ExportSegmentMatches <- function(refsequence, -->
<!--                                  sequences, -->
<!--                                  IDsuffix = NULL, -->
<!--                                  outfile=NULL) { -->
<!--   #### N-to-1 alignments of all spike-coding sequences #### -->
<!--   ## Get sequences for reference and query sequences -->
<!--   alignmentsNto1 <- alignNtoOne( -->
<!--     refSequence = refsequence,  -->
<!--     querySeq = sequences,  -->
<!--     type = "global-local") -->
<!--   if (!is.null(outfile)) { -->
<!--     ## Write the reference sequence in the output fle -->
<!--     writeXStringSet(refsequence, filepath = outfile, format = "fasta") -->
<!--     i <- 1 -->
<!--     nbAlignments <- length(alignmentsNto1$alignments) -->
<!--     for (i in 1:nbAlignments) { -->
<!--       sequencesName <- names(sequences[i]) -->
<!--       alignment <- alignmentsNto1$alignments[[i]] -->
<!--       subject <- subject(alignment) -->
<!--       ## Suppress the dashes from the alignment to get the raw sequence -->
<!--       sequence <- as.character(subject) -->
<!--       sequenceDesaligned <- gsub(pattern = "-", replacement = "", x = sequence) -->
<!--       seqStringSet <- DNAStringSet(x = sequenceDesaligned) #, start = start(subject), end=end(subject)) -->
<!--       ## Define a sequence ID for the fasta header -->
<!--       sequenceID <- sequencesName -->
<!--       if (!is.null(IDsuffix)) { -->
<!--         sequenceID <- paste0(sequenceID, IDsuffix) -->
<!--       }  -->
<!--       sequenceID <- paste0(sequenceID, "_", start(subject), "-", end(subject)) -->
<!--       names(seqStringSet) <- sequenceID -->
<!--       #  -->
<!--       ## Write pairwise alignment (temporarily disaactivated) -->
<!--       # alignmentFile <- paste0("pairwise-alignment_",  -->
<!--       #                         # gsub(pattern = "/", replacement = "-", x = sequencesName),  -->
<!--       #                         ".txt") -->
<!--       # writePairwiseAlignments(x = alignment, file = outfile) -->
<!--       ## Append the sequence to the file -->
<!--       message("\tAppending sequence ", i, "/", nbAlignments, "\t", sequenceID) -->
<!--       writeXStringSet(seqStringSet, -->
<!--                       filepath = outfile, format = "fasta", append = TRUE) -->
<!--     } -->
<!--     message("\tExported alignments to\t", outfile) -->
<!--   }  -->
<!--   return(alignmentsNto1) -->
<!-- } -->
<!-- ``` -->
<!-- ```{r Sgene_exportseq} -->
<!-- #### Export S gene sequences found in the reference sequences #### -->
<!-- # toString(refSeq) -->
<!-- writeXStringSet( -->
<!--   refSeq,  -->
<!--   filepath = file.path( -->
<!--     dir$outseq,  -->
<!--     paste0("S-gene_", refSequenceName, ".fasta")),  -->
<!--   format = "fasta") -->
<!-- ## Export matches to S-gene -->
<!-- outfiles["S gene matches"] <- file.path( -->
<!--   dir$outseq,  -->
<!--   paste0("S-gene_", refSequenceName, "_matches.fasta")) -->
<!-- featureAlignmentsNto1All <- ExportSegmentMatches( -->
<!--   refsequence = refSeq, -->
<!--   sequences = sequences,  -->
<!--   IDsuffix = "_S-gene", -->
<!--   outfile = outfiles["S gene matches"]) -->
<!-- kable(featureAlignmentsNto1All$stats[order(featureAlignmentsNto1All$stat$score, decreasing = TRUE),],  -->
<!--         caption = "N-to-one alignment of S genes") -->
<!-- ``` -->
<!-- ## Export matches to the coding sequence of the S1 cleavage produc -->
<!-- ## N-to-1 alignemnts of S1 coding sequences -->
<!-- ```{r S1_align_queries} -->
<!-- refS1Limits <- refSpikeLimits[1] + 3*c(13, 685) -->
<!-- #### N-to-1 alignments of S1-coding sequences #### -->
<!-- ## Get sequences for reference and query sequences -->
<!-- refS1 <- subseq(sequences[refSequenceName], start = refS1Limits[1], end = refS1Limits[2]) -->
<!-- S1Nto1 <- alignNtoOne(refSequence = refS1, querySeq = querySeq, type = "global-local") -->
<!-- kable(S1Nto1$stats[order(S1Nto1$stat$score, decreasing = TRUE),],  -->
<!--       caption = "N-to-one alignment of S1 coding sequence") -->
<!-- ``` -->
<!-- ### S1 gene PIP plot -->
<!-- ```{r S1_pip, fig.width=10, fig.height=5, out.width="100%", fig.cap="Percent Identical Positions profile over the whole sequences of SARS-CoV-2. "} -->
<!-- ## PIP profile of S1 N-to-1 alignments -->
<!-- plotPIPprofiles(alignments = S1Nto1$alignments,  -->
<!--                 windowSize = 200,  -->
<!--                 legend = paste0(names(S1Nto1$alignments), " (", round(digits = 2, S1Nto1$stats$pid), "%)"), -->
<!--                 main = paste0("S1 coding sequence - PIP profile", "\nRef: ", refSequenceName), -->
<!--                 legendMargin = 0, legendCorner = "bottomleft", legendCex = 0.8, ylim = c(30, 100)) -->
<!-- ``` -->
</div>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">kable</span>(<span class="kw">t</span>(<span class="kw">as.data.frame</span>(dir)), <span class="dt">col.names =</span> <span class="st">&quot;Dir&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Directories&quot;</span>)</a></code></pre></div>
<table>
<caption>Directories</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">Dir</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>main</td>
<td align="left">..</td>
</tr>
<tr class="even">
<td>R</td>
<td align="left">../scripts/R</td>
</tr>
<tr class="odd">
<td>outseq</td>
<td align="left">../results/spike_proteins/PIP_profiles</td>
</tr>
<tr class="even">
<td>sequences</td>
<td align="left">../data/GISAID_genomes</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">kable</span>(outfiles, <span class="dt">col.names =</span> <span class="st">&quot;File&quot;</span>, <span class="dt">caption =</span> <span class="st">&quot;Output files&quot;</span>)</a></code></pre></div>
<table>
<caption>Output files</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">File</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HuCoV2_WH01_2019 alignments</td>
<td align="left">../results/spike_proteins/PIP_profiles/one-to-n_alignments_ref_HuCoV2_WH01_2019</td>
</tr>
<tr class="even">
<td>HuSARS-Frankfurt-1_2003 alignments</td>
<td align="left">../results/spike_proteins/PIP_profiles/one-to-n_alignments_ref_HuSARS-Frankfurt-1_2003</td>
</tr>
<tr class="odd">
<td>Sequence alignments - SARS</td>
<td align="left">../results/spike_proteins/PIP_profiles/spike-protein_alignments_ref_HuSARS-Frankfurt-1_2003</td>
</tr>
</tbody>
</table>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Biostrings_2.52.0   XVector_0.24.0      IRanges_2.18.3      S4Vectors_0.22.1    BiocGenerics_0.30.0 knitr_1.28         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4          digest_0.6.25       magrittr_1.5        evaluate_0.14       highr_0.8           zlibbioc_1.30.0     rlang_0.4.5         stringi_1.4.6       rmarkdown_2.1       tools_3.6.1         stringr_1.4.0       xfun_0.12           yaml_2.2.1          compiler_3.6.1     
[15] BiocManager_1.30.10 htmltools_0.4.0    </code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
